{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "76e413f3-bf9f-4e2f-8758-0379f2a2b0cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pickle as pkl\n",
    "from gprofiler import GProfiler\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8f33c235-07ec-46e3-894f-c07305806dd2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# loading a pickle file for visualization\n",
    "#with open (\"C:\\Users\\Echeh\\Downloads\\Masters Pipeline\\mergedbf_unusual_marked_with_20000cluster_label (1).pkl\", 'rb') as file:\n",
    "  #  data = pickle.load(file)\n",
    "print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "638b51eb-04e8-4bcd-9d06-85a8660a4871",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>SH</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>PIEZ1_CAEEL_10_VVKLLLPAALLAAAI</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>17001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>PIEZ1_CAEEL_15_LPAALLAAAIIRPSF</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425155</th>\n",
       "      <td>95</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>EHPRLHSYSVTPSAP</td>\n",
       "      <td>X5ML99_CAEEL_95_EHPRLHSYSVTPSAP</td>\n",
       "      <td>EHPRLHSYSVTPSAP</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>54</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425156</th>\n",
       "      <td>100</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>HSYSVTPSAPAMETA</td>\n",
       "      <td>X5ML99_CAEEL_100_HSYSVTPSAPAMETA</td>\n",
       "      <td>HSYSVTPSAPAMETA</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425157</th>\n",
       "      <td>105</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>TPSAPAMETAFGHDY</td>\n",
       "      <td>X5ML99_CAEEL_105_TPSAPAMETAFGHDY</td>\n",
       "      <td>TPSAPAMETAFGHDY</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425158</th>\n",
       "      <td>110</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>AMETAFGHDYDPPPS</td>\n",
       "      <td>X5ML99_CAEEL_110_AMETAFGHDYDPPPS</td>\n",
       "      <td>AMETAFGHDYDPPPS</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425159</th>\n",
       "      <td>115</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>FGHDYDPPPSYTEAM</td>\n",
       "      <td>X5ML99_CAEEL_115_FGHDYDPPPSYTEAM</td>\n",
       "      <td>FGHDYDPPPSYTEAM</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>18760</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2425160 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start       Entry    Entry name  \\\n",
       "0            0  A0A061ACU2   PIEZ1_CAEEL   \n",
       "1            5  A0A061ACU2   PIEZ1_CAEEL   \n",
       "2           10  A0A061ACU2   PIEZ1_CAEEL   \n",
       "3           15  A0A061ACU2   PIEZ1_CAEEL   \n",
       "4           20  A0A061ACU2   PIEZ1_CAEEL   \n",
       "...        ...         ...           ...   \n",
       "2425155     95      X5ML99  X5ML99_CAEEL   \n",
       "2425156    100      X5ML99  X5ML99_CAEEL   \n",
       "2425157    105      X5ML99  X5ML99_CAEEL   \n",
       "2425158    110      X5ML99  X5ML99_CAEEL   \n",
       "2425159    115      X5ML99  X5ML99_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "0        Piezo-type mechanosensitive ion channel compon...  MTVPPLLKSCVVKLL   \n",
       "1        Piezo-type mechanosensitive ion channel compon...  LLKSCVVKLLLPAAL   \n",
       "2        Piezo-type mechanosensitive ion channel compon...  VVKLLLPAALLAAAI   \n",
       "3        Piezo-type mechanosensitive ion channel compon...  LPAALLAAAIIRPSF   \n",
       "4        Piezo-type mechanosensitive ion channel compon...  LAAAIIRPSFLSIGY   \n",
       "...                                                    ...              ...   \n",
       "2425155                 EGF-like domain-containing protein  EHPRLHSYSVTPSAP   \n",
       "2425156                 EGF-like domain-containing protein  HSYSVTPSAPAMETA   \n",
       "2425157                 EGF-like domain-containing protein  TPSAPAMETAFGHDY   \n",
       "2425158                 EGF-like domain-containing protein  AMETAFGHDYDPPPS   \n",
       "2425159                 EGF-like domain-containing protein  FGHDYDPPPSYTEAM   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "0           PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL  MTVPPLLKSCVVKLL       0       4   \n",
       "1           PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL  LLKSCVVKLLLPAAL       0       2   \n",
       "2          PIEZ1_CAEEL_10_VVKLLLPAALLAAAI  VVKLLLPAALLAAAI       0       2   \n",
       "3          PIEZ1_CAEEL_15_LPAALLAAAIIRPSF  LPAALLAAAIIRPSF       0       4   \n",
       "4          PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY  LAAAIIRPSFLSIGY       1       2   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2425155   X5ML99_CAEEL_95_EHPRLHSYSVTPSAP  EHPRLHSYSVTPSAP       0       6   \n",
       "2425156  X5ML99_CAEEL_100_HSYSVTPSAPAMETA  HSYSVTPSAPAMETA       0       4   \n",
       "2425157  X5ML99_CAEEL_105_TPSAPAMETAFGHDY  TPSAPAMETAFGHDY       1       4   \n",
       "2425158  X5ML99_CAEEL_110_AMETAFGHDYDPPPS  AMETAFGHDYDPPPS       1       6   \n",
       "2425159  X5ML99_CAEEL_115_FGHDYDPPPSYTEAM  FGHDYDPPPSYTEAM       1       6   \n",
       "\n",
       "         CB  ...  SH  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \n",
       "0        13  ...   1  1    2    0  56  False  False  False     False   14908  \n",
       "1        14  ...   1  0    2    0  58  False  False  False     False    7734  \n",
       "2        14  ...   0  0    1    0  58  False  False  False     False   17001  \n",
       "3        13  ...   0  0    0    4  56  False  False  False     False    7557  \n",
       "4        13  ...   0  0    0    4  57  False  False  False     False    8636  \n",
       "...      ..  ...  .. ..  ...  ...  ..    ...    ...    ...       ...     ...  \n",
       "2425155  12  ...   0  0    0    4  54  False  False  False     False    8981  \n",
       "2425156  13  ...   0  1    0    0  56  False  False  False     False   16050  \n",
       "2425157  12  ...   0  1    0    0  55  False  False  False     False   16050  \n",
       "2425158  11  ...   0  1    0    0  53  False  False  False     False    7246  \n",
       "2425159  11  ...   0  1    0    0  53  False  False  False     False   18760  \n",
       "\n",
       "[2425160 rows x 31 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# loading a pickle file into pandas for visualization\n",
    "df_pkl  = pd.read_pickle (r\"C:\\Users\\Echeh\\Downloads\\Masters Pipeline\\mergedbf_unusual_marked_with_20000cluster_label (1).pkl\")\n",
    "df_pkl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "971dea0d-f514-4281-8c03-d838fae1c25c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#reading the other dataframe to get the column Gene Names (Primary)\n",
    "pf=pd.read_excel(r\"C:\\Users\\Echeh\\Downloads\\Masters Pipeline\\uniprotkb_C_elegans_AND_model_organism_2025_01_20.xlsx\")\n",
    "pf=pf[[\"Gene Names (primary)\",\"Entry\"]].set_index(\"Entry\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "65cd2df9-8de1-40fd-a9a6-8c94b898b032",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Joined the pf and df_pkl dataframes, using entry column\n",
    "df_pkl=df_pkl.join(pf, on=\"Entry\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c400184b-449b-48ce-9153-5dd359c2e19a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# indexed the dataframe df_pkl to give us peptides that have entry name with PIEZ1_CAEEL only\n",
    "#index = df_pkl[\"Entry name\"] == \"PIEZ1_CAEEL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0c4c457d-ce29-4bc9-be8e-cd29cc2c532c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_pkl[index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "48a36aef-3452-4f0a-87d0-8974892c3762",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Gene Names (primary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10947</th>\n",
       "      <td>4155</td>\n",
       "      <td>G4SLH0</td>\n",
       "      <td>TTN1_CAEEL</td>\n",
       "      <td>Titin homolog (EC 2.7.11.1)</td>\n",
       "      <td>ATLKCIVVGTPLPDV</td>\n",
       "      <td>TTN1_CAEEL_4155_ATLKCIVVGTPLPDV</td>\n",
       "      <td>ATLKCIVVGTPLPDV</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>ttn-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96923</th>\n",
       "      <td>330</td>\n",
       "      <td>P34271</td>\n",
       "      <td>CUP4_CAEEL</td>\n",
       "      <td>Acetylcholine receptor-like protein cup-4 (Coe...</td>\n",
       "      <td>PLSVSEIPFCIKLIG</td>\n",
       "      <td>CUP4_CAEEL_330_PLSVSEIPFCIKLIG</td>\n",
       "      <td>PLSVSEIPFCIKLIG</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>cup-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234219</th>\n",
       "      <td>30</td>\n",
       "      <td>Q9XUC2</td>\n",
       "      <td>IFT22_CAEEL</td>\n",
       "      <td>Intraflagellar transport associated protein 2</td>\n",
       "      <td>ILVLGPPKAGKTTLC</td>\n",
       "      <td>IFT22_CAEEL_30_ILVLGPPKAGKTTLC</td>\n",
       "      <td>ILVLGPPKAGKTTLC</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>ifta-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236662</th>\n",
       "      <td>150</td>\n",
       "      <td>Q9XW92</td>\n",
       "      <td>VATA_CAEEL</td>\n",
       "      <td>V-type proton ATPase catalytic subunit A (V-AT...</td>\n",
       "      <td>IKHKILLPPSACGTI</td>\n",
       "      <td>VATA_CAEEL_150_IKHKILLPPSACGTI</td>\n",
       "      <td>IKHKILLPPSACGTI</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>vha-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2347550</th>\n",
       "      <td>580</td>\n",
       "      <td>Q9XW73</td>\n",
       "      <td>Q9XW73_CAEEL</td>\n",
       "      <td>Mediator of RNA polymerase II transcription su...</td>\n",
       "      <td>TPFIIVCIKSPSLLV</td>\n",
       "      <td>Q9XW73_CAEEL_580_TPFIIVCIKSPSLLV</td>\n",
       "      <td>TPFIIVCIKSPSLLV</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2360262</th>\n",
       "      <td>30</td>\n",
       "      <td>Q9XXF0</td>\n",
       "      <td>Q9XXF0_CAEEL</td>\n",
       "      <td>Uncharacterized protein</td>\n",
       "      <td>LTPIITAQCSKKPKG</td>\n",
       "      <td>Q9XXF0_CAEEL_30_LTPIITAQCSKKPKG</td>\n",
       "      <td>LTPIITAQCSKKPKG</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2360816</th>\n",
       "      <td>305</td>\n",
       "      <td>Q9XXI0</td>\n",
       "      <td>Q9XXI0_CAEEL</td>\n",
       "      <td>Arrestin C-terminal-like domain-containing pro...</td>\n",
       "      <td>KKLRCELPITVGTIP</td>\n",
       "      <td>Q9XXI0_CAEEL_305_KKLRCELPITVGTIP</td>\n",
       "      <td>KKLRCELPITVGTIP</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>arrd-8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2390435</th>\n",
       "      <td>135</td>\n",
       "      <td>U4PR82</td>\n",
       "      <td>U4PR82_CAEEL</td>\n",
       "      <td>Pyr_redox_2 domain-containing protein</td>\n",
       "      <td>VVACGKTSPRKLQLP</td>\n",
       "      <td>U4PR82_CAEEL_135_VVACGKTSPRKLQLP</td>\n",
       "      <td>VVACGKTSPRKLQLP</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>osgn-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2398461</th>\n",
       "      <td>205</td>\n",
       "      <td>V6CK14</td>\n",
       "      <td>V6CK14_CAEEL</td>\n",
       "      <td>FBD domain-containing protein</td>\n",
       "      <td>CIVNLGIPKTKTFPL</td>\n",
       "      <td>V6CK14_CAEEL_205_CIVNLGIPKTKTFPL</td>\n",
       "      <td>CIVNLGIPKTKTFPL</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>him-19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start       Entry    Entry name  \\\n",
       "0            0  A0A061ACU2   PIEZ1_CAEEL   \n",
       "10947     4155      G4SLH0    TTN1_CAEEL   \n",
       "96923      330      P34271    CUP4_CAEEL   \n",
       "234219      30      Q9XUC2   IFT22_CAEEL   \n",
       "236662     150      Q9XW92    VATA_CAEEL   \n",
       "...        ...         ...           ...   \n",
       "2347550    580      Q9XW73  Q9XW73_CAEEL   \n",
       "2360262     30      Q9XXF0  Q9XXF0_CAEEL   \n",
       "2360816    305      Q9XXI0  Q9XXI0_CAEEL   \n",
       "2390435    135      U4PR82  U4PR82_CAEEL   \n",
       "2398461    205      V6CK14  V6CK14_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "0        Piezo-type mechanosensitive ion channel compon...  MTVPPLLKSCVVKLL   \n",
       "10947                          Titin homolog (EC 2.7.11.1)  ATLKCIVVGTPLPDV   \n",
       "96923    Acetylcholine receptor-like protein cup-4 (Coe...  PLSVSEIPFCIKLIG   \n",
       "234219       Intraflagellar transport associated protein 2  ILVLGPPKAGKTTLC   \n",
       "236662   V-type proton ATPase catalytic subunit A (V-AT...  IKHKILLPPSACGTI   \n",
       "...                                                    ...              ...   \n",
       "2347550  Mediator of RNA polymerase II transcription su...  TPFIIVCIKSPSLLV   \n",
       "2360262                            Uncharacterized protein  LTPIITAQCSKKPKG   \n",
       "2360816  Arrestin C-terminal-like domain-containing pro...  KKLRCELPITVGTIP   \n",
       "2390435              Pyr_redox_2 domain-containing protein  VVACGKTSPRKLQLP   \n",
       "2398461                      FBD domain-containing protein  CIVNLGIPKTKTFPL   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "0           PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL  MTVPPLLKSCVVKLL       0       4   \n",
       "10947     TTN1_CAEEL_4155_ATLKCIVVGTPLPDV  ATLKCIVVGTPLPDV       1       4   \n",
       "96923      CUP4_CAEEL_330_PLSVSEIPFCIKLIG  PLSVSEIPFCIKLIG       1       4   \n",
       "234219     IFT22_CAEEL_30_ILVLGPPKAGKTTLC  ILVLGPPKAGKTTLC       2       4   \n",
       "236662     VATA_CAEEL_150_IKHKILLPPSACGTI  IKHKILLPPSACGTI       1       4   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2347550  Q9XW73_CAEEL_580_TPFIIVCIKSPSLLV  TPFIIVCIKSPSLLV       0       4   \n",
       "2360262   Q9XXF0_CAEEL_30_LTPIITAQCSKKPKG  LTPIITAQCSKKPKG       1       4   \n",
       "2360816  Q9XXI0_CAEEL_305_KKLRCELPITVGTIP  KKLRCELPITVGTIP       1       4   \n",
       "2390435  U4PR82_CAEEL_135_VVACGKTSPRKLQLP  VVACGKTSPRKLQLP       1       4   \n",
       "2398461  V6CK14_CAEEL_205_CIVNLGIPKTKTFPL  CIVNLGIPKTKTFPL       1       4   \n",
       "\n",
       "         CB  ...  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \\\n",
       "0        13  ...  1    2    0  56  False  False  False     False   14908   \n",
       "10947    12  ...  0    1    0  55  False  False  False     False   14908   \n",
       "96923    12  ...  0    1    0  55  False  False  False     False   14908   \n",
       "234219   11  ...  0    2    0  54   True  False  False     False   14908   \n",
       "236662   12  ...  0    2    0  55  False  False  False     False   14908   \n",
       "...      ..  ... ..  ...  ...  ..    ...    ...    ...       ...     ...   \n",
       "2347550  13  ...  0    1    0  56  False  False  False     False   14908   \n",
       "2360262  12  ...  0    3    0  55  False  False  False     False   14908   \n",
       "2360816  12  ...  0    2    4  55  False  False  False     False   14908   \n",
       "2390435  12  ...  0    2    4  55  False  False  False     False   14908   \n",
       "2398461  12  ...  0    2    0  55  False  False  False     False   14908   \n",
       "\n",
       "         Gene Names (primary)  \n",
       "0                      pezo-1  \n",
       "10947                   ttn-1  \n",
       "96923                   cup-4  \n",
       "234219                 ifta-2  \n",
       "236662                 vha-13  \n",
       "...                       ...  \n",
       "2347550                   NaN  \n",
       "2360262                   NaN  \n",
       "2360816                arrd-8  \n",
       "2390435                osgn-1  \n",
       "2398461                him-19  \n",
       "\n",
       "[95 rows x 32 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# indexed the dataframe df_pkl[ bravo] to give us peptides that have Kclust equal to 14908 only\n",
    "bravo = df_pkl[\"Kclust\"] == 14908\n",
    "df_pkl[bravo]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4efe32f8-d31e-490d-b970-174269009253",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MYO2_CAEEL\n",
      "8493  does not have myos.\n",
      "pixel_MYO2_CAEEL_0.xlsx\n",
      "pixel_MYO2_CAEEL_5.xlsx\n",
      "5650  does not have myos.\n",
      "pixel_MYO2_CAEEL_10.xlsx\n",
      "11549  does not have myos.\n",
      "pixel_MYO2_CAEEL_15.xlsx\n",
      "pixel_MYO2_CAEEL_20.xlsx\n",
      "15658  does not have myos.\n",
      "pixel_MYO2_CAEEL_25.xlsx\n",
      "5377  does not have myos.\n",
      "pixel_MYO2_CAEEL_30.xlsx\n",
      "2241  does not have myos.\n",
      "pixel_MYO2_CAEEL_35.xlsx\n",
      "pixel_MYO2_CAEEL_40.xlsx\n",
      "9889  does not have myos.\n",
      "pixel_MYO2_CAEEL_45.xlsx\n",
      "9310  does not have myos.\n",
      "pixel_MYO2_CAEEL_50.xlsx\n",
      "pixel_MYO2_CAEEL_55.xlsx\n",
      "pixel_MYO2_CAEEL_60.xlsx\n",
      "6643  does not have myos.\n",
      "pixel_MYO2_CAEEL_65.xlsx\n",
      "3553  does not have myos.\n",
      "pixel_MYO2_CAEEL_70.xlsx\n",
      "pixel_MYO2_CAEEL_75.xlsx\n",
      "pixel_MYO2_CAEEL_80.xlsx\n",
      "10161  does not have myos.\n",
      "pixel_MYO2_CAEEL_85.xlsx\n",
      "18521  does not have myos.\n",
      "pixel_MYO2_CAEEL_90.xlsx\n",
      "486  does not have myos.\n",
      "pixel_MYO2_CAEEL_95.xlsx\n",
      "pixel_MYO2_CAEEL_100.xlsx\n",
      "pixel_MYO2_CAEEL_105.xlsx\n",
      "pixel_MYO2_CAEEL_110.xlsx\n",
      "pixel_MYO2_CAEEL_115.xlsx\n",
      "7327  does not have myos.\n",
      "pixel_MYO2_CAEEL_120.xlsx\n",
      "4642  does not have myos.\n",
      "pixel_MYO2_CAEEL_125.xlsx\n",
      "8406  does not have myos.\n",
      "pixel_MYO2_CAEEL_130.xlsx\n",
      "pixel_MYO2_CAEEL_135.xlsx\n",
      "pixel_MYO2_CAEEL_140.xlsx\n",
      "pixel_MYO2_CAEEL_145.xlsx\n",
      "12095  does not have myos.\n",
      "pixel_MYO2_CAEEL_150.xlsx\n",
      "pixel_MYO2_CAEEL_155.xlsx\n",
      "pixel_MYO2_CAEEL_160.xlsx\n",
      "pixel_MYO2_CAEEL_165.xlsx\n",
      "pixel_MYO2_CAEEL_170.xlsx\n",
      "pixel_MYO2_CAEEL_175.xlsx\n",
      "pixel_MYO2_CAEEL_180.xlsx\n",
      "pixel_MYO2_CAEEL_185.xlsx\n",
      "pixel_MYO2_CAEEL_190.xlsx\n",
      "pixel_MYO2_CAEEL_195.xlsx\n",
      "12307  does not have myos.\n",
      "pixel_MYO2_CAEEL_200.xlsx\n",
      "pixel_MYO2_CAEEL_205.xlsx\n",
      "13163  does not have myos.\n",
      "pixel_MYO2_CAEEL_210.xlsx\n",
      "17005  does not have myos.\n",
      "pixel_MYO2_CAEEL_215.xlsx\n",
      "pixel_MYO2_CAEEL_220.xlsx\n",
      "7222  does not have myos.\n",
      "pixel_MYO2_CAEEL_225.xlsx\n",
      "pixel_MYO2_CAEEL_230.xlsx\n",
      "pixel_MYO2_CAEEL_235.xlsx\n",
      "pixel_MYO2_CAEEL_240.xlsx\n",
      "3207  does not have myos.\n",
      "pixel_MYO2_CAEEL_245.xlsx\n",
      "14477  does not have myos.\n",
      "pixel_MYO2_CAEEL_250.xlsx\n",
      "5569  does not have myos.\n",
      "pixel_MYO2_CAEEL_255.xlsx\n",
      "8557  does not have myos.\n",
      "pixel_MYO2_CAEEL_260.xlsx\n",
      "pixel_MYO2_CAEEL_265.xlsx\n",
      "7884  does not have myos.\n",
      "pixel_MYO2_CAEEL_270.xlsx\n",
      "6170  does not have myos.\n",
      "pixel_MYO2_CAEEL_275.xlsx\n",
      "4950  does not have myos.\n",
      "pixel_MYO2_CAEEL_280.xlsx\n",
      "pixel_MYO2_CAEEL_285.xlsx\n",
      "4788  does not have myos.\n",
      "pixel_MYO2_CAEEL_290.xlsx\n",
      "pixel_MYO2_CAEEL_295.xlsx\n",
      "3048  does not have myos.\n",
      "pixel_MYO2_CAEEL_300.xlsx\n",
      "pixel_MYO2_CAEEL_305.xlsx\n",
      "13670  does not have myos.\n",
      "pixel_MYO2_CAEEL_310.xlsx\n",
      "796  does not have myos.\n",
      "pixel_MYO2_CAEEL_315.xlsx\n",
      "pixel_MYO2_CAEEL_320.xlsx\n",
      "1701  does not have myos.\n",
      "pixel_MYO2_CAEEL_325.xlsx\n",
      "3307  does not have myos.\n",
      "pixel_MYO2_CAEEL_330.xlsx\n",
      "15343  does not have myos.\n",
      "pixel_MYO2_CAEEL_335.xlsx\n",
      "pixel_MYO2_CAEEL_340.xlsx\n",
      "4074  does not have myos.\n",
      "pixel_MYO2_CAEEL_345.xlsx\n",
      "pixel_MYO2_CAEEL_350.xlsx\n",
      "13452  does not have myos.\n",
      "pixel_MYO2_CAEEL_355.xlsx\n",
      "14451  does not have myos.\n",
      "pixel_MYO2_CAEEL_360.xlsx\n",
      "pixel_MYO2_CAEEL_365.xlsx\n",
      "8127  does not have myos.\n",
      "pixel_MYO2_CAEEL_370.xlsx\n",
      "1181  does not have myos.\n",
      "pixel_MYO2_CAEEL_375.xlsx\n",
      "8039  does not have myos.\n",
      "pixel_MYO2_CAEEL_380.xlsx\n",
      "13111  does not have myos.\n",
      "pixel_MYO2_CAEEL_385.xlsx\n",
      "16447  does not have myos.\n",
      "pixel_MYO2_CAEEL_390.xlsx\n",
      "14980  does not have myos.\n",
      "pixel_MYO2_CAEEL_395.xlsx\n",
      "18773  does not have myos.\n",
      "pixel_MYO2_CAEEL_400.xlsx\n",
      "pixel_MYO2_CAEEL_405.xlsx\n",
      "2439  does not have myos.\n",
      "pixel_MYO2_CAEEL_410.xlsx\n",
      "2439  does not have myos.\n",
      "pixel_MYO2_CAEEL_415.xlsx\n",
      "12716  does not have myos.\n",
      "pixel_MYO2_CAEEL_420.xlsx\n",
      "713  does not have myos.\n",
      "pixel_MYO2_CAEEL_425.xlsx\n",
      "4049  does not have myos.\n",
      "pixel_MYO2_CAEEL_430.xlsx\n",
      "14336  does not have myos.\n",
      "pixel_MYO2_CAEEL_435.xlsx\n",
      "2353  does not have myos.\n",
      "pixel_MYO2_CAEEL_440.xlsx\n",
      "12760  does not have myos.\n",
      "pixel_MYO2_CAEEL_445.xlsx\n",
      "pixel_MYO2_CAEEL_450.xlsx\n",
      "15200  does not have myos.\n",
      "pixel_MYO2_CAEEL_455.xlsx\n",
      "11943  does not have myos.\n",
      "pixel_MYO2_CAEEL_460.xlsx\n",
      "3656  does not have myos.\n",
      "pixel_MYO2_CAEEL_465.xlsx\n",
      "pixel_MYO2_CAEEL_470.xlsx\n",
      "pixel_MYO2_CAEEL_475.xlsx\n",
      "16951  does not have myos.\n",
      "pixel_MYO2_CAEEL_480.xlsx\n",
      "15083  does not have myos.\n",
      "pixel_MYO2_CAEEL_485.xlsx\n",
      "pixel_MYO2_CAEEL_490.xlsx\n",
      "12701  does not have myos.\n",
      "pixel_MYO2_CAEEL_495.xlsx\n",
      "pixel_MYO2_CAEEL_500.xlsx\n",
      "6736  does not have myos.\n",
      "pixel_MYO2_CAEEL_505.xlsx\n",
      "pixel_MYO2_CAEEL_510.xlsx\n",
      "pixel_MYO2_CAEEL_515.xlsx\n",
      "1906  does not have myos.\n",
      "pixel_MYO2_CAEEL_520.xlsx\n",
      "pixel_MYO2_CAEEL_525.xlsx\n",
      "pixel_MYO2_CAEEL_530.xlsx\n",
      "pixel_MYO2_CAEEL_535.xlsx\n",
      "5428  does not have myos.\n",
      "pixel_MYO2_CAEEL_540.xlsx\n",
      "pixel_MYO2_CAEEL_545.xlsx\n",
      "12721  does not have myos.\n",
      "pixel_MYO2_CAEEL_550.xlsx\n",
      "pixel_MYO2_CAEEL_555.xlsx\n",
      "7312  does not have myos.\n",
      "pixel_MYO2_CAEEL_560.xlsx\n",
      "pixel_MYO2_CAEEL_565.xlsx\n",
      "12629  does not have myos.\n",
      "pixel_MYO2_CAEEL_570.xlsx\n",
      "4615  does not have myos.\n",
      "pixel_MYO2_CAEEL_575.xlsx\n",
      "14035  does not have myos.\n",
      "pixel_MYO2_CAEEL_580.xlsx\n",
      "15681  does not have myos.\n",
      "pixel_MYO2_CAEEL_585.xlsx\n",
      "5823  does not have myos.\n",
      "pixel_MYO2_CAEEL_590.xlsx\n",
      "pixel_MYO2_CAEEL_595.xlsx\n",
      "3553  does not have myos.\n",
      "pixel_MYO2_CAEEL_600.xlsx\n",
      "16813  does not have myos.\n",
      "pixel_MYO2_CAEEL_605.xlsx\n",
      "14800  does not have myos.\n",
      "pixel_MYO2_CAEEL_610.xlsx\n",
      "19289  does not have myos.\n",
      "pixel_MYO2_CAEEL_615.xlsx\n",
      "7309  does not have myos.\n",
      "pixel_MYO2_CAEEL_620.xlsx\n",
      "pixel_MYO2_CAEEL_625.xlsx\n",
      "16458  does not have myos.\n",
      "pixel_MYO2_CAEEL_630.xlsx\n",
      "10979  does not have myos.\n",
      "pixel_MYO2_CAEEL_635.xlsx\n",
      "9317  does not have myos.\n",
      "pixel_MYO2_CAEEL_640.xlsx\n",
      "13702  does not have myos.\n",
      "pixel_MYO2_CAEEL_645.xlsx\n",
      "2251  does not have myos.\n",
      "pixel_MYO2_CAEEL_650.xlsx\n",
      "762  does not have myos.\n",
      "pixel_MYO2_CAEEL_655.xlsx\n",
      "19460  does not have myos.\n",
      "pixel_MYO2_CAEEL_660.xlsx\n",
      "pixel_MYO2_CAEEL_665.xlsx\n",
      "pixel_MYO2_CAEEL_670.xlsx\n",
      "19234  does not have myos.\n",
      "pixel_MYO2_CAEEL_675.xlsx\n",
      "10244  does not have myos.\n",
      "pixel_MYO2_CAEEL_680.xlsx\n",
      "pixel_MYO2_CAEEL_685.xlsx\n",
      "pixel_MYO2_CAEEL_690.xlsx\n",
      "pixel_MYO2_CAEEL_695.xlsx\n",
      "1642  does not have myos.\n",
      "pixel_MYO2_CAEEL_700.xlsx\n",
      "254  does not have myos.\n",
      "pixel_MYO2_CAEEL_705.xlsx\n",
      "7390  does not have myos.\n",
      "pixel_MYO2_CAEEL_710.xlsx\n",
      "18438  does not have myos.\n",
      "pixel_MYO2_CAEEL_715.xlsx\n",
      "pixel_MYO2_CAEEL_720.xlsx\n",
      "pixel_MYO2_CAEEL_725.xlsx\n",
      "16902  does not have myos.\n",
      "pixel_MYO2_CAEEL_730.xlsx\n",
      "pixel_MYO2_CAEEL_735.xlsx\n",
      "pixel_MYO2_CAEEL_740.xlsx\n",
      "1063  does not have myos.\n",
      "pixel_MYO2_CAEEL_745.xlsx\n",
      "211  does not have myos.\n",
      "pixel_MYO2_CAEEL_750.xlsx\n",
      "7803  does not have myos.\n",
      "pixel_MYO2_CAEEL_755.xlsx\n",
      "4331  does not have myos.\n",
      "pixel_MYO2_CAEEL_760.xlsx\n",
      "pixel_MYO2_CAEEL_765.xlsx\n",
      "pixel_MYO2_CAEEL_770.xlsx\n",
      "11324  does not have myos.\n",
      "pixel_MYO2_CAEEL_775.xlsx\n",
      "pixel_MYO2_CAEEL_780.xlsx\n",
      "pixel_MYO2_CAEEL_785.xlsx\n",
      "5952  does not have myos.\n",
      "pixel_MYO2_CAEEL_790.xlsx\n",
      "16411  does not have myos.\n",
      "pixel_MYO2_CAEEL_795.xlsx\n",
      "57  does not have myos.\n",
      "pixel_MYO2_CAEEL_800.xlsx\n",
      "19386  does not have myos.\n",
      "pixel_MYO2_CAEEL_805.xlsx\n",
      "14367  does not have myos.\n",
      "pixel_MYO2_CAEEL_810.xlsx\n",
      "6397  does not have myos.\n",
      "pixel_MYO2_CAEEL_815.xlsx\n",
      "11088  does not have myos.\n",
      "pixel_MYO2_CAEEL_820.xlsx\n",
      "14418  does not have myos.\n",
      "pixel_MYO2_CAEEL_825.xlsx\n",
      "pixel_MYO2_CAEEL_830.xlsx\n",
      "12834  does not have myos.\n",
      "pixel_MYO2_CAEEL_835.xlsx\n",
      "12567  does not have myos.\n",
      "pixel_MYO2_CAEEL_840.xlsx\n",
      "1380  does not have myos.\n",
      "pixel_MYO2_CAEEL_845.xlsx\n",
      "19599  does not have myos.\n",
      "pixel_MYO2_CAEEL_850.xlsx\n",
      "pixel_MYO2_CAEEL_855.xlsx\n",
      "2670  does not have myos.\n",
      "pixel_MYO2_CAEEL_860.xlsx\n",
      "14394  does not have myos.\n",
      "pixel_MYO2_CAEEL_865.xlsx\n",
      "14394  does not have myos.\n",
      "pixel_MYO2_CAEEL_870.xlsx\n",
      "13227  does not have myos.\n",
      "pixel_MYO2_CAEEL_875.xlsx\n",
      "pixel_MYO2_CAEEL_880.xlsx\n",
      "11969  does not have myos.\n",
      "pixel_MYO2_CAEEL_885.xlsx\n",
      "pixel_MYO2_CAEEL_890.xlsx\n",
      "18287  does not have myos.\n",
      "pixel_MYO2_CAEEL_895.xlsx\n",
      "pixel_MYO2_CAEEL_900.xlsx\n",
      "pixel_MYO2_CAEEL_905.xlsx\n",
      "11886  does not have myos.\n",
      "pixel_MYO2_CAEEL_910.xlsx\n",
      "16684  does not have myos.\n",
      "pixel_MYO2_CAEEL_915.xlsx\n",
      "pixel_MYO2_CAEEL_920.xlsx\n",
      "pixel_MYO2_CAEEL_925.xlsx\n",
      "pixel_MYO2_CAEEL_930.xlsx\n",
      "pixel_MYO2_CAEEL_935.xlsx\n",
      "pixel_MYO2_CAEEL_940.xlsx\n",
      "13227  does not have myos.\n",
      "pixel_MYO2_CAEEL_945.xlsx\n",
      "4754  does not have myos.\n",
      "pixel_MYO2_CAEEL_950.xlsx\n",
      "696  does not have myos.\n",
      "pixel_MYO2_CAEEL_955.xlsx\n",
      "9340  does not have myos.\n",
      "pixel_MYO2_CAEEL_960.xlsx\n",
      "pixel_MYO2_CAEEL_965.xlsx\n",
      "pixel_MYO2_CAEEL_970.xlsx\n",
      "pixel_MYO2_CAEEL_975.xlsx\n",
      "pixel_MYO2_CAEEL_980.xlsx\n",
      "pixel_MYO2_CAEEL_985.xlsx\n",
      "1143  does not have myos.\n",
      "pixel_MYO2_CAEEL_990.xlsx\n",
      "pixel_MYO2_CAEEL_995.xlsx\n",
      "3195  does not have myos.\n",
      "pixel_MYO2_CAEEL_1000.xlsx\n",
      "6779  does not have myos.\n",
      "pixel_MYO2_CAEEL_1005.xlsx\n",
      "pixel_MYO2_CAEEL_1010.xlsx\n",
      "pixel_MYO2_CAEEL_1015.xlsx\n",
      "pixel_MYO2_CAEEL_1020.xlsx\n",
      "11780  does not have myos.\n",
      "pixel_MYO2_CAEEL_1025.xlsx\n",
      "8563  does not have myos.\n",
      "pixel_MYO2_CAEEL_1030.xlsx\n",
      "4504  does not have myos.\n",
      "pixel_MYO2_CAEEL_1035.xlsx\n",
      "pixel_MYO2_CAEEL_1040.xlsx\n",
      "2828  does not have myos.\n",
      "pixel_MYO2_CAEEL_1045.xlsx\n",
      "10512  does not have myos.\n",
      "pixel_MYO2_CAEEL_1050.xlsx\n",
      "13351  does not have myos.\n",
      "pixel_MYO2_CAEEL_1055.xlsx\n",
      "12984  does not have myos.\n",
      "pixel_MYO2_CAEEL_1060.xlsx\n",
      "pixel_MYO2_CAEEL_1065.xlsx\n",
      "10471  does not have myos.\n",
      "pixel_MYO2_CAEEL_1070.xlsx\n",
      "pixel_MYO2_CAEEL_1075.xlsx\n",
      "pixel_MYO2_CAEEL_1080.xlsx\n",
      "pixel_MYO2_CAEEL_1085.xlsx\n",
      "pixel_MYO2_CAEEL_1090.xlsx\n",
      "1252  does not have myos.\n",
      "pixel_MYO2_CAEEL_1095.xlsx\n",
      "370  does not have myos.\n",
      "pixel_MYO2_CAEEL_1100.xlsx\n",
      "9719  does not have myos.\n",
      "pixel_MYO2_CAEEL_1105.xlsx\n",
      "pixel_MYO2_CAEEL_1110.xlsx\n",
      "pixel_MYO2_CAEEL_1115.xlsx\n",
      "pixel_MYO2_CAEEL_1120.xlsx\n",
      "pixel_MYO2_CAEEL_1125.xlsx\n",
      "pixel_MYO2_CAEEL_1130.xlsx\n",
      "16591  does not have myos.\n",
      "pixel_MYO2_CAEEL_1135.xlsx\n",
      "16591  does not have myos.\n",
      "pixel_MYO2_CAEEL_1140.xlsx\n",
      "pixel_MYO2_CAEEL_1145.xlsx\n",
      "12640  does not have myos.\n",
      "pixel_MYO2_CAEEL_1150.xlsx\n",
      "1285  does not have myos.\n",
      "pixel_MYO2_CAEEL_1155.xlsx\n",
      "pixel_MYO2_CAEEL_1160.xlsx\n",
      "1059  does not have myos.\n",
      "pixel_MYO2_CAEEL_1165.xlsx\n",
      "pixel_MYO2_CAEEL_1170.xlsx\n",
      "pixel_MYO2_CAEEL_1175.xlsx\n",
      "pixel_MYO2_CAEEL_1180.xlsx\n",
      "pixel_MYO2_CAEEL_1185.xlsx\n",
      "11611  does not have myos.\n",
      "pixel_MYO2_CAEEL_1190.xlsx\n",
      "pixel_MYO2_CAEEL_1195.xlsx\n",
      "pixel_MYO2_CAEEL_1200.xlsx\n",
      "16189  does not have myos.\n",
      "pixel_MYO2_CAEEL_1205.xlsx\n",
      "2471  does not have myos.\n",
      "pixel_MYO2_CAEEL_1210.xlsx\n",
      "pixel_MYO2_CAEEL_1215.xlsx\n",
      "16837  does not have myos.\n",
      "pixel_MYO2_CAEEL_1220.xlsx\n",
      "15271  does not have myos.\n",
      "pixel_MYO2_CAEEL_1225.xlsx\n",
      "7036  does not have myos.\n",
      "pixel_MYO2_CAEEL_1230.xlsx\n",
      "9182  does not have myos.\n",
      "pixel_MYO2_CAEEL_1235.xlsx\n",
      "7129  does not have myos.\n",
      "pixel_MYO2_CAEEL_1240.xlsx\n",
      "pixel_MYO2_CAEEL_1245.xlsx\n",
      "pixel_MYO2_CAEEL_1250.xlsx\n",
      "12127  does not have myos.\n",
      "pixel_MYO2_CAEEL_1255.xlsx\n",
      "16208  does not have myos.\n",
      "pixel_MYO2_CAEEL_1260.xlsx\n",
      "pixel_MYO2_CAEEL_1265.xlsx\n",
      "pixel_MYO2_CAEEL_1270.xlsx\n",
      "pixel_MYO2_CAEEL_1275.xlsx\n",
      "pixel_MYO2_CAEEL_1280.xlsx\n",
      "pixel_MYO2_CAEEL_1285.xlsx\n",
      "pixel_MYO2_CAEEL_1290.xlsx\n",
      "pixel_MYO2_CAEEL_1295.xlsx\n",
      "11619  does not have myos.\n",
      "pixel_MYO2_CAEEL_1300.xlsx\n",
      "pixel_MYO2_CAEEL_1305.xlsx\n",
      "pixel_MYO2_CAEEL_1310.xlsx\n",
      "pixel_MYO2_CAEEL_1315.xlsx\n",
      "pixel_MYO2_CAEEL_1320.xlsx\n",
      "pixel_MYO2_CAEEL_1325.xlsx\n",
      "pixel_MYO2_CAEEL_1330.xlsx\n",
      "pixel_MYO2_CAEEL_1335.xlsx\n",
      "pixel_MYO2_CAEEL_1340.xlsx\n",
      "pixel_MYO2_CAEEL_1345.xlsx\n",
      "pixel_MYO2_CAEEL_1350.xlsx\n",
      "2143  does not have myos.\n",
      "pixel_MYO2_CAEEL_1355.xlsx\n",
      "1181  does not have myos.\n",
      "pixel_MYO2_CAEEL_1360.xlsx\n",
      "pixel_MYO2_CAEEL_1365.xlsx\n",
      "144  does not have myos.\n",
      "pixel_MYO2_CAEEL_1370.xlsx\n",
      "13132  does not have myos.\n",
      "pixel_MYO2_CAEEL_1375.xlsx\n",
      "2489  does not have myos.\n",
      "pixel_MYO2_CAEEL_1380.xlsx\n",
      "pixel_MYO2_CAEEL_1385.xlsx\n",
      "16996  does not have myos.\n",
      "pixel_MYO2_CAEEL_1390.xlsx\n",
      "pixel_MYO2_CAEEL_1395.xlsx\n",
      "19537  does not have myos.\n",
      "pixel_MYO2_CAEEL_1400.xlsx\n",
      "16755  does not have myos.\n",
      "pixel_MYO2_CAEEL_1405.xlsx\n",
      "14610  does not have myos.\n",
      "pixel_MYO2_CAEEL_1410.xlsx\n",
      "7410  does not have myos.\n",
      "pixel_MYO2_CAEEL_1415.xlsx\n",
      "10400  does not have myos.\n",
      "pixel_MYO2_CAEEL_1420.xlsx\n",
      "11417  does not have myos.\n",
      "pixel_MYO2_CAEEL_1425.xlsx\n",
      "pixel_MYO2_CAEEL_1430.xlsx\n",
      "13287  does not have myos.\n",
      "pixel_MYO2_CAEEL_1435.xlsx\n",
      "5322  does not have myos.\n",
      "pixel_MYO2_CAEEL_1440.xlsx\n",
      "pixel_MYO2_CAEEL_1445.xlsx\n",
      "3171  does not have myos.\n",
      "pixel_MYO2_CAEEL_1450.xlsx\n",
      "pixel_MYO2_CAEEL_1455.xlsx\n",
      "1725  does not have myos.\n",
      "pixel_MYO2_CAEEL_1460.xlsx\n",
      "pixel_MYO2_CAEEL_1465.xlsx\n",
      "pixel_MYO2_CAEEL_1470.xlsx\n",
      "7837  does not have myos.\n",
      "pixel_MYO2_CAEEL_1475.xlsx\n",
      "13407  does not have myos.\n",
      "pixel_MYO2_CAEEL_1480.xlsx\n",
      "pixel_MYO2_CAEEL_1485.xlsx\n",
      "8593  does not have myos.\n",
      "pixel_MYO2_CAEEL_1490.xlsx\n",
      "pixel_MYO2_CAEEL_1495.xlsx\n",
      "17415  does not have myos.\n",
      "pixel_MYO2_CAEEL_1500.xlsx\n",
      "pixel_MYO2_CAEEL_1505.xlsx\n",
      "pixel_MYO2_CAEEL_1510.xlsx\n",
      "18374  does not have myos.\n",
      "pixel_MYO2_CAEEL_1515.xlsx\n",
      "19061  does not have myos.\n",
      "pixel_MYO2_CAEEL_1520.xlsx\n",
      "17415  does not have myos.\n",
      "pixel_MYO2_CAEEL_1525.xlsx\n",
      "536  does not have myos.\n",
      "pixel_MYO2_CAEEL_1530.xlsx\n",
      "pixel_MYO2_CAEEL_1535.xlsx\n",
      "pixel_MYO2_CAEEL_1540.xlsx\n",
      "pixel_MYO2_CAEEL_1545.xlsx\n",
      "pixel_MYO2_CAEEL_1550.xlsx\n",
      "pixel_MYO2_CAEEL_1555.xlsx\n",
      "pixel_MYO2_CAEEL_1560.xlsx\n",
      "15820  does not have myos.\n",
      "pixel_MYO2_CAEEL_1565.xlsx\n",
      "16033  does not have myos.\n",
      "pixel_MYO2_CAEEL_1570.xlsx\n",
      "3506  does not have myos.\n",
      "pixel_MYO2_CAEEL_1575.xlsx\n",
      "540  does not have myos.\n",
      "pixel_MYO2_CAEEL_1580.xlsx\n",
      "pixel_MYO2_CAEEL_1585.xlsx\n",
      "pixel_MYO2_CAEEL_1590.xlsx\n",
      "4431  does not have myos.\n",
      "pixel_MYO2_CAEEL_1595.xlsx\n",
      "9708  does not have myos.\n",
      "pixel_MYO2_CAEEL_1600.xlsx\n",
      "9639  does not have myos.\n",
      "pixel_MYO2_CAEEL_1605.xlsx\n",
      "pixel_MYO2_CAEEL_1610.xlsx\n",
      "pixel_MYO2_CAEEL_1615.xlsx\n",
      "pixel_MYO2_CAEEL_1620.xlsx\n",
      "pixel_MYO2_CAEEL_1625.xlsx\n",
      "pixel_MYO2_CAEEL_1630.xlsx\n",
      "pixel_MYO2_CAEEL_1635.xlsx\n",
      "pixel_MYO2_CAEEL_1640.xlsx\n",
      "pixel_MYO2_CAEEL_1645.xlsx\n",
      "pixel_MYO2_CAEEL_1650.xlsx\n",
      "pixel_MYO2_CAEEL_1655.xlsx\n",
      "pixel_MYO2_CAEEL_1660.xlsx\n",
      "pixel_MYO2_CAEEL_1665.xlsx\n",
      "pixel_MYO2_CAEEL_1670.xlsx\n",
      "pixel_MYO2_CAEEL_1675.xlsx\n",
      "pixel_MYO2_CAEEL_1680.xlsx\n",
      "pixel_MYO2_CAEEL_1685.xlsx\n",
      "pixel_MYO2_CAEEL_1690.xlsx\n",
      "13227  does not have myos.\n",
      "pixel_MYO2_CAEEL_1695.xlsx\n",
      "pixel_MYO2_CAEEL_1700.xlsx\n",
      "15408  does not have myos.\n",
      "pixel_MYO2_CAEEL_1705.xlsx\n",
      "10639  does not have myos.\n",
      "pixel_MYO2_CAEEL_1710.xlsx\n",
      "4431  does not have myos.\n",
      "pixel_MYO2_CAEEL_1715.xlsx\n",
      "9708  does not have myos.\n",
      "pixel_MYO2_CAEEL_1720.xlsx\n",
      "6612  does not have myos.\n",
      "pixel_MYO2_CAEEL_1725.xlsx\n",
      "pixel_MYO2_CAEEL_1730.xlsx\n",
      "pixel_MYO2_CAEEL_1735.xlsx\n",
      "pixel_MYO2_CAEEL_1740.xlsx\n",
      "pixel_MYO2_CAEEL_1745.xlsx\n",
      "pixel_MYO2_CAEEL_1750.xlsx\n",
      "pixel_MYO2_CAEEL_1755.xlsx\n",
      "pixel_MYO2_CAEEL_1760.xlsx\n",
      "14974  does not have myos.\n",
      "pixel_MYO2_CAEEL_1765.xlsx\n",
      "pixel_MYO2_CAEEL_1770.xlsx\n",
      "pixel_MYO2_CAEEL_1775.xlsx\n",
      "2640  does not have myos.\n",
      "pixel_MYO2_CAEEL_1780.xlsx\n",
      "pixel_MYO2_CAEEL_1785.xlsx\n",
      "pixel_MYO2_CAEEL_1790.xlsx\n",
      "pixel_MYO2_CAEEL_1795.xlsx\n",
      "pixel_MYO2_CAEEL_1800.xlsx\n",
      "pixel_MYO2_CAEEL_1805.xlsx\n",
      "10979  does not have myos.\n",
      "pixel_MYO2_CAEEL_1810.xlsx\n",
      "1063  does not have myos.\n",
      "pixel_MYO2_CAEEL_1815.xlsx\n",
      "4208  does not have myos.\n",
      "pixel_MYO2_CAEEL_1820.xlsx\n",
      "pixel_MYO2_CAEEL_1825.xlsx\n",
      "15278  does not have myos.\n",
      "pixel_MYO2_CAEEL_1830.xlsx\n",
      "669  does not have myos.\n",
      "pixel_MYO2_CAEEL_1835.xlsx\n",
      "19061  does not have myos.\n",
      "pixel_MYO2_CAEEL_1840.xlsx\n",
      "669  does not have myos.\n",
      "pixel_MYO2_CAEEL_1845.xlsx\n",
      "pixel_MYO2_CAEEL_1850.xlsx\n",
      "4767  does not have myos.\n",
      "pixel_MYO2_CAEEL_1855.xlsx\n",
      "pixel_MYO2_CAEEL_1860.xlsx\n",
      "17913  does not have myos.\n",
      "pixel_MYO2_CAEEL_1865.xlsx\n",
      "6042  does not have myos.\n",
      "pixel_MYO2_CAEEL_1870.xlsx\n",
      "10216  does not have myos.\n",
      "pixel_MYO2_CAEEL_1875.xlsx\n",
      "13163  does not have myos.\n",
      "pixel_MYO2_CAEEL_1880.xlsx\n",
      "13176  does not have myos.\n",
      "pixel_MYO2_CAEEL_1885.xlsx\n",
      "841  does not have myos.\n",
      "pixel_MYO2_CAEEL_1890.xlsx\n",
      "pixel_MYO2_CAEEL_1895.xlsx\n",
      "180  does not have myos.\n",
      "pixel_MYO2_CAEEL_1900.xlsx\n",
      "843  does not have myos.\n",
      "pixel_MYO2_CAEEL_1905.xlsx\n",
      "pixel_MYO2_CAEEL_1910.xlsx\n",
      "18374  does not have myos.\n",
      "pixel_MYO2_CAEEL_1915.xlsx\n",
      "6077  does not have myos.\n",
      "pixel_MYO2_CAEEL_1920.xlsx\n",
      "6818  does not have myos.\n",
      "pixel_MYO2_CAEEL_1925.xlsx\n",
      "6818  does not have myos.\n",
      "pixel_MYO2_CAEEL_1930.xlsx\n"
     ]
    }
   ],
   "source": [
    "# Using the for loop to iterate throughin order to get the excel file that contains number of GO Terms , start and without myos column\n",
    "peptide_of_interest= [\"MYO2_CAEEL\"]\n",
    "orows=[]\n",
    "for name in peptide_of_interest:\n",
    "    print(name)\n",
    "\n",
    "    bravo = df_pkl[\"Entry name\"] == name\n",
    "    mylist=[]\n",
    "    for i, row in df_pkl[bravo].iterrows():\n",
    "        orow={}\n",
    "        orow[\"Start\"]=row[\"Start\"]\n",
    "        orow[\"Kclust\"]=row[\"Kclust\"]\n",
    "        mylist.append(row[\"Kclust\"])\n",
    "        \n",
    "        pixo = df_pkl[\"Kclust\"]==row[\"Kclust\"]\n",
    "        testdf=df_pkl[pixo]\n",
    "        testdf2=testdf[testdf[\"Gene Names (primary)\"].str.contains(\"somestuff\")==False]\n",
    "        testdf3=testdf[testdf[\"Gene Names (primary)\"].str.contains(\"myo-3|myo-1|unc-54|myo-6|myo-5\")==False]\n",
    "        if len(testdf3)==len(testdf2):\n",
    "            print(row[\"Kclust\"],\" does not have myos.\")\n",
    "            orow[\"Othermyos\"]=False\n",
    "        df_pkl[pixo].to_excel(\"pixel_%s_%s.xlsx\"%(name, row[\"Start\"]))\n",
    "        print(\"pixel_%s_%s.xlsx\"%(name, row[\"Start\"]))\n",
    "\n",
    "        #print(df_pkl[pixo])\n",
    "        orows.append(orow)\n",
    "    resdf=pd.DataFrame(orows)\n",
    "    resdf.to_excel(\"resdf.xlsx\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7087473f-927b-4e7e-b959-97e331c5788a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Gene Names (primary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7734</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>PIEZ1_CAEEL_10_VVKLLLPAALLAAAI</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>17001</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>PIEZ1_CAEEL_15_LPAALLAAAIIRPSF</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7557</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8636</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425117</th>\n",
       "      <td>170</td>\n",
       "      <td>X5ML89</td>\n",
       "      <td>X5ML89_CAEEL</td>\n",
       "      <td>RING-type domain-containing protein</td>\n",
       "      <td>NDELDREMERPPAHR</td>\n",
       "      <td>X5ML89_CAEEL_170_NDELDREMERPPAHR</td>\n",
       "      <td>NDELDREMERPPAHR</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>11071</td>\n",
       "      <td>grif-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425118</th>\n",
       "      <td>175</td>\n",
       "      <td>X5ML89</td>\n",
       "      <td>X5ML89_CAEEL</td>\n",
       "      <td>RING-type domain-containing protein</td>\n",
       "      <td>REMERPPAHRLPRSS</td>\n",
       "      <td>X5ML89_CAEEL_175_REMERPPAHRLPRSS</td>\n",
       "      <td>REMERPPAHRLPRSS</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>54</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>17294</td>\n",
       "      <td>grif-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425119</th>\n",
       "      <td>180</td>\n",
       "      <td>X5ML89</td>\n",
       "      <td>X5ML89_CAEEL</td>\n",
       "      <td>RING-type domain-containing protein</td>\n",
       "      <td>PPAHRLPRSSTPAPA</td>\n",
       "      <td>X5ML89_CAEEL_180_PPAHRLPRSSTPAPA</td>\n",
       "      <td>PPAHRLPRSSTPAPA</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>50</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>10416</td>\n",
       "      <td>grif-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425120</th>\n",
       "      <td>185</td>\n",
       "      <td>X5ML89</td>\n",
       "      <td>X5ML89_CAEEL</td>\n",
       "      <td>RING-type domain-containing protein</td>\n",
       "      <td>LPRSSTPAPAPPFDN</td>\n",
       "      <td>X5ML89_CAEEL_185_LPRSSTPAPAPPFDN</td>\n",
       "      <td>LPRSSTPAPAPPFDN</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>50</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8758</td>\n",
       "      <td>grif-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425121</th>\n",
       "      <td>190</td>\n",
       "      <td>X5ML89</td>\n",
       "      <td>X5ML89_CAEEL</td>\n",
       "      <td>RING-type domain-containing protein</td>\n",
       "      <td>TPAPAPPFDNSVAYY</td>\n",
       "      <td>X5ML89_CAEEL_190_TPAPAPPFDNSVAYY</td>\n",
       "      <td>TPAPAPPFDNSVAYY</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1194</td>\n",
       "      <td>grif-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1689956 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start       Entry    Entry name  \\\n",
       "0            0  A0A061ACU2   PIEZ1_CAEEL   \n",
       "1            5  A0A061ACU2   PIEZ1_CAEEL   \n",
       "2           10  A0A061ACU2   PIEZ1_CAEEL   \n",
       "3           15  A0A061ACU2   PIEZ1_CAEEL   \n",
       "4           20  A0A061ACU2   PIEZ1_CAEEL   \n",
       "...        ...         ...           ...   \n",
       "2425117    170      X5ML89  X5ML89_CAEEL   \n",
       "2425118    175      X5ML89  X5ML89_CAEEL   \n",
       "2425119    180      X5ML89  X5ML89_CAEEL   \n",
       "2425120    185      X5ML89  X5ML89_CAEEL   \n",
       "2425121    190      X5ML89  X5ML89_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "0        Piezo-type mechanosensitive ion channel compon...  MTVPPLLKSCVVKLL   \n",
       "1        Piezo-type mechanosensitive ion channel compon...  LLKSCVVKLLLPAAL   \n",
       "2        Piezo-type mechanosensitive ion channel compon...  VVKLLLPAALLAAAI   \n",
       "3        Piezo-type mechanosensitive ion channel compon...  LPAALLAAAIIRPSF   \n",
       "4        Piezo-type mechanosensitive ion channel compon...  LAAAIIRPSFLSIGY   \n",
       "...                                                    ...              ...   \n",
       "2425117                RING-type domain-containing protein  NDELDREMERPPAHR   \n",
       "2425118                RING-type domain-containing protein  REMERPPAHRLPRSS   \n",
       "2425119                RING-type domain-containing protein  PPAHRLPRSSTPAPA   \n",
       "2425120                RING-type domain-containing protein  LPRSSTPAPAPPFDN   \n",
       "2425121                RING-type domain-containing protein  TPAPAPPFDNSVAYY   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "0           PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL  MTVPPLLKSCVVKLL       0       4   \n",
       "1           PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL  LLKSCVVKLLLPAAL       0       2   \n",
       "2          PIEZ1_CAEEL_10_VVKLLLPAALLAAAI  VVKLLLPAALLAAAI       0       2   \n",
       "3          PIEZ1_CAEEL_15_LPAALLAAAIIRPSF  LPAALLAAAIIRPSF       0       4   \n",
       "4          PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY  LAAAIIRPSFLSIGY       1       2   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2425117  X5ML89_CAEEL_170_NDELDREMERPPAHR  NDELDREMERPPAHR       0       4   \n",
       "2425118  X5ML89_CAEEL_175_REMERPPAHRLPRSS  REMERPPAHRLPRSS       0       6   \n",
       "2425119  X5ML89_CAEEL_180_PPAHRLPRSSTPAPA  PPAHRLPRSSTPAPA       0      10   \n",
       "2425120  X5ML89_CAEEL_185_LPRSSTPAPAPPFDN  LPRSSTPAPAPPFDN       0      10   \n",
       "2425121  X5ML89_CAEEL_190_TPAPAPPFDNSVAYY  TPAPAPPFDNSVAYY       0       8   \n",
       "\n",
       "         CB  ...  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \\\n",
       "0        13  ...  1    2    0  56  False  False  False     False   14908   \n",
       "1        14  ...  0    2    0  58  False  False  False     False    7734   \n",
       "2        14  ...  0    1    0  58  False  False  False     False   17001   \n",
       "3        13  ...  0    0    4  56  False  False  False     False    7557   \n",
       "4        13  ...  0    0    4  57  False  False  False     False    8636   \n",
       "...      ..  ... ..  ...  ...  ..    ...    ...    ...       ...     ...   \n",
       "2425117  13  ...  1    0   12  56  False  False  False     False   11071   \n",
       "2425118  12  ...  1    0   16  54  False  False  False     False   17294   \n",
       "2425119  10  ...  0    0    8  50  False  False  False     False   10416   \n",
       "2425120  10  ...  0    0    4  50  False  False  False     False    8758   \n",
       "2425121  11  ...  0    0    0  52  False  False  False     False    1194   \n",
       "\n",
       "         Gene Names (primary)  \n",
       "0                      pezo-1  \n",
       "1                      pezo-1  \n",
       "2                      pezo-1  \n",
       "3                      pezo-1  \n",
       "4                      pezo-1  \n",
       "...                       ...  \n",
       "2425117                grif-1  \n",
       "2425118                grif-1  \n",
       "2425119                grif-1  \n",
       "2425120                grif-1  \n",
       "2425121                grif-1  \n",
       "\n",
       "[1689956 rows x 32 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Showing gene Names column that does not contain myo-3\n",
    "df_pkl[df_pkl[\"Gene Names (primary)\"].str.contains(\"myo-2\")==False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1047423e-d96a-4e69-a1ea-83285ddb60f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Gene Names (primary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25807</th>\n",
       "      <td>45</td>\n",
       "      <td>G5ECR9</td>\n",
       "      <td>UNC55_CAEEL</td>\n",
       "      <td>Nuclear hormone receptor unc-55 (Uncoordinated...</td>\n",
       "      <td>KRSIRRSLSYTCRAT</td>\n",
       "      <td>UNC55_CAEEL_45_KRSIRRSLSYTCRAT</td>\n",
       "      <td>KRSIRRSLSYTCRAT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>unc-55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33020</th>\n",
       "      <td>45</td>\n",
       "      <td>G5EE01</td>\n",
       "      <td>PTEN_CAEEL</td>\n",
       "      <td>Phosphatidylinositol 3,4,5-trisphosphate 3-pho...</td>\n",
       "      <td>RHIFRTAVSSNRCRT</td>\n",
       "      <td>PTEN_CAEEL_45_RHIFRTAVSSNRCRT</td>\n",
       "      <td>RHIFRTAVSSNRCRT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>daf-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41823</th>\n",
       "      <td>105</td>\n",
       "      <td>G5EFF5</td>\n",
       "      <td>DAF12_CAEEL</td>\n",
       "      <td>Nuclear hormone receptor family member daf-12 ...</td>\n",
       "      <td>DSSEESRRRQKTCRV</td>\n",
       "      <td>DAF12_CAEEL_105_DSSEESRRRQKTCRV</td>\n",
       "      <td>DSSEESRRRQKTCRV</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>daf-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90123</th>\n",
       "      <td>1925</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>ADSAEQCLVRIRSRT</td>\n",
       "      <td>MYO2_CAEEL_1925_ADSAEQCLVRIRSRT</td>\n",
       "      <td>ADSAEQCLVRIRSRT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90124</th>\n",
       "      <td>1930</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>QCLVRIRSRTRANAE</td>\n",
       "      <td>MYO2_CAEEL_1930_QCLVRIRSRTRANAE</td>\n",
       "      <td>QCLVRIRSRTRANAE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377172</th>\n",
       "      <td>860</td>\n",
       "      <td>U4PC56</td>\n",
       "      <td>U4PC56_CAEEL</td>\n",
       "      <td>Transcription Factor ThreeC subunit (GTF3C hom...</td>\n",
       "      <td>VDIVSSSARVSVRCR</td>\n",
       "      <td>U4PC56_CAEEL_860_VDIVSSSARVSVRCR</td>\n",
       "      <td>VDIVSSSARVSVRCR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>tftc-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2383325</th>\n",
       "      <td>270</td>\n",
       "      <td>U4PF33</td>\n",
       "      <td>U4PF33_CAEEL</td>\n",
       "      <td>BUB1 N-terminal domain-containing protein</td>\n",
       "      <td>GSIRSSFRSRRNTDC</td>\n",
       "      <td>U4PF33_CAEEL_270_GSIRSSFRSRRNTDC</td>\n",
       "      <td>GSIRSSFRSRRNTDC</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>59</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2387181</th>\n",
       "      <td>870</td>\n",
       "      <td>U4PMB9</td>\n",
       "      <td>U4PMB9_CAEEL</td>\n",
       "      <td>GPS domain-containing protein</td>\n",
       "      <td>SGTRTNRRCSSDRLL</td>\n",
       "      <td>U4PMB9_CAEEL_870_SGTRTNRRCSSDRLL</td>\n",
       "      <td>SGTRTNRRCSSDRLL</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>59</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>fmil-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2393396</th>\n",
       "      <td>815</td>\n",
       "      <td>U4PRL9</td>\n",
       "      <td>U4PRL9_CAEEL</td>\n",
       "      <td>GPS domain-containing protein</td>\n",
       "      <td>TRTNRRCSSDRLLAL</td>\n",
       "      <td>U4PRL9_CAEEL_815_TRTNRRCSSDRLLAL</td>\n",
       "      <td>TRTNRRCSSDRLLAL</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>fmil-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2404284</th>\n",
       "      <td>35</td>\n",
       "      <td>V6CLM0</td>\n",
       "      <td>V6CLM0_CAEEL</td>\n",
       "      <td>NTR domain-containing protein</td>\n",
       "      <td>CQSLGREIRESASRR</td>\n",
       "      <td>V6CLM0_CAEEL_35_CQSLGREIRESASRR</td>\n",
       "      <td>CQSLGREIRESASRR</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>59</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>sfrp-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start   Entry    Entry name  \\\n",
       "25807       45  G5ECR9   UNC55_CAEEL   \n",
       "33020       45  G5EE01    PTEN_CAEEL   \n",
       "41823      105  G5EFF5   DAF12_CAEEL   \n",
       "90123     1925  P12845    MYO2_CAEEL   \n",
       "90124     1930  P12845    MYO2_CAEEL   \n",
       "...        ...     ...           ...   \n",
       "2377172    860  U4PC56  U4PC56_CAEEL   \n",
       "2383325    270  U4PF33  U4PF33_CAEEL   \n",
       "2387181    870  U4PMB9  U4PMB9_CAEEL   \n",
       "2393396    815  U4PRL9  U4PRL9_CAEEL   \n",
       "2404284     35  V6CLM0  V6CLM0_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "25807    Nuclear hormone receptor unc-55 (Uncoordinated...  KRSIRRSLSYTCRAT   \n",
       "33020    Phosphatidylinositol 3,4,5-trisphosphate 3-pho...  RHIFRTAVSSNRCRT   \n",
       "41823    Nuclear hormone receptor family member daf-12 ...  DSSEESRRRQKTCRV   \n",
       "90123              Myosin-2 (Myosin heavy chain C) (MHC C)  ADSAEQCLVRIRSRT   \n",
       "90124              Myosin-2 (Myosin heavy chain C) (MHC C)  QCLVRIRSRTRANAE   \n",
       "...                                                    ...              ...   \n",
       "2377172  Transcription Factor ThreeC subunit (GTF3C hom...  VDIVSSSARVSVRCR   \n",
       "2383325          BUB1 N-terminal domain-containing protein  GSIRSSFRSRRNTDC   \n",
       "2387181                      GPS domain-containing protein  SGTRTNRRCSSDRLL   \n",
       "2393396                      GPS domain-containing protein  TRTNRRCSSDRLLAL   \n",
       "2404284                      NTR domain-containing protein  CQSLGREIRESASRR   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "25807      UNC55_CAEEL_45_KRSIRRSLSYTCRAT  KRSIRRSLSYTCRAT       0       0   \n",
       "33020       PTEN_CAEEL_45_RHIFRTAVSSNRCRT  RHIFRTAVSSNRCRT       0       0   \n",
       "41823     DAF12_CAEEL_105_DSSEESRRRQKTCRV  DSSEESRRRQKTCRV       0       0   \n",
       "90123     MYO2_CAEEL_1925_ADSAEQCLVRIRSRT  ADSAEQCLVRIRSRT       0       0   \n",
       "90124     MYO2_CAEEL_1930_QCLVRIRSRTRANAE  QCLVRIRSRTRANAE       0       0   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2377172  U4PC56_CAEEL_860_VDIVSSSARVSVRCR  VDIVSSSARVSVRCR       0       0   \n",
       "2383325  U4PF33_CAEEL_270_GSIRSSFRSRRNTDC  GSIRSSFRSRRNTDC       1       0   \n",
       "2387181  U4PMB9_CAEEL_870_SGTRTNRRCSSDRLL  SGTRTNRRCSSDRLL       1       0   \n",
       "2393396  U4PRL9_CAEEL_815_TRTNRRCSSDRLLAL  TRTNRRCSSDRLLAL       0       0   \n",
       "2404284   V6CLM0_CAEEL_35_CQSLGREIRESASRR  CQSLGREIRESASRR       1       0   \n",
       "\n",
       "         CB  ...  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \\\n",
       "25807    15  ...  0    1   16  60  False  False  False     False    6818   \n",
       "33020    15  ...  0    0   16  60  False  False  False     False    6818   \n",
       "41823    15  ...  0    1   16  60  False  False  False     False    6818   \n",
       "90123    15  ...  0    0   12  60  False  False  False     False    6818   \n",
       "90124    15  ...  0    0   16  60  False  False  False     False    6818   \n",
       "...      ..  ... ..  ...  ...  ..    ...    ...    ...       ...     ...   \n",
       "2377172  15  ...  0    0   12  60  False  False  False     False    6818   \n",
       "2383325  14  ...  0    0   16  59   True  False  False     False    6818   \n",
       "2387181  14  ...  0    0   16  59  False  False  False     False    6818   \n",
       "2393396  15  ...  0    0   16  60  False  False  False     False    6818   \n",
       "2404284  14  ...  0    0   16  59  False  False  False     False    6818   \n",
       "\n",
       "         Gene Names (primary)  \n",
       "25807                  unc-55  \n",
       "33020                  daf-18  \n",
       "41823                  daf-12  \n",
       "90123                   myo-2  \n",
       "90124                   myo-2  \n",
       "...                       ...  \n",
       "2377172                tftc-1  \n",
       "2383325                   NaN  \n",
       "2387181                fmil-1  \n",
       "2393396                fmil-1  \n",
       "2404284                sfrp-1  \n",
       "\n",
       "[143 rows x 32 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "testdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5617a3aa-d908-403c-81d1-fcad81190134",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Gene Names (primary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7734</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>PIEZ1_CAEEL_10_VVKLLLPAALLAAAI</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>17001</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>PIEZ1_CAEEL_15_LPAALLAAAIIRPSF</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7557</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8636</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425155</th>\n",
       "      <td>95</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>EHPRLHSYSVTPSAP</td>\n",
       "      <td>X5ML99_CAEEL_95_EHPRLHSYSVTPSAP</td>\n",
       "      <td>EHPRLHSYSVTPSAP</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>54</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8981</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425156</th>\n",
       "      <td>100</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>HSYSVTPSAPAMETA</td>\n",
       "      <td>X5ML99_CAEEL_100_HSYSVTPSAPAMETA</td>\n",
       "      <td>HSYSVTPSAPAMETA</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16050</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425157</th>\n",
       "      <td>105</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>TPSAPAMETAFGHDY</td>\n",
       "      <td>X5ML99_CAEEL_105_TPSAPAMETAFGHDY</td>\n",
       "      <td>TPSAPAMETAFGHDY</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16050</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425158</th>\n",
       "      <td>110</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>AMETAFGHDYDPPPS</td>\n",
       "      <td>X5ML99_CAEEL_110_AMETAFGHDYDPPPS</td>\n",
       "      <td>AMETAFGHDYDPPPS</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7246</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425159</th>\n",
       "      <td>115</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>FGHDYDPPPSYTEAM</td>\n",
       "      <td>X5ML99_CAEEL_115_FGHDYDPPPSYTEAM</td>\n",
       "      <td>FGHDYDPPPSYTEAM</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>18760</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2425160 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start       Entry    Entry name  \\\n",
       "0            0  A0A061ACU2   PIEZ1_CAEEL   \n",
       "1            5  A0A061ACU2   PIEZ1_CAEEL   \n",
       "2           10  A0A061ACU2   PIEZ1_CAEEL   \n",
       "3           15  A0A061ACU2   PIEZ1_CAEEL   \n",
       "4           20  A0A061ACU2   PIEZ1_CAEEL   \n",
       "...        ...         ...           ...   \n",
       "2425155     95      X5ML99  X5ML99_CAEEL   \n",
       "2425156    100      X5ML99  X5ML99_CAEEL   \n",
       "2425157    105      X5ML99  X5ML99_CAEEL   \n",
       "2425158    110      X5ML99  X5ML99_CAEEL   \n",
       "2425159    115      X5ML99  X5ML99_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "0        Piezo-type mechanosensitive ion channel compon...  MTVPPLLKSCVVKLL   \n",
       "1        Piezo-type mechanosensitive ion channel compon...  LLKSCVVKLLLPAAL   \n",
       "2        Piezo-type mechanosensitive ion channel compon...  VVKLLLPAALLAAAI   \n",
       "3        Piezo-type mechanosensitive ion channel compon...  LPAALLAAAIIRPSF   \n",
       "4        Piezo-type mechanosensitive ion channel compon...  LAAAIIRPSFLSIGY   \n",
       "...                                                    ...              ...   \n",
       "2425155                 EGF-like domain-containing protein  EHPRLHSYSVTPSAP   \n",
       "2425156                 EGF-like domain-containing protein  HSYSVTPSAPAMETA   \n",
       "2425157                 EGF-like domain-containing protein  TPSAPAMETAFGHDY   \n",
       "2425158                 EGF-like domain-containing protein  AMETAFGHDYDPPPS   \n",
       "2425159                 EGF-like domain-containing protein  FGHDYDPPPSYTEAM   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "0           PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL  MTVPPLLKSCVVKLL       0       4   \n",
       "1           PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL  LLKSCVVKLLLPAAL       0       2   \n",
       "2          PIEZ1_CAEEL_10_VVKLLLPAALLAAAI  VVKLLLPAALLAAAI       0       2   \n",
       "3          PIEZ1_CAEEL_15_LPAALLAAAIIRPSF  LPAALLAAAIIRPSF       0       4   \n",
       "4          PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY  LAAAIIRPSFLSIGY       1       2   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2425155   X5ML99_CAEEL_95_EHPRLHSYSVTPSAP  EHPRLHSYSVTPSAP       0       6   \n",
       "2425156  X5ML99_CAEEL_100_HSYSVTPSAPAMETA  HSYSVTPSAPAMETA       0       4   \n",
       "2425157  X5ML99_CAEEL_105_TPSAPAMETAFGHDY  TPSAPAMETAFGHDY       1       4   \n",
       "2425158  X5ML99_CAEEL_110_AMETAFGHDYDPPPS  AMETAFGHDYDPPPS       1       6   \n",
       "2425159  X5ML99_CAEEL_115_FGHDYDPPPSYTEAM  FGHDYDPPPSYTEAM       1       6   \n",
       "\n",
       "         CB  ...  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \\\n",
       "0        13  ...  1    2    0  56  False  False  False     False   14908   \n",
       "1        14  ...  0    2    0  58  False  False  False     False    7734   \n",
       "2        14  ...  0    1    0  58  False  False  False     False   17001   \n",
       "3        13  ...  0    0    4  56  False  False  False     False    7557   \n",
       "4        13  ...  0    0    4  57  False  False  False     False    8636   \n",
       "...      ..  ... ..  ...  ...  ..    ...    ...    ...       ...     ...   \n",
       "2425155  12  ...  0    0    4  54  False  False  False     False    8981   \n",
       "2425156  13  ...  1    0    0  56  False  False  False     False   16050   \n",
       "2425157  12  ...  1    0    0  55  False  False  False     False   16050   \n",
       "2425158  11  ...  1    0    0  53  False  False  False     False    7246   \n",
       "2425159  11  ...  1    0    0  53  False  False  False     False   18760   \n",
       "\n",
       "         Gene Names (primary)  \n",
       "0                      pezo-1  \n",
       "1                      pezo-1  \n",
       "2                      pezo-1  \n",
       "3                      pezo-1  \n",
       "4                      pezo-1  \n",
       "...                       ...  \n",
       "2425155                   NaN  \n",
       "2425156                   NaN  \n",
       "2425157                   NaN  \n",
       "2425158                   NaN  \n",
       "2425159                   NaN  \n",
       "\n",
       "[2425160 rows x 32 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Dataframe containg the column of interest- Gene Names (primary)\n",
    "df_pkl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "64a03fee-7011-4fe2-bef0-553bf799e15e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Gene Names (primary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89738</th>\n",
       "      <td>0</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>MDYENDPGWKYLRRS</td>\n",
       "      <td>MYO2_CAEEL_0_MDYENDPGWKYLRRS</td>\n",
       "      <td>MDYENDPGWKYLRRS</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>57</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8493</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89739</th>\n",
       "      <td>5</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>DPGWKYLRRSREEML</td>\n",
       "      <td>MYO2_CAEEL_5_DPGWKYLRRSREEML</td>\n",
       "      <td>DPGWKYLRRSREEML</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>57</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>744</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89740</th>\n",
       "      <td>10</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>YLRRSREEMLQDQSR</td>\n",
       "      <td>MYO2_CAEEL_10_YLRRSREEMLQDQSR</td>\n",
       "      <td>YLRRSREEMLQDQSR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>5650</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89741</th>\n",
       "      <td>15</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>REEMLQDQSRAYDSK</td>\n",
       "      <td>MYO2_CAEEL_15_REEMLQDQSRAYDSK</td>\n",
       "      <td>REEMLQDQSRAYDSK</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>11549</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89742</th>\n",
       "      <td>20</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>QDQSRAYDSKKNVWI</td>\n",
       "      <td>MYO2_CAEEL_20_QDQSRAYDSKKNVWI</td>\n",
       "      <td>QDQSRAYDSKKNVWI</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>11268</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90120</th>\n",
       "      <td>1910</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>KYRTVQLSLETAEER</td>\n",
       "      <td>MYO2_CAEEL_1910_KYRTVQLSLETAEER</td>\n",
       "      <td>KYRTVQLSLETAEER</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1339</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90121</th>\n",
       "      <td>1915</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>QLSLETAEERADSAE</td>\n",
       "      <td>MYO2_CAEEL_1915_QLSLETAEERADSAE</td>\n",
       "      <td>QLSLETAEERADSAE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>18374</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90122</th>\n",
       "      <td>1920</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>TAEERADSAEQCLVR</td>\n",
       "      <td>MYO2_CAEEL_1920_TAEERADSAEQCLVR</td>\n",
       "      <td>TAEERADSAEQCLVR</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6077</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90123</th>\n",
       "      <td>1925</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>ADSAEQCLVRIRSRT</td>\n",
       "      <td>MYO2_CAEEL_1925_ADSAEQCLVRIRSRT</td>\n",
       "      <td>ADSAEQCLVRIRSRT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90124</th>\n",
       "      <td>1930</td>\n",
       "      <td>P12845</td>\n",
       "      <td>MYO2_CAEEL</td>\n",
       "      <td>Myosin-2 (Myosin heavy chain C) (MHC C)</td>\n",
       "      <td>QCLVRIRSRTRANAE</td>\n",
       "      <td>MYO2_CAEEL_1930_QCLVRIRSRTRANAE</td>\n",
       "      <td>QCLVRIRSRTRANAE</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>60</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>6818</td>\n",
       "      <td>myo-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Start   Entry  Entry name                            Protein names  \\\n",
       "89738      0  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "89739      5  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "89740     10  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "89741     15  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "89742     20  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "...      ...     ...         ...                                      ...   \n",
       "90120   1910  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "90121   1915  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "90122   1920  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "90123   1925  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "90124   1930  P12845  MYO2_CAEEL  Myosin-2 (Myosin heavy chain C) (MHC C)   \n",
       "\n",
       "               Peptide                     Prot_res_pep               aa  \\\n",
       "89738  MDYENDPGWKYLRRS     MYO2_CAEEL_0_MDYENDPGWKYLRRS  MDYENDPGWKYLRRS   \n",
       "89739  DPGWKYLRRSREEML     MYO2_CAEEL_5_DPGWKYLRRSREEML  DPGWKYLRRSREEML   \n",
       "89740  YLRRSREEMLQDQSR    MYO2_CAEEL_10_YLRRSREEMLQDQSR  YLRRSREEMLQDQSR   \n",
       "89741  REEMLQDQSRAYDSK    MYO2_CAEEL_15_REEMLQDQSRAYDSK  REEMLQDQSRAYDSK   \n",
       "89742  QDQSRAYDSKKNVWI    MYO2_CAEEL_20_QDQSRAYDSKKNVWI  QDQSRAYDSKKNVWI   \n",
       "...                ...                              ...              ...   \n",
       "90120  KYRTVQLSLETAEER  MYO2_CAEEL_1910_KYRTVQLSLETAEER  KYRTVQLSLETAEER   \n",
       "90121  QLSLETAEERADSAE  MYO2_CAEEL_1915_QLSLETAEERADSAE  QLSLETAEERADSAE   \n",
       "90122  TAEERADSAEQCLVR  MYO2_CAEEL_1920_TAEERADSAEQCLVR  TAEERADSAEQCLVR   \n",
       "90123  ADSAEQCLVRIRSRT  MYO2_CAEEL_1925_ADSAEQCLVRIRSRT  ADSAEQCLVRIRSRT   \n",
       "90124  QCLVRIRSRTRANAE  MYO2_CAEEL_1930_QCLVRIRSRTRANAE  QCLVRIRSRTRANAE   \n",
       "\n",
       "       CA-Gly  Pro-MC  CB  ...  S  NH3  Arg  MC  LastC  ManyH  SeCys  \\\n",
       "89738       1       2  13  ...  1    1    8  57  False  False  False   \n",
       "89739       1       2  13  ...  1    1   12  57  False  False  False   \n",
       "89740       0       0  15  ...  1    0   16  60  False  False  False   \n",
       "89741       0       0  15  ...  1    1    8  60  False  False  False   \n",
       "89742       0       0  15  ...  0    2    4  60  False  False  False   \n",
       "...       ...     ...  ..  ... ..  ...  ...  ..    ...    ...    ...   \n",
       "90120       0       0  15  ...  0    1    8  60  False  False  False   \n",
       "90121       0       0  15  ...  0    0    4  60  False  False  False   \n",
       "90122       0       0  15  ...  0    0    8  60  False  False  False   \n",
       "90123       0       0  15  ...  0    0   12  60  False  False  False   \n",
       "90124       0       0  15  ...  0    0   16  60  False  False  False   \n",
       "\n",
       "       PyrroLys  Kclust  Gene Names (primary)  \n",
       "89738     False    8493                 myo-2  \n",
       "89739     False     744                 myo-2  \n",
       "89740     False    5650                 myo-2  \n",
       "89741     False   11549                 myo-2  \n",
       "89742     False   11268                 myo-2  \n",
       "...         ...     ...                   ...  \n",
       "90120     False    1339                 myo-2  \n",
       "90121     False   18374                 myo-2  \n",
       "90122     False    6077                 myo-2  \n",
       "90123     False    6818                 myo-2  \n",
       "90124     False    6818                 myo-2  \n",
       "\n",
       "[387 rows x 32 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Identifying the entry name in the dataframe using indexing\n",
    "ben=df_pkl[\"Entry name\"]== \"MYO2_CAEEL\"\n",
    "\n",
    "df_pkl[ben]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "075ceedd-31b0-40b3-9095-58a075c1e433",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Start                                                      1930\n",
       "Entry                                                    P12845\n",
       "Entry name                                           MYO2_CAEEL\n",
       "Protein names           Myosin-2 (Myosin heavy chain C) (MHC C)\n",
       "Peptide                                         QCLVRIRSRTRANAE\n",
       "Prot_res_pep                    MYO2_CAEEL_1930_QCLVRIRSRTRANAE\n",
       "aa                                              QCLVRIRSRTRANAE\n",
       "CA-Gly                                                        0\n",
       "Pro-MC                                                        0\n",
       "CB                                                           15\n",
       "CB-Pro                                                        0\n",
       "Carboxyl                                                      3\n",
       "Amide                                                         6\n",
       "His                                                           0\n",
       "Trp                                                           0\n",
       "Phe-Tyr                                                       0\n",
       "OH-Tyr                                                        0\n",
       "CG                                                           12\n",
       "CD                                                            7\n",
       "CE                                                            0\n",
       "OH                                                            2\n",
       "SH                                                            1\n",
       "S                                                             0\n",
       "NH3                                                           0\n",
       "Arg                                                          16\n",
       "MC                                                           60\n",
       "LastC                                                     False\n",
       "ManyH                                                     False\n",
       "SeCys                                                     False\n",
       "PyrroLys                                                  False\n",
       "Kclust                                                     6818\n",
       "Gene Names (primary)                                      myo-2\n",
       "Name: 90124, dtype: object"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Showing key and values associated with each column in the dataframe\n",
    "row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9f0f3e02-75ac-40b8-bbc5-01777abdda63",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n# Using TOPPGENE FOR ENRICHMENT ANALYSIS\\n\\nimport requests\\nimport pandas as pd\\nimport mygene\\n\\n\\ndef functional_enrichment_analysis(gene_symbols):\\n    \\'\\'\\'\\n    Perform functional enrichment analysis using the ToppGene API.\\n\\n    Args:\\n        gene_symbols (list): A list of human gene symbols.\\n\\n    Returns:\\n        pd.DataFrame: A DataFrame containing the functional enrichment results.\\n    \\'\\'\\'\\n    # Convert gene symbols to Entrez IDs using a mapping API or a predefined mapping file\\n\\n    mg = mygene.MyGeneInfo()\\n    result = mg.querymany(gene_symbols, scopes=\"uniprot\", fields=\"entrezgene\", species=6239)\\n    strlist = [d[\\'entrezgene\\'] for d in result if \\'entrezgene\\' in d]\\n    gene_list = [int(item) for item in strlist]\\n    print(gene_list)\\n\\n    # Define the API endpoint\\n    api_url = \"https://toppgene.cchmc.org/API/enrich\"\\n\\n    # Construct the payload\\n    payload = {\\n        \"Genes\": gene_list,\\n        \"Species\": 6239\\n    }\\n\\n    # Make the API request\\n    try:\\n        response = requests.post(api_url, json=payload)\\n        response.raise_for_status()\\n    except requests.exceptions.RequestException as e:\\n        raise RuntimeError(f\"API request failed: {e}\")\\n\\n    # Parse the response\\n    results = response.json()\\n    print(results)\\n    if \"Annotations\" not in results:\\n        raise ValueError(\"Unexpected response format or no annotations found.\")\\n\\n    # Convert annotations to a DataFrame\\n    data = []\\n    for annotation in results[\"Annotations\"]:\\n        genes_in_term_in_query = annotation.get(\"Genes\", [])\\n        genes_in_term_in_query = [d[\\'Symbol\\'] for d in genes_in_term_in_query if \\'Symbol\\' in d]\\n\\n        data.append({\\n            \"Category\": annotation.get(\"Category\"),\\n            \"ID\": annotation.get(\"ID\"),\\n            \"Name\": annotation.get(\"Name\"),\\n            \"PValue\": annotation.get(\"PValue\"),\\n            \"QValueFDRBH\": annotation.get(\"QValueFDRBH\"),\\n            \"QValueFDRBY\": annotation.get(\"QValueFDRBY\"),\\n            \"QValueBonferroni\": annotation.get(\"QValueBonferroni\"),\\n            \"TotalGenes\": annotation.get(\"TotalGenes\"),\\n            \"GenesInTerm\": annotation.get(\"GenesInTerm\"),\\n            \"GenesInQuery\": annotation.get(\"GenesInQuery\"),\\n            \"GenesInTermInQuery\": annotation.get(\"GenesInTermInQuery\"),\\n            \"IntersectionWithQuery\": genes_in_term_in_query\\n        })\\n\\n    return pd.DataFrame(data)\\n\\n    Example usage:\\n    gene_symbols = [\"SRF\", \"GATA4\", \"HAND2\", \"NKX2-5\", \"HAND1\", \"MEF2A\", \"TBX5\"]\\n    gene_symbols = [\"G5EBL3\", \"C8JQR2\", \"C4ALD5\"]\\n    gene_symbols = list(df_pkl[df_pkl[\\'Kclust\\']==1001][\\'Entry\\'])\\n    result_df = functional_enrichment_analysis(gene_symbols)\\n    result_df\\n'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "# Using TOPPGENE FOR ENRICHMENT ANALYSIS\n",
    "\n",
    "import requests\n",
    "import pandas as pd\n",
    "import mygene\n",
    "\n",
    "\n",
    "def functional_enrichment_analysis(gene_symbols):\n",
    "    '''\n",
    "    Perform functional enrichment analysis using the ToppGene API.\n",
    "\n",
    "    Args:\n",
    "        gene_symbols (list): A list of human gene symbols.\n",
    "\n",
    "    Returns:\n",
    "        pd.DataFrame: A DataFrame containing the functional enrichment results.\n",
    "    '''\n",
    "    # Convert gene symbols to Entrez IDs using a mapping API or a predefined mapping file\n",
    "\n",
    "    mg = mygene.MyGeneInfo()\n",
    "    result = mg.querymany(gene_symbols, scopes=\"uniprot\", fields=\"entrezgene\", species=6239)\n",
    "    strlist = [d['entrezgene'] for d in result if 'entrezgene' in d]\n",
    "    gene_list = [int(item) for item in strlist]\n",
    "    print(gene_list)\n",
    "\n",
    "    # Define the API endpoint\n",
    "    api_url = \"https://toppgene.cchmc.org/API/enrich\"\n",
    "\n",
    "    # Construct the payload\n",
    "    payload = {\n",
    "        \"Genes\": gene_list,\n",
    "        \"Species\": 6239\n",
    "    }\n",
    "\n",
    "    # Make the API request\n",
    "    try:\n",
    "        response = requests.post(api_url, json=payload)\n",
    "        response.raise_for_status()\n",
    "    except requests.exceptions.RequestException as e:\n",
    "        raise RuntimeError(f\"API request failed: {e}\")\n",
    "\n",
    "    # Parse the response\n",
    "    results = response.json()\n",
    "    print(results)\n",
    "    if \"Annotations\" not in results:\n",
    "        raise ValueError(\"Unexpected response format or no annotations found.\")\n",
    "\n",
    "    # Convert annotations to a DataFrame\n",
    "    data = []\n",
    "    for annotation in results[\"Annotations\"]:\n",
    "        genes_in_term_in_query = annotation.get(\"Genes\", [])\n",
    "        genes_in_term_in_query = [d['Symbol'] for d in genes_in_term_in_query if 'Symbol' in d]\n",
    "\n",
    "        data.append({\n",
    "            \"Category\": annotation.get(\"Category\"),\n",
    "            \"ID\": annotation.get(\"ID\"),\n",
    "            \"Name\": annotation.get(\"Name\"),\n",
    "            \"PValue\": annotation.get(\"PValue\"),\n",
    "            \"QValueFDRBH\": annotation.get(\"QValueFDRBH\"),\n",
    "            \"QValueFDRBY\": annotation.get(\"QValueFDRBY\"),\n",
    "            \"QValueBonferroni\": annotation.get(\"QValueBonferroni\"),\n",
    "            \"TotalGenes\": annotation.get(\"TotalGenes\"),\n",
    "            \"GenesInTerm\": annotation.get(\"GenesInTerm\"),\n",
    "            \"GenesInQuery\": annotation.get(\"GenesInQuery\"),\n",
    "            \"GenesInTermInQuery\": annotation.get(\"GenesInTermInQuery\"),\n",
    "            \"IntersectionWithQuery\": genes_in_term_in_query\n",
    "        })\n",
    "\n",
    "    return pd.DataFrame(data)\n",
    "\n",
    "    Example usage:\n",
    "    gene_symbols = [\"SRF\", \"GATA4\", \"HAND2\", \"NKX2-5\", \"HAND1\", \"MEF2A\", \"TBX5\"]\n",
    "    gene_symbols = [\"G5EBL3\", \"C8JQR2\", \"C4ALD5\"]\n",
    "    gene_symbols = list(df_pkl[df_pkl['Kclust']==1001]['Entry'])\n",
    "    result_df = functional_enrichment_analysis(gene_symbols)\n",
    "    result_df\n",
    "\"\"\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d6708722-3b42-4773-a544-4fe998676d98",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Echeh\\AppData\\Local\\Temp\\ipykernel_15152\\2057579436.py:17: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  pd.concat(GOs).to_excel('cluster_MYO2_GOs.xlsx')\n"
     ]
    }
   ],
   "source": [
    "# USING Gprofiler for enrichment analysis, to get excel files and does not need to be re-run\n",
    "import numpy as np\n",
    "GOs=[]\n",
    "for i in mylist:\n",
    "    subdf=df_pkl[df_pkl.Kclust==i]\n",
    "    gp = GProfiler(return_dataframe=True)\n",
    "    myquery=list(subdf.dropna()['Gene Names (primary)'])\n",
    "    results = gp.profile(organism='celegans', query=myquery,no_evidences=False)\n",
    "    results['Kclust']=i\n",
    "    results['qgenelist']=str(list(subdf['Gene Names (primary)']))\n",
    "    results=results[results[\"source\"]!=\"TF\"]\n",
    "    results=results[results[\"source\"]!=\"MIRNA\"]\n",
    "    results=results[results[\"source\"]!=\"HP\"]\n",
    "    results.to_excel('cluster_%i_GOs.xlsx' % i )\n",
    "    GOs.append(results)\n",
    "    #subdf.to_excel('cluster_MYO3_%i.xlsx' %i)\n",
    "pd.concat(GOs).to_excel('cluster_MYO2_GOs.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "dcd0333e-4292-404c-a55b-9308b94003d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Echeh\\AppData\\Local\\Temp\\ipykernel_15152\\379820850.py:2: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  pd.concat(GOs)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>qgenelist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0033186</td>\n",
       "      <td>CAF-1 complex</td>\n",
       "      <td>0.005045</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A conserved heterotrimeric protein complex th...</td>\n",
       "      <td>2</td>\n",
       "      <td>71</td>\n",
       "      <td>2</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0032991]</td>\n",
       "      <td>[chaf-2, chaf-1]</td>\n",
       "      <td>[[IBA], [IBA]]</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005737</td>\n",
       "      <td>cytoplasm</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The contents of a cell excluding the plasma m...</td>\n",
       "      <td>4587</td>\n",
       "      <td>71</td>\n",
       "      <td>42</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.591549</td>\n",
       "      <td>0.009156</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005622, GO:0110165]</td>\n",
       "      <td>[uba-1, pak-2, daf-9, cpna-1, hda-4, myo-2, lr...</td>\n",
       "      <td>[[IBA], [IBA], [IDA, IBA], [IDA, IEA], [ISS], ...</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:BP</td>\n",
       "      <td>GO:0007167</td>\n",
       "      <td>enzyme-linked receptor protein signaling pathway</td>\n",
       "      <td>0.021501</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The series of molecular signals initiated by ...</td>\n",
       "      <td>144</td>\n",
       "      <td>67</td>\n",
       "      <td>7</td>\n",
       "      <td>10936</td>\n",
       "      <td>0.104478</td>\n",
       "      <td>0.048611</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0007166]</td>\n",
       "      <td>[hda-4, src-2, unc-2, let-23, daf-4, rgl-1, ro...</td>\n",
       "      <td>[[IMP], [IBA], [IGI], [IDA, ISS, IEA], [IMP, I...</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005859</td>\n",
       "      <td>muscle myosin complex</td>\n",
       "      <td>0.030056</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A filament of myosin found in a muscle cell o...</td>\n",
       "      <td>4</td>\n",
       "      <td>71</td>\n",
       "      <td>2</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0016460, GO:0043292]</td>\n",
       "      <td>[myo-2, mlc-3]</td>\n",
       "      <td>[[ISS], [IBA]]</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0003824</td>\n",
       "      <td>catalytic activity</td>\n",
       "      <td>0.018819</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Catalysis of a biochemical reaction at physio...</td>\n",
       "      <td>4722</td>\n",
       "      <td>101</td>\n",
       "      <td>60</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.594059</td>\n",
       "      <td>0.012706</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0003674]</td>\n",
       "      <td>[ksr-2, gcy-23, plc-1, cam-1, src-2, dkf-2, un...</td>\n",
       "      <td>[[IGI, IBA, IEA], [ISS, IBA, IEA], [IDA, IBA, ...</td>\n",
       "      <td>744</td>\n",
       "      <td>['ksr-2', 'gcy-23', 'plc-1', 'cam-1', 'src-2',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0099513</td>\n",
       "      <td>polymeric cytoskeletal fiber</td>\n",
       "      <td>0.025148</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A component of the cytoskeleton consisting of...</td>\n",
       "      <td>167</td>\n",
       "      <td>41</td>\n",
       "      <td>5</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.121951</td>\n",
       "      <td>0.029940</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005856, GO:0099512]</td>\n",
       "      <td>[sma-1, mua-3, ify-1, alp-1, dlc-3]</td>\n",
       "      <td>[[IEA], [IDA], [IDA], [IBA], [IEA]]</td>\n",
       "      <td>6077</td>\n",
       "      <td>['sma-1', 'sma-1', 'gbf-1', 'unc-94', 'glh-3',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP1546</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>0.003061</td>\n",
       "      <td>True</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>106</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>897</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.047170</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "      <td>[daf-18, daf-12, let-363, lin-17, sfrp-1]</td>\n",
       "      <td>[[WP], [WP], [WP], [WP], [WP]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REAC</td>\n",
       "      <td>REAC:R-CEL-390666</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>0.035684</td>\n",
       "      <td>True</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>4004</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[REAC:R-CEL-375280]</td>\n",
       "      <td>[dop-1, dop-6]</td>\n",
       "      <td>[[REAC], [REAC]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP1546</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>0.003061</td>\n",
       "      <td>True</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>106</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>897</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.047170</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "      <td>[daf-18, daf-12, let-363, lin-17, sfrp-1]</td>\n",
       "      <td>[[WP], [WP], [WP], [WP], [WP]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REAC</td>\n",
       "      <td>REAC:R-CEL-390666</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>0.035684</td>\n",
       "      <td>True</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>4004</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[REAC:R-CEL-375280]</td>\n",
       "      <td>[dop-1, dop-6]</td>\n",
       "      <td>[[REAC], [REAC]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15271 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   source             native  \\\n",
       "0   GO:CC         GO:0033186   \n",
       "1   GO:CC         GO:0005737   \n",
       "2   GO:BP         GO:0007167   \n",
       "3   GO:CC         GO:0005859   \n",
       "0   GO:MF         GO:0003824   \n",
       "..    ...                ...   \n",
       "23  GO:CC         GO:0099513   \n",
       "0      WP          WP:WP1546   \n",
       "1    REAC  REAC:R-CEL-390666   \n",
       "0      WP          WP:WP1546   \n",
       "1    REAC  REAC:R-CEL-390666   \n",
       "\n",
       "                                                name   p_value significant  \\\n",
       "0                                      CAF-1 complex  0.005045        True   \n",
       "1                                          cytoplasm  0.006956        True   \n",
       "2   enzyme-linked receptor protein signaling pathway  0.021501        True   \n",
       "3                              muscle myosin complex  0.030056        True   \n",
       "0                                 catalytic activity  0.018819        True   \n",
       "..                                               ...       ...         ...   \n",
       "23                      polymeric cytoskeletal fiber  0.025148        True   \n",
       "0                    Metapathway signal transduction  0.003061        True   \n",
       "1                                Serotonin receptors  0.035684        True   \n",
       "0                    Metapathway signal transduction  0.003061        True   \n",
       "1                                Serotonin receptors  0.035684        True   \n",
       "\n",
       "                                          description term_size query_size  \\\n",
       "0   \"A conserved heterotrimeric protein complex th...         2         71   \n",
       "1   \"The contents of a cell excluding the plasma m...      4587         71   \n",
       "2   \"The series of molecular signals initiated by ...       144         67   \n",
       "3   \"A filament of myosin found in a muscle cell o...         4         71   \n",
       "0   \"Catalysis of a biochemical reaction at physio...      4722        101   \n",
       "..                                                ...       ...        ...   \n",
       "23  \"A component of the cytoskeleton consisting of...       167         41   \n",
       "0                     Metapathway signal transduction       106          7   \n",
       "1                                 Serotonin receptors         6         15   \n",
       "0                     Metapathway signal transduction       106          7   \n",
       "1                                 Serotonin receptors         6         15   \n",
       "\n",
       "   intersection_size effective_domain_size  precision    recall    query  \\\n",
       "0                  2                 12900   0.028169  1.000000  query_1   \n",
       "1                 42                 12900   0.591549  0.009156  query_1   \n",
       "2                  7                 10936   0.104478  0.048611  query_1   \n",
       "3                  2                 12900   0.028169  0.500000  query_1   \n",
       "0                 60                 11728   0.594059  0.012706  query_1   \n",
       "..               ...                   ...        ...       ...      ...   \n",
       "23                 5                 12900   0.121951  0.029940  query_1   \n",
       "0                  5                   897   0.714286  0.047170  query_1   \n",
       "1                  2                  4004   0.133333  0.333333  query_1   \n",
       "0                  5                   897   0.714286  0.047170  query_1   \n",
       "1                  2                  4004   0.133333  0.333333  query_1   \n",
       "\n",
       "                     parents  \\\n",
       "0               [GO:0032991]   \n",
       "1   [GO:0005622, GO:0110165]   \n",
       "2               [GO:0007166]   \n",
       "3   [GO:0016460, GO:0043292]   \n",
       "0               [GO:0003674]   \n",
       "..                       ...   \n",
       "23  [GO:0005856, GO:0099512]   \n",
       "0                [WP:000000]   \n",
       "1        [REAC:R-CEL-375280]   \n",
       "0                [WP:000000]   \n",
       "1        [REAC:R-CEL-375280]   \n",
       "\n",
       "                                        intersections  \\\n",
       "0                                    [chaf-2, chaf-1]   \n",
       "1   [uba-1, pak-2, daf-9, cpna-1, hda-4, myo-2, lr...   \n",
       "2   [hda-4, src-2, unc-2, let-23, daf-4, rgl-1, ro...   \n",
       "3                                      [myo-2, mlc-3]   \n",
       "0   [ksr-2, gcy-23, plc-1, cam-1, src-2, dkf-2, un...   \n",
       "..                                                ...   \n",
       "23                [sma-1, mua-3, ify-1, alp-1, dlc-3]   \n",
       "0           [daf-18, daf-12, let-363, lin-17, sfrp-1]   \n",
       "1                                      [dop-1, dop-6]   \n",
       "0           [daf-18, daf-12, let-363, lin-17, sfrp-1]   \n",
       "1                                      [dop-1, dop-6]   \n",
       "\n",
       "                                            evidences  Kclust  \\\n",
       "0                                      [[IBA], [IBA]]    8493   \n",
       "1   [[IBA], [IBA], [IDA, IBA], [IDA, IEA], [ISS], ...    8493   \n",
       "2   [[IMP], [IBA], [IGI], [IDA, ISS, IEA], [IMP, I...    8493   \n",
       "3                                      [[ISS], [IBA]]    8493   \n",
       "0   [[IGI, IBA, IEA], [ISS, IBA, IEA], [IDA, IBA, ...     744   \n",
       "..                                                ...     ...   \n",
       "23                [[IEA], [IDA], [IDA], [IBA], [IEA]]    6077   \n",
       "0                      [[WP], [WP], [WP], [WP], [WP]]    6818   \n",
       "1                                    [[REAC], [REAC]]    6818   \n",
       "0                      [[WP], [WP], [WP], [WP], [WP]]    6818   \n",
       "1                                    [[REAC], [REAC]]    6818   \n",
       "\n",
       "                                            qgenelist  \n",
       "0   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "1   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "2   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "3   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "0   ['ksr-2', 'gcy-23', 'plc-1', 'cam-1', 'src-2',...  \n",
       "..                                                ...  \n",
       "23  ['sma-1', 'sma-1', 'gbf-1', 'unc-94', 'glh-3',...  \n",
       "0   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "1   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "0   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "1   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "\n",
       "[15271 rows x 18 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# This give the gene onotology terms of the proteins after being enrinched\n",
    "pd.concat(GOs)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6ac2722-ead1-414a-b1b3-f59cc645b08a",
   "metadata": {},
   "source": [
    "df_pkl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "77faa2eb-0715-4a70-b41b-11d9be8e86b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Echeh\\AppData\\Local\\Temp\\ipykernel_15152\\3727389055.py:1: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  newdf = pd.concat(GOs)[pd.concat(GOs)[\"source\"]!=\"TF\"]\n"
     ]
    }
   ],
   "source": [
    "newdf = pd.concat(GOs)[pd.concat(GOs)[\"source\"]!=\"TF\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "31b57e88-5626-4159-9bc8-d57019c6757f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>qgenelist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0033186</td>\n",
       "      <td>CAF-1 complex</td>\n",
       "      <td>0.005045</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A conserved heterotrimeric protein complex th...</td>\n",
       "      <td>2</td>\n",
       "      <td>71</td>\n",
       "      <td>2</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0032991]</td>\n",
       "      <td>[chaf-2, chaf-1]</td>\n",
       "      <td>[[IBA], [IBA]]</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005737</td>\n",
       "      <td>cytoplasm</td>\n",
       "      <td>0.006956</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The contents of a cell excluding the plasma m...</td>\n",
       "      <td>4587</td>\n",
       "      <td>71</td>\n",
       "      <td>42</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.591549</td>\n",
       "      <td>0.009156</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005622, GO:0110165]</td>\n",
       "      <td>[uba-1, pak-2, daf-9, cpna-1, hda-4, myo-2, lr...</td>\n",
       "      <td>[[IBA], [IBA], [IDA, IBA], [IDA, IEA], [ISS], ...</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:BP</td>\n",
       "      <td>GO:0007167</td>\n",
       "      <td>enzyme-linked receptor protein signaling pathway</td>\n",
       "      <td>0.021501</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The series of molecular signals initiated by ...</td>\n",
       "      <td>144</td>\n",
       "      <td>67</td>\n",
       "      <td>7</td>\n",
       "      <td>10936</td>\n",
       "      <td>0.104478</td>\n",
       "      <td>0.048611</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0007166]</td>\n",
       "      <td>[hda-4, src-2, unc-2, let-23, daf-4, rgl-1, ro...</td>\n",
       "      <td>[[IMP], [IBA], [IGI], [IDA, ISS, IEA], [IMP, I...</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005859</td>\n",
       "      <td>muscle myosin complex</td>\n",
       "      <td>0.030056</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A filament of myosin found in a muscle cell o...</td>\n",
       "      <td>4</td>\n",
       "      <td>71</td>\n",
       "      <td>2</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0016460, GO:0043292]</td>\n",
       "      <td>[myo-2, mlc-3]</td>\n",
       "      <td>[[ISS], [IBA]]</td>\n",
       "      <td>8493</td>\n",
       "      <td>['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0003824</td>\n",
       "      <td>catalytic activity</td>\n",
       "      <td>0.018819</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Catalysis of a biochemical reaction at physio...</td>\n",
       "      <td>4722</td>\n",
       "      <td>101</td>\n",
       "      <td>60</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.594059</td>\n",
       "      <td>0.012706</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0003674]</td>\n",
       "      <td>[ksr-2, gcy-23, plc-1, cam-1, src-2, dkf-2, un...</td>\n",
       "      <td>[[IGI, IBA, IEA], [ISS, IBA, IEA], [IDA, IBA, ...</td>\n",
       "      <td>744</td>\n",
       "      <td>['ksr-2', 'gcy-23', 'plc-1', 'cam-1', 'src-2',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0099513</td>\n",
       "      <td>polymeric cytoskeletal fiber</td>\n",
       "      <td>0.025148</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A component of the cytoskeleton consisting of...</td>\n",
       "      <td>167</td>\n",
       "      <td>41</td>\n",
       "      <td>5</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.121951</td>\n",
       "      <td>0.029940</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005856, GO:0099512]</td>\n",
       "      <td>[sma-1, mua-3, ify-1, alp-1, dlc-3]</td>\n",
       "      <td>[[IEA], [IDA], [IDA], [IBA], [IEA]]</td>\n",
       "      <td>6077</td>\n",
       "      <td>['sma-1', 'sma-1', 'gbf-1', 'unc-94', 'glh-3',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP1546</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>0.003061</td>\n",
       "      <td>True</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>106</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>897</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.047170</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "      <td>[daf-18, daf-12, let-363, lin-17, sfrp-1]</td>\n",
       "      <td>[[WP], [WP], [WP], [WP], [WP]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REAC</td>\n",
       "      <td>REAC:R-CEL-390666</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>0.035684</td>\n",
       "      <td>True</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>4004</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[REAC:R-CEL-375280]</td>\n",
       "      <td>[dop-1, dop-6]</td>\n",
       "      <td>[[REAC], [REAC]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WP</td>\n",
       "      <td>WP:WP1546</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>0.003061</td>\n",
       "      <td>True</td>\n",
       "      <td>Metapathway signal transduction</td>\n",
       "      <td>106</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>897</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.047170</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[WP:000000]</td>\n",
       "      <td>[daf-18, daf-12, let-363, lin-17, sfrp-1]</td>\n",
       "      <td>[[WP], [WP], [WP], [WP], [WP]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>REAC</td>\n",
       "      <td>REAC:R-CEL-390666</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>0.035684</td>\n",
       "      <td>True</td>\n",
       "      <td>Serotonin receptors</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>4004</td>\n",
       "      <td>0.133333</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[REAC:R-CEL-375280]</td>\n",
       "      <td>[dop-1, dop-6]</td>\n",
       "      <td>[[REAC], [REAC]]</td>\n",
       "      <td>6818</td>\n",
       "      <td>['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15271 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   source             native  \\\n",
       "0   GO:CC         GO:0033186   \n",
       "1   GO:CC         GO:0005737   \n",
       "2   GO:BP         GO:0007167   \n",
       "3   GO:CC         GO:0005859   \n",
       "0   GO:MF         GO:0003824   \n",
       "..    ...                ...   \n",
       "23  GO:CC         GO:0099513   \n",
       "0      WP          WP:WP1546   \n",
       "1    REAC  REAC:R-CEL-390666   \n",
       "0      WP          WP:WP1546   \n",
       "1    REAC  REAC:R-CEL-390666   \n",
       "\n",
       "                                                name   p_value significant  \\\n",
       "0                                      CAF-1 complex  0.005045        True   \n",
       "1                                          cytoplasm  0.006956        True   \n",
       "2   enzyme-linked receptor protein signaling pathway  0.021501        True   \n",
       "3                              muscle myosin complex  0.030056        True   \n",
       "0                                 catalytic activity  0.018819        True   \n",
       "..                                               ...       ...         ...   \n",
       "23                      polymeric cytoskeletal fiber  0.025148        True   \n",
       "0                    Metapathway signal transduction  0.003061        True   \n",
       "1                                Serotonin receptors  0.035684        True   \n",
       "0                    Metapathway signal transduction  0.003061        True   \n",
       "1                                Serotonin receptors  0.035684        True   \n",
       "\n",
       "                                          description term_size query_size  \\\n",
       "0   \"A conserved heterotrimeric protein complex th...         2         71   \n",
       "1   \"The contents of a cell excluding the plasma m...      4587         71   \n",
       "2   \"The series of molecular signals initiated by ...       144         67   \n",
       "3   \"A filament of myosin found in a muscle cell o...         4         71   \n",
       "0   \"Catalysis of a biochemical reaction at physio...      4722        101   \n",
       "..                                                ...       ...        ...   \n",
       "23  \"A component of the cytoskeleton consisting of...       167         41   \n",
       "0                     Metapathway signal transduction       106          7   \n",
       "1                                 Serotonin receptors         6         15   \n",
       "0                     Metapathway signal transduction       106          7   \n",
       "1                                 Serotonin receptors         6         15   \n",
       "\n",
       "   intersection_size effective_domain_size  precision    recall    query  \\\n",
       "0                  2                 12900   0.028169  1.000000  query_1   \n",
       "1                 42                 12900   0.591549  0.009156  query_1   \n",
       "2                  7                 10936   0.104478  0.048611  query_1   \n",
       "3                  2                 12900   0.028169  0.500000  query_1   \n",
       "0                 60                 11728   0.594059  0.012706  query_1   \n",
       "..               ...                   ...        ...       ...      ...   \n",
       "23                 5                 12900   0.121951  0.029940  query_1   \n",
       "0                  5                   897   0.714286  0.047170  query_1   \n",
       "1                  2                  4004   0.133333  0.333333  query_1   \n",
       "0                  5                   897   0.714286  0.047170  query_1   \n",
       "1                  2                  4004   0.133333  0.333333  query_1   \n",
       "\n",
       "                     parents  \\\n",
       "0               [GO:0032991]   \n",
       "1   [GO:0005622, GO:0110165]   \n",
       "2               [GO:0007166]   \n",
       "3   [GO:0016460, GO:0043292]   \n",
       "0               [GO:0003674]   \n",
       "..                       ...   \n",
       "23  [GO:0005856, GO:0099512]   \n",
       "0                [WP:000000]   \n",
       "1        [REAC:R-CEL-375280]   \n",
       "0                [WP:000000]   \n",
       "1        [REAC:R-CEL-375280]   \n",
       "\n",
       "                                        intersections  \\\n",
       "0                                    [chaf-2, chaf-1]   \n",
       "1   [uba-1, pak-2, daf-9, cpna-1, hda-4, myo-2, lr...   \n",
       "2   [hda-4, src-2, unc-2, let-23, daf-4, rgl-1, ro...   \n",
       "3                                      [myo-2, mlc-3]   \n",
       "0   [ksr-2, gcy-23, plc-1, cam-1, src-2, dkf-2, un...   \n",
       "..                                                ...   \n",
       "23                [sma-1, mua-3, ify-1, alp-1, dlc-3]   \n",
       "0           [daf-18, daf-12, let-363, lin-17, sfrp-1]   \n",
       "1                                      [dop-1, dop-6]   \n",
       "0           [daf-18, daf-12, let-363, lin-17, sfrp-1]   \n",
       "1                                      [dop-1, dop-6]   \n",
       "\n",
       "                                            evidences  Kclust  \\\n",
       "0                                      [[IBA], [IBA]]    8493   \n",
       "1   [[IBA], [IBA], [IDA, IBA], [IDA, IEA], [ISS], ...    8493   \n",
       "2   [[IMP], [IBA], [IGI], [IDA, ISS, IEA], [IMP, I...    8493   \n",
       "3                                      [[ISS], [IBA]]    8493   \n",
       "0   [[IGI, IBA, IEA], [ISS, IBA, IEA], [IDA, IBA, ...     744   \n",
       "..                                                ...     ...   \n",
       "23                [[IEA], [IDA], [IDA], [IBA], [IEA]]    6077   \n",
       "0                      [[WP], [WP], [WP], [WP], [WP]]    6818   \n",
       "1                                    [[REAC], [REAC]]    6818   \n",
       "0                      [[WP], [WP], [WP], [WP], [WP]]    6818   \n",
       "1                                    [[REAC], [REAC]]    6818   \n",
       "\n",
       "                                            qgenelist  \n",
       "0   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "1   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "2   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "3   ['uba-1', 'pak-2', 'daf-9', 'cpna-1', 'jmjc-1'...  \n",
       "0   ['ksr-2', 'gcy-23', 'plc-1', 'cam-1', 'src-2',...  \n",
       "..                                                ...  \n",
       "23  ['sma-1', 'sma-1', 'gbf-1', 'unc-94', 'glh-3',...  \n",
       "0   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "1   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "0   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "1   ['unc-55', 'daf-18', 'daf-12', 'myo-2', 'myo-2...  \n",
       "\n",
       "[15271 rows x 18 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "newdf \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "7ebd8099-4b37-441a-96bc-36e06c2d7f13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Othermyos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>8493</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>744</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>5650</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>11549</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>11268</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>1910</td>\n",
       "      <td>1339</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>383</th>\n",
       "      <td>1915</td>\n",
       "      <td>18374</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>384</th>\n",
       "      <td>1920</td>\n",
       "      <td>6077</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>1925</td>\n",
       "      <td>6818</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>1930</td>\n",
       "      <td>6818</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>387 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Start  Kclust Othermyos\n",
       "0        0    8493     False\n",
       "1        5     744       NaN\n",
       "2       10    5650     False\n",
       "3       15   11549     False\n",
       "4       20   11268       NaN\n",
       "..     ...     ...       ...\n",
       "382   1910    1339       NaN\n",
       "383   1915   18374     False\n",
       "384   1920    6077     False\n",
       "385   1925    6818     False\n",
       "386   1930    6818     False\n",
       "\n",
       "[387 rows x 3 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "c4b92a78-3062-4a73-bf1d-556f2b16c1da",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Used For loop to iterate the resdf file then specifying the number of Go terms associated in each Kclust files\n",
    "lengo=[]\n",
    "for i,row in resdf.iterrows():\n",
    "    Kclust = row[\"Kclust\"]\n",
    "    godf=pd.read_excel(\"cluster_%i_GOs.xlsx\" % Kclust)\n",
    "    lengo.append(len(godf))\n",
    "resdf[\"Number of GO term\"]=lengo\n",
    "resdf.to_excel(\"resdf2.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c8b14c2d-7436-4105-bc67-1e4282214f5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "26e4057e-4190-4aff-8d17-5efb286a8aac",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib qt\n",
    "\n",
    "fig,ax = plt.subplots(1, figsize=(16,10))\n",
    "ax.plot(resdf[\"Start\"], resdf[\"Number of GO term\"])\n",
    "\n",
    "font1 ={\"size\": 40}\n",
    "font2 = {\"size\": 20}\n",
    "\n",
    "ax.set_title(\"GO terms-function of sequence position\", fontdict= font1)\n",
    "ax.set_xlabel(\"Start\", fontdict = font2)\n",
    "ax.set_ylabel(\"Number of GO term\", fontdict = font2)\n",
    "ax.tick_params(axis='both', which='major', labelsize=18)\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "fd628552-edb9-4043-8e7b-94ce3456129d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "4f1cbaec-3e78-4b18-93f7-993bb4202c77",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example bin size (I can choose to adjust this when I want)\n",
    "bin_size = 50\n",
    "\n",
    "# Bin amino acid positions (I binned the amino acid positions in order to accomodate all start ids opn the map)\n",
    "resdf[\"AA_bin\"] = (resdf[\"Start\"] // bin_size) * bin_size\n",
    "\n",
    "# Aggregate the number of GO terms per bin\n",
    "binned_data = resdf.groupby(\"AA_bin\")[\"Number of GO term\"].sum().reset_index()\n",
    "\n",
    "#  Then Prepared 2D array for heatmap (1 row, many columns)\n",
    "heat_data = np.array([binned_data[\"Number of GO term\"].values])\n",
    "\n",
    "# Plot heatmap\n",
    "#plt.figure(figsize=(20, 3)) - this gave a smaller heat map\n",
    "fig,ax = plt.subplots(1, figsize=(12, 8))\n",
    "ax = sns.heatmap(\n",
    "    heat_data,\n",
    "    cmap=\"YlGnBu\",\n",
    "    cbar_kws={'label': 'Number of GO terms'},\n",
    "    xticklabels=binned_data[\"AA_bin\"].astype(str),\n",
    "    yticklabels=False\n",
    ")\n",
    "# Increasing the colorbar label (legend title) font size\n",
    "colorbar = ax.collections[0].colorbar\n",
    "colorbar.ax.set_ylabel('Number of GO terms', fontsize=25) \n",
    "\n",
    "# formatting the map to my choice settings\n",
    "plt.title(\"GO terms-function of sequence position\", fontsize=40)\n",
    "plt.xlabel(f\"Amino Acid Position (bins of {bin_size})\", fontsize=35)\n",
    "plt.ylabel(\"\")\n",
    "ax.tick_params(axis='x', labelsize=12, rotation=90)\n",
    "plt.tight_layout()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "3856ee8e-791f-441d-8c50-333afd6fcd6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# reading file that I extracted from uniprot to develop my question even further\n",
    "df_domain =pd.read_excel(r\"C:\\Users\\Echeh\\Downloads\\Masters Pipeline\\domain.xlsx\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "6c081fd1-cd9a-461f-a307-43d67ca9e03f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Type</th>\n",
       "      <th>Position(s)</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Domain</td>\n",
       "      <td>28-77</td>\n",
       "      <td>Myosin N-terminal SH3-like</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Domain</td>\n",
       "      <td>81-794</td>\n",
       "      <td>Myosin motor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Region</td>\n",
       "      <td>668-690</td>\n",
       "      <td>Actin-binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Region</td>\n",
       "      <td>773-787</td>\n",
       "      <td>Actin-binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Region</td>\n",
       "      <td>856-1947</td>\n",
       "      <td>Rodlike tail (S2 and LMM domains)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Region</td>\n",
       "      <td>1047-1075</td>\n",
       "      <td>Disordered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Compositional bias</td>\n",
       "      <td>1053-1075</td>\n",
       "      <td>Basic and acidic residues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Region</td>\n",
       "      <td>1119-1161</td>\n",
       "      <td>Disordered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Compositional bias</td>\n",
       "      <td>1125-1155</td>\n",
       "      <td>Basic and acidic residues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Compositional bias</td>\n",
       "      <td>1231-1245</td>\n",
       "      <td>Basic and acidic residues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Region</td>\n",
       "      <td>1231-1253</td>\n",
       "      <td>Disordered</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Compositional bias</td>\n",
       "      <td>1516-1529</td>\n",
       "      <td>Basic and acidic residues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Region</td>\n",
       "      <td>1516-1537</td>\n",
       "      <td>Disordered</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Type Position(s)                        Description\n",
       "0               Domain       28-77         Myosin N-terminal SH3-like\n",
       "1               Domain      81-794                     Myosin motor  \n",
       "2               Region     668-690                      Actin-binding\n",
       "3               Region     773-787                      Actin-binding\n",
       "4               Region    856-1947  Rodlike tail (S2 and LMM domains)\n",
       "5               Region   1047-1075                         Disordered\n",
       "6   Compositional bias   1053-1075          Basic and acidic residues\n",
       "7               Region   1119-1161                         Disordered\n",
       "8   Compositional bias   1125-1155          Basic and acidic residues\n",
       "9   Compositional bias   1231-1245          Basic and acidic residues\n",
       "10              Region   1231-1253                         Disordered\n",
       "11  Compositional bias   1516-1529          Basic and acidic residues\n",
       "12              Region   1516-1537                         Disordered"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_domain\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "14e547d1-02f7-4637-9846-631e7ec44163",
   "metadata": {},
   "outputs": [],
   "source": [
    "# the ID name for the C.elegans used was P12845 · MYO2_CAEEL\n",
    "df_domain\n",
    "\n",
    "import numpy as np\n",
    "import matplotlib.patches as patches\n",
    "\n",
    "# the Parameter I used\n",
    "bin_size = 50\n",
    "\n",
    "# Bin resdf for GO terms \n",
    "resdf[\"AA_bin\"] = (resdf[\"Start\"] // bin_size) * bin_size\n",
    "binned_data = resdf.groupby(\"AA_bin\")[\"Number of GO term\"].sum().reset_index()\n",
    "heat_data = np.array([binned_data[\"Number of GO term\"].values])\n",
    "\n",
    "# I also binned the df_domain plot axis to ease visualization \n",
    "df_domain[[\"Start\", \"End\"]] = df_domain[\"Position(s)\"].str.split(\"-\", expand=True).astype(int)\n",
    "df_domain[\"Bin_start\"] = (df_domain[\"Start\"] // bin_size) * bin_size\n",
    "df_domain[\"Bin_end\"] = (df_domain[\"End\"] // bin_size) * bin_size\n",
    "\n",
    "# Then I added some Domain colors for the legend(key identification) \n",
    "colors = {\n",
    "    \"Domain\": \"red\",\n",
    "    \"Region\": \"blue\",\n",
    "    \"Compositional bias\": \"purple\"\n",
    "}\n",
    "\n",
    "# Also Assigned y-positions per domain type ---\n",
    "unique_types = df_domain[\"Type\"].unique()\n",
    "y_positions = {typ: i for i, typ in enumerate(unique_types[::-1])}\n",
    "\n",
    "# created 2 subplots to aid visualization\n",
    "fig, (ax1, ax2) = plt.subplots(\n",
    "    2, 1, figsize=(16, 8), sharex=True,\n",
    "    gridspec_kw={\"height_ratios\": [1, 3]},\n",
    "    constrained_layout=True\n",
    ")\n",
    "\n",
    "# The Top subplot showing the Domain bars\n",
    "for _, row in df_domain.iterrows():\n",
    "    x_start_bin = row[\"Bin_start\"]\n",
    "    x_end_bin = row[\"Bin_end\"]\n",
    "\n",
    "    # Skip if domain is out of binned range\n",
    "    if x_end_bin < binned_data[\"AA_bin\"].min() or x_start_bin > binned_data[\"AA_bin\"].max():\n",
    "        continue\n",
    "\n",
    "    try:\n",
    "        x_start_idx = binned_data[binned_data[\"AA_bin\"] == x_start_bin].index[0]\n",
    "    except IndexError:\n",
    "        continue  # skip if start bin not found\n",
    "\n",
    "    try:\n",
    "        x_end_idx = binned_data[binned_data[\"AA_bin\"] == x_end_bin].index[0]\n",
    "    except IndexError:\n",
    "        x_end_idx = x_start_idx  # treated as a narrow feature\n",
    "\n",
    "    y = y_positions[row[\"Type\"]]\n",
    "\n",
    "    ax1.barh(\n",
    "        y=y,\n",
    "        width=x_end_idx - x_start_idx + 1,\n",
    "        left=x_start_idx,\n",
    "        height=0.8,\n",
    "        color=colors.get(row[\"Type\"], \"gray\"),\n",
    "        edgecolor='black',\n",
    "        alpha=0.7\n",
    "    )\n",
    "\n",
    "# Formatting domain track\n",
    "ax1.set_yticks(list(y_positions.values()))\n",
    "ax1.set_yticklabels(list(y_positions.keys()), fontsize=15)\n",
    "ax1.set_xlim([0, len(binned_data) - 1])\n",
    "ax1.set_title(\"Protein Domain Architecture\", fontsize=18)\n",
    "ax1.set_ylabel(\"Domain Features\", fontsize=18)\n",
    "\n",
    "# Legend\n",
    "legend_elements = [\n",
    "    patches.Patch(facecolor=color, edgecolor='black', label=typ)\n",
    "    for typ, color in colors.items()\n",
    "]\n",
    "ax1.legend(handles=legend_elements, title=\"Domain Features\", fontsize=14, loc='upper right', bbox_to_anchor=(1.15, 1))\n",
    "\n",
    "# BOTTOM SUBPLOT: Heatmap \n",
    "heatmap=sns.heatmap(\n",
    "    heat_data,\n",
    "    cmap=\"YlGnBu\",\n",
    "    cbar_kws={'label': 'Number of GO terms'},\n",
    "    xticklabels=binned_data[\"AA_bin\"].astype(str),\n",
    "    yticklabels=False,\n",
    "    ax=ax2\n",
    ")\n",
    "# Increasing the colorbar label (legend title) font size\n",
    "colorbar = heatmap.collections[0].colorbar\n",
    "colorbar.ax.set_ylabel('Number of GO terms', fontsize=25) \n",
    "\n",
    "\n",
    "# Formatting x-axis labels\n",
    "xticks = binned_data[\"AA_bin\"].astype(str)\n",
    "ax2.set_xticks(np.arange(len(xticks))[::4])\n",
    "ax2.set_xticklabels(xticks[::4], rotation=90, fontsize=12, fontweight=\"bold\")\n",
    "\n",
    "# Axis labels\n",
    "ax2.set_title(\"GO Terms Across Amino Acid Positions\", fontsize=20, fontweight=\"bold\")\n",
    "ax2.set_xlabel(f\"Amino Acid Position (bins of {bin_size})\", fontsize=14, fontweight=\"bold\")\n",
    "\n",
    "plt.show()\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e171611f-d137-4d39-9442-ad5dd0525ba2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This Hovers the description tags over the heat map plot, not necessarily used but some descriptions were overlapping\n",
    "from matplotlib.patches import Rectangle\n",
    "\n",
    "# Your bin size\n",
    "bin_size = 50\n",
    "\n",
    "# Bin GO term data\n",
    "resdf[\"AA_bin\"] = (resdf[\"Start\"] // bin_size) * bin_size\n",
    "binned_data = resdf.groupby(\"AA_bin\")[\"Number of GO term\"].sum().reset_index()\n",
    "heat_data = np.array([binned_data[\"Number of GO term\"].values])\n",
    "\n",
    "# Processed domain table\n",
    "df_domain[\"Start\"] = df_domain[\"Position(s)\"].str.extract(r\"(\\d+)\", expand=False).astype(int)\n",
    "df_domain[\"End\"] = df_domain[\"Position(s)\"].str.extract(r\"-(\\d+)\", expand=False).astype(int)\n",
    "\n",
    "# Assigned y positions to avoid overlap (just stack them: 1.2, 1.4, 1.6, ...)\n",
    "label_y_positions = np.linspace(0.3, 0.8, len(df_domain))  # Customize upper limit based on number of domains\n",
    "\n",
    "# Plotted heatmap\n",
    "fig, ax = plt.subplots(1, figsize=(10, 3))\n",
    "sns.heatmap(\n",
    "    heat_data,\n",
    "    cmap=\"YlGnBu\",\n",
    "    cbar_kws={'label': 'Number of GO terms'},\n",
    "    xticklabels=binned_data[\"AA_bin\"].astype(str),\n",
    "    yticklabels=False,\n",
    "    ax=ax\n",
    ")\n",
    "\n",
    "# Drew some rectangles and stack text labels in them usoing thick lines as the border of each description\n",
    "for idx, (i, row) in enumerate(df_domain.iterrows()):\n",
    "    start_bin = (row[\"Start\"] // bin_size)\n",
    "    end_bin = (row[\"End\"] // bin_size)\n",
    "    rect = Rectangle(\n",
    "        (start_bin, 0), \n",
    "        end_bin - start_bin + 1, \n",
    "        1,\n",
    "        linewidth=2.5,\n",
    "        edgecolor='red',\n",
    "        facecolor='none',\n",
    "        linestyle='--'\n",
    "    )\n",
    "    ax.add_patch(rect)\n",
    "    \n",
    "    # Placed label at increasing y axis\n",
    "    ax.text(\n",
    "        x=(start_bin + end_bin) / 2,\n",
    "        y=label_y_positions[idx],\n",
    "        s=row[\"Description\"],\n",
    "        ha='center',\n",
    "        va='bottom',\n",
    "        fontsize=20,\n",
    "        rotation=90,\n",
    "        color='red',\n",
    "        clip_on=False\n",
    "    )\n",
    "\n",
    "# Formatting the heatmap\n",
    "plt.title(\"GO terms - function of sequence position\", fontsize=30, fontweight=\"bold\")\n",
    "plt.xlabel(f\"Amino Acid Position (bins of {bin_size})\", fontsize=18, fontweight=\"bold\")\n",
    "plt.ylabel(\"\")\n",
    "ax.tick_params(axis='x', labelsize=10, rotation=60, fontweight=\"bold\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "3e6de41d-37cf-48c5-b5f1-a03a834f145e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "\n",
    "# ID of C. elegans protein used: P12845 · MYO2_CAEEL\n",
    "\n",
    "# The Parameter I used\n",
    "bin_size = 50\n",
    "\n",
    "# Then I Processed GO term data (resdf)  \n",
    "resdf[\"AA_bin\"] = (resdf[\"Start\"] // bin_size) * bin_size\n",
    "binned_data = resdf.groupby(\"AA_bin\")[\"Number of GO term\"].sum().reset_index()\n",
    "heat_data = np.array([binned_data[\"Number of GO term\"].values])\n",
    "\n",
    "# I further Processed domain data (df_domain) to get bin size 50 for 0 - 49, 50 - 99\n",
    "df_domain[[\"Start\", \"End\"]] = df_domain[\"Position(s)\"].str.split(\"-\", expand=True).astype(int)\n",
    "df_domain[\"Bin_start\"] = (df_domain[\"Start\"] // bin_size) * bin_size\n",
    "df_domain[\"Bin_end\"] = (df_domain[\"End\"] // bin_size) * bin_size\n",
    "\n",
    "# Clean and standardize domain descriptions\n",
    "df_domain[\"Description\"] = df_domain[\"Description\"].str.strip()\n",
    "\n",
    "# Added Color mapping for domain descriptions to be  visualize on the sub plot\n",
    "colors = {\n",
    "    \"Myosin N-terminal SH3-like\": \"blue\",\n",
    "    \"Myosin motor\": \"red\",\n",
    "    \"Actin-binding\": \"violet\",\n",
    "    \"Rodlike tail (S2 and LMM domains)\": \"purple\",\n",
    "    \"Disordered\": \"green\",\n",
    "    \"Basic and acidic residues\": \"orange\"\n",
    "}\n",
    "\n",
    "# Assigned vertical positions for each domain description\n",
    "unique_types = df_domain[\"Description\"].unique()\n",
    "y_positions = {desc: i for i, desc in enumerate(reversed(unique_types))}\n",
    "\n",
    "# Created 2 subplots: Top for domain descriptions, Bottom for GO term heatmap\n",
    "fig, (ax1, ax2) = plt.subplots(\n",
    "    2, 1,\n",
    "    figsize=(12, 8),\n",
    "    sharex=True,\n",
    "    gridspec_kw={\"height_ratios\": [1, 2]},\n",
    "    constrained_layout=True\n",
    ")\n",
    "\n",
    "# Plot domain descriptions in top subplot (ax1)\n",
    "for _, row in df_domain.iterrows():\n",
    "    start_bin = row[\"Bin_start\"]\n",
    "    end_bin = row[\"Bin_end\"]\n",
    "    desc = row[\"Description\"]  # cleaned version used for matching\n",
    "\n",
    "    # Skip domain if completely out of range\n",
    "    if end_bin < binned_data[\"AA_bin\"].min() or start_bin > binned_data[\"AA_bin\"].max():\n",
    "        continue\n",
    "\n",
    "    # Map bin positions to indices for plotting width\n",
    "    try:\n",
    "        start_idx = binned_data[binned_data[\"AA_bin\"] == start_bin].index[0]\n",
    "    except IndexError:\n",
    "        continue\n",
    "\n",
    "    try:\n",
    "        end_idx = binned_data[binned_data[\"AA_bin\"] == end_bin].index[0]\n",
    "    except IndexError:\n",
    "        end_idx = start_idx  # fallback if only start is found\n",
    "\n",
    "    y = y_positions[desc]\n",
    "    color = colors.get(desc, \"gray\")  # Use gray if no match found\n",
    "\n",
    "    ax1.barh(\n",
    "        y=y,\n",
    "        width=end_idx - start_idx + 1,\n",
    "        left=start_idx,\n",
    "        height=0.8,\n",
    "        color=color,\n",
    "        edgecolor='black',\n",
    "        alpha=0.7\n",
    "    )\n",
    "\n",
    "# Formating top subplot\n",
    "ax1.set_yticks(list(y_positions.values()))\n",
    "ax1.set_yticklabels(list(y_positions.keys()), fontsize=13)\n",
    "ax1.set_xlim([0, len(binned_data) - 1])\n",
    "ax1.set_title(\"Protein Domain Descriptions\", fontsize=16)\n",
    "ax1.set_ylabel(\"Description\", fontsize=14)\n",
    "\n",
    "# Added legend to top plot\n",
    "legend_elements = [\n",
    "    patches.Patch(facecolor=color, edgecolor='black', label=desc)\n",
    "    for desc, color in colors.items()\n",
    "]\n",
    "ax1.legend(\n",
    "    handles=legend_elements,\n",
    "    title=\"Domain type\",\n",
    "    fontsize=12,\n",
    "    title_fontsize=13,\n",
    "    loc='upper right',\n",
    "    bbox_to_anchor=(1.15, 1)\n",
    ")\n",
    "\n",
    "# Then Plotted heatmap in bottom subplot (ax2) \n",
    "heatmap=sns.heatmap(\n",
    "    heat_data,\n",
    "    cmap=\"YlGnBu\",\n",
    "    cbar_kws={'label': 'Number of GO terms'},\n",
    "    xticklabels=binned_data[\"AA_bin\"].astype(str),\n",
    "    yticklabels=False,\n",
    "    ax=ax2\n",
    ")\n",
    "# Increasing the colorbar label (legend title) font size\n",
    "colorbar = heatmap.collections[0].colorbar\n",
    "colorbar.ax.set_ylabel('Number of GO terms', fontsize=25) \n",
    "\n",
    "\n",
    "\n",
    "# Formatting bottom subplot\n",
    "xticks = binned_data[\"AA_bin\"].astype(str)\n",
    "ax2.set_xticks(np.arange(len(xticks))[::4])\n",
    "ax2.set_xticklabels(xticks[::4], rotation=90, fontsize=12, fontweight=\"bold\")\n",
    "\n",
    "ax2.set_title(\"GO Terms Across Amino Acid Positions\", fontsize=18, fontweight=\"bold\")\n",
    "ax2.set_xlabel(f\"Amino Acid Position (binned, size={bin_size})\", fontsize=14, fontweight=\"bold\")\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "80dffeef-7c79-4c93-980b-90775b4ffe1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.patches import Patch"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eaf352b7-be66-4787-81fb-8ce7d04c57d0",
   "metadata": {},
   "source": [
    "# Reading Excel file with high GO terms in relation to various regions for feature extraction\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "e4fc66e2-b908-4ed6-bf1d-05a92f42eb62",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extracting columns\n",
    "columns_to_extract = [\n",
    "    'CA-Gly', 'Pro-MC', 'CB', 'CB-Pro', 'Carboxyl',\n",
    "    'Amide', 'His', 'Trp', 'Phe-Tyr', 'OH-Tyr', 'CG',\n",
    "    'CD', 'CE', 'OH', 'SH', 'S', 'NH3', 'Arg', 'MC'\n",
    "]\n",
    "\n",
    "# define standardization function\n",
    "def standardize_max(reference_data, data):\n",
    "    max_values = np.max(reference_data, axis=1)\n",
    "    min_values = np.min(reference_data, axis=1)\n",
    "    return [\n",
    "        (col - min_val) / (max_val - min_val) if max_val != min_val else col * 0\n",
    "        for col, max_val, min_val in zip(data, max_values, min_values)\n",
    "    ]\n",
    "    \n",
    "   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "1ad5db4f-8648-49aa-8878-3ebcdbd049ff",
   "metadata": {},
   "outputs": [],
   "source": [
    " df_20_000 = pd.read_pickle('mergedbf_unusual_marked_with_20000cluster_label (1).pkl')\n",
    "df_20_000 = df_20_000.rename(columns={'Kclust':'cluster'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "cbbd603e-6d96-4e9f-98a2-167b4204b5d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2425160\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "17768"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(df_20_000))\n",
    "len(df_20_000['cluster'].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71b50b25-67ef-4270-8f74-e06d88fd15c8",
   "metadata": {},
   "source": [
    "c.elegans 1780 start(AA position) Reading Excel file with high GO terms in relation to rod-like tail region for feature extraction with 82 GO terms\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "0d6eee5d-74d2-4ffc-8a7b-c4ca2c9111f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 2640] # c.elegans 1780 start(AA position) Reading Excel file with high GO terms in relation to rod-like\n",
    "# tail region for feature extraction with 82 GO terms\n",
    "\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_MYO2_CAEEL_1780 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_2640_pos_1780'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bf79032-2d52-400a-a30e-d8be212e8249",
   "metadata": {},
   "source": [
    "c.elegans 1300 start(AA position) Reading Excel file with high GO terms in relation to disorderd and tail region with 103 GO terms for feature extraction\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "bc45ec07-896d-42ba-8870-668e4a97ddb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 11619] # c.elegans 1300 start(AA position) Reading Excel file with high GO terms in relation to disorderd and rodlike tail \n",
    "#region with 103 GO terms for feature extraction\n",
    "\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_MYO2_CAEEL_1300 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_11619_pos_1300'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edb5cc03-6f2d-492f-83a3-0cede4f1af81",
   "metadata": {},
   "source": [
    "c.elegans 150 start(AA position) Reading Excel file with high GO terms in relation to myosin motor region with 7 GO terms for feature extraction\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "edcfa839-dcbe-4914-ac3d-b9c987f677d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 12095] # c.elegans 150 start(AA position) Reading Excel file with high GO terms in relation to myosin motor \n",
    "#region with 7 GO terms for feature extraction\n",
    "\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_MYO2_CAEEL_150 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_12095_pos_150'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1cafcab-c43f-4841-af44-143cc338a936",
   "metadata": {},
   "source": [
    "c.elegans 45 start(AA position) Reading Excel file with high GO terms in relation to myosin motor and myosin N-terminal SH3-like region with 54 GO terms for feature extraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "90d280ae-ff28-4360-8477-e826aa9dc217",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 9889] # c.elegans 45 start(AA position) Reading Excel file with high GO terms in relation to myosin motor and myosin\n",
    "#N-terminal SH3-like region with 54 GO terms for feature extraction\n",
    "\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_MYO2_CAEEL_45 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_9889_pos_45'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c3aa506-50f8-4b4b-b74e-17bc46ffc4bc",
   "metadata": {},
   "source": [
    "c.elegans 760 start(AA position) Reading Excel file with high GO terms in relation to Actin binding and myosin motor region with 98 GO terms for feature extraction\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "cbaa3a93-4f58-48df-80a7-db7c85aa77d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 4331] # c.elegans 760 start(AA position) Reading Excel file with high GO terms in relation to Actin binding and myosin \n",
    "# motor region with 98 GO terms for feature extraction\n",
    "\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_MYO2_CAEEL_760 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_4331_pos_760'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77ea87e8-a59e-42d8-ab23-aa7a52731178",
   "metadata": {},
   "source": [
    "c.elegans 1165 start(AA position) Reading Excel file with high GO terms in relation to basic and acidic residue, disorderd and rod-like tail region with 72 GO terms for feature extraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "e89cfb6b-d77c-429d-8137-f68054d469ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 1059] # c.elegans 1165 start(AA position) Reading Excel file with high GO terms in relation to basic and acidic residue\n",
    "# disorderd and rod-like tail region with 72 GO terms for feature extraction\n",
    "\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_MYO2_CAEEL_1165 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_1059_pos_1165'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72908be1-fdcc-4936-92e5-22545dd4b80a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
