{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2aadcc33-a752-4c67-85c2-9393b9d11412",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pickle as pkl\n",
    "from gprofiler import GProfiler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d009e732-7235-476c-b47a-0ddba5bd5318",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>SH</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL</td>\n",
       "      <td>LLKSCVVKLLLPAAL</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>PIEZ1_CAEEL_10_VVKLLLPAALLAAAI</td>\n",
       "      <td>VVKLLLPAALLAAAI</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>17001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>PIEZ1_CAEEL_15_LPAALLAAAIIRPSF</td>\n",
       "      <td>LPAALLAAAIIRPSF</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY</td>\n",
       "      <td>LAAAIIRPSFLSIGY</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425155</th>\n",
       "      <td>95</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>EHPRLHSYSVTPSAP</td>\n",
       "      <td>X5ML99_CAEEL_95_EHPRLHSYSVTPSAP</td>\n",
       "      <td>EHPRLHSYSVTPSAP</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>54</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425156</th>\n",
       "      <td>100</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>HSYSVTPSAPAMETA</td>\n",
       "      <td>X5ML99_CAEEL_100_HSYSVTPSAPAMETA</td>\n",
       "      <td>HSYSVTPSAPAMETA</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425157</th>\n",
       "      <td>105</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>TPSAPAMETAFGHDY</td>\n",
       "      <td>X5ML99_CAEEL_105_TPSAPAMETAFGHDY</td>\n",
       "      <td>TPSAPAMETAFGHDY</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425158</th>\n",
       "      <td>110</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>AMETAFGHDYDPPPS</td>\n",
       "      <td>X5ML99_CAEEL_110_AMETAFGHDYDPPPS</td>\n",
       "      <td>AMETAFGHDYDPPPS</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425159</th>\n",
       "      <td>115</td>\n",
       "      <td>X5ML99</td>\n",
       "      <td>X5ML99_CAEEL</td>\n",
       "      <td>EGF-like domain-containing protein</td>\n",
       "      <td>FGHDYDPPPSYTEAM</td>\n",
       "      <td>X5ML99_CAEEL_115_FGHDYDPPPSYTEAM</td>\n",
       "      <td>FGHDYDPPPSYTEAM</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>18760</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2425160 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start       Entry    Entry name  \\\n",
       "0            0  A0A061ACU2   PIEZ1_CAEEL   \n",
       "1            5  A0A061ACU2   PIEZ1_CAEEL   \n",
       "2           10  A0A061ACU2   PIEZ1_CAEEL   \n",
       "3           15  A0A061ACU2   PIEZ1_CAEEL   \n",
       "4           20  A0A061ACU2   PIEZ1_CAEEL   \n",
       "...        ...         ...           ...   \n",
       "2425155     95      X5ML99  X5ML99_CAEEL   \n",
       "2425156    100      X5ML99  X5ML99_CAEEL   \n",
       "2425157    105      X5ML99  X5ML99_CAEEL   \n",
       "2425158    110      X5ML99  X5ML99_CAEEL   \n",
       "2425159    115      X5ML99  X5ML99_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "0        Piezo-type mechanosensitive ion channel compon...  MTVPPLLKSCVVKLL   \n",
       "1        Piezo-type mechanosensitive ion channel compon...  LLKSCVVKLLLPAAL   \n",
       "2        Piezo-type mechanosensitive ion channel compon...  VVKLLLPAALLAAAI   \n",
       "3        Piezo-type mechanosensitive ion channel compon...  LPAALLAAAIIRPSF   \n",
       "4        Piezo-type mechanosensitive ion channel compon...  LAAAIIRPSFLSIGY   \n",
       "...                                                    ...              ...   \n",
       "2425155                 EGF-like domain-containing protein  EHPRLHSYSVTPSAP   \n",
       "2425156                 EGF-like domain-containing protein  HSYSVTPSAPAMETA   \n",
       "2425157                 EGF-like domain-containing protein  TPSAPAMETAFGHDY   \n",
       "2425158                 EGF-like domain-containing protein  AMETAFGHDYDPPPS   \n",
       "2425159                 EGF-like domain-containing protein  FGHDYDPPPSYTEAM   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "0           PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL  MTVPPLLKSCVVKLL       0       4   \n",
       "1           PIEZ1_CAEEL_5_LLKSCVVKLLLPAAL  LLKSCVVKLLLPAAL       0       2   \n",
       "2          PIEZ1_CAEEL_10_VVKLLLPAALLAAAI  VVKLLLPAALLAAAI       0       2   \n",
       "3          PIEZ1_CAEEL_15_LPAALLAAAIIRPSF  LPAALLAAAIIRPSF       0       4   \n",
       "4          PIEZ1_CAEEL_20_LAAAIIRPSFLSIGY  LAAAIIRPSFLSIGY       1       2   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2425155   X5ML99_CAEEL_95_EHPRLHSYSVTPSAP  EHPRLHSYSVTPSAP       0       6   \n",
       "2425156  X5ML99_CAEEL_100_HSYSVTPSAPAMETA  HSYSVTPSAPAMETA       0       4   \n",
       "2425157  X5ML99_CAEEL_105_TPSAPAMETAFGHDY  TPSAPAMETAFGHDY       1       4   \n",
       "2425158  X5ML99_CAEEL_110_AMETAFGHDYDPPPS  AMETAFGHDYDPPPS       1       6   \n",
       "2425159  X5ML99_CAEEL_115_FGHDYDPPPSYTEAM  FGHDYDPPPSYTEAM       1       6   \n",
       "\n",
       "         CB  ...  SH  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \n",
       "0        13  ...   1  1    2    0  56  False  False  False     False   14908  \n",
       "1        14  ...   1  0    2    0  58  False  False  False     False    7734  \n",
       "2        14  ...   0  0    1    0  58  False  False  False     False   17001  \n",
       "3        13  ...   0  0    0    4  56  False  False  False     False    7557  \n",
       "4        13  ...   0  0    0    4  57  False  False  False     False    8636  \n",
       "...      ..  ...  .. ..  ...  ...  ..    ...    ...    ...       ...     ...  \n",
       "2425155  12  ...   0  0    0    4  54  False  False  False     False    8981  \n",
       "2425156  13  ...   0  1    0    0  56  False  False  False     False   16050  \n",
       "2425157  12  ...   0  1    0    0  55  False  False  False     False   16050  \n",
       "2425158  11  ...   0  1    0    0  53  False  False  False     False    7246  \n",
       "2425159  11  ...   0  1    0    0  53  False  False  False     False   18760  \n",
       "\n",
       "[2425160 rows x 31 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# loading a pickle file into pandas for visualization\n",
    "df_pkl  = pd.read_pickle (r\"C:\\Users\\Echeh\\Music\\Master pipeline\\mergedbf_unusual_marked_with_20000cluster_label (1).pkl\")\n",
    "df_pkl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "10e005c0-b683-4589-886e-e0808c5f8388",
   "metadata": {},
   "outputs": [],
   "source": [
    "#reading the other dataframe to get the column Gene Names (Primary)\n",
    "pf=pd.read_excel(r\"C:\\Users\\Echeh\\Music\\Master pipeline\\uniprotkb_C_elegans_AND_model_organism_2025_01_20.xlsx\")\n",
    "pf=pf[[\"Gene Names (primary)\",\"Entry\"]].set_index(\"Entry\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2a3e784a-e8f4-4c1c-af53-9e79730ea0d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Joined the pf and df_pkl dataframes, using entry column\n",
    "df_pkl=df_pkl.join(pf, on=\"Entry\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e588c5bb-3a42-4cb3-a0ca-68a40efbab40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Prot_res_pep</th>\n",
       "      <th>aa</th>\n",
       "      <th>CA-Gly</th>\n",
       "      <th>Pro-MC</th>\n",
       "      <th>CB</th>\n",
       "      <th>...</th>\n",
       "      <th>S</th>\n",
       "      <th>NH3</th>\n",
       "      <th>Arg</th>\n",
       "      <th>MC</th>\n",
       "      <th>LastC</th>\n",
       "      <th>ManyH</th>\n",
       "      <th>SeCys</th>\n",
       "      <th>PyrroLys</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Gene Names (primary)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A0A061ACU2</td>\n",
       "      <td>PIEZ1_CAEEL</td>\n",
       "      <td>Piezo-type mechanosensitive ion channel compon...</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL</td>\n",
       "      <td>MTVPPLLKSCVVKLL</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>pezo-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10947</th>\n",
       "      <td>4155</td>\n",
       "      <td>G4SLH0</td>\n",
       "      <td>TTN1_CAEEL</td>\n",
       "      <td>Titin homolog (EC 2.7.11.1)</td>\n",
       "      <td>ATLKCIVVGTPLPDV</td>\n",
       "      <td>TTN1_CAEEL_4155_ATLKCIVVGTPLPDV</td>\n",
       "      <td>ATLKCIVVGTPLPDV</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>ttn-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96923</th>\n",
       "      <td>330</td>\n",
       "      <td>P34271</td>\n",
       "      <td>CUP4_CAEEL</td>\n",
       "      <td>Acetylcholine receptor-like protein cup-4 (Coe...</td>\n",
       "      <td>PLSVSEIPFCIKLIG</td>\n",
       "      <td>CUP4_CAEEL_330_PLSVSEIPFCIKLIG</td>\n",
       "      <td>PLSVSEIPFCIKLIG</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>cup-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234219</th>\n",
       "      <td>30</td>\n",
       "      <td>Q9XUC2</td>\n",
       "      <td>IFT22_CAEEL</td>\n",
       "      <td>Intraflagellar transport associated protein 2</td>\n",
       "      <td>ILVLGPPKAGKTTLC</td>\n",
       "      <td>IFT22_CAEEL_30_ILVLGPPKAGKTTLC</td>\n",
       "      <td>ILVLGPPKAGKTTLC</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>54</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>ifta-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236662</th>\n",
       "      <td>150</td>\n",
       "      <td>Q9XW92</td>\n",
       "      <td>VATA_CAEEL</td>\n",
       "      <td>V-type proton ATPase catalytic subunit A (V-AT...</td>\n",
       "      <td>IKHKILLPPSACGTI</td>\n",
       "      <td>VATA_CAEEL_150_IKHKILLPPSACGTI</td>\n",
       "      <td>IKHKILLPPSACGTI</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>vha-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2347550</th>\n",
       "      <td>580</td>\n",
       "      <td>Q9XW73</td>\n",
       "      <td>Q9XW73_CAEEL</td>\n",
       "      <td>Mediator of RNA polymerase II transcription su...</td>\n",
       "      <td>TPFIIVCIKSPSLLV</td>\n",
       "      <td>Q9XW73_CAEEL_580_TPFIIVCIKSPSLLV</td>\n",
       "      <td>TPFIIVCIKSPSLLV</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2360262</th>\n",
       "      <td>30</td>\n",
       "      <td>Q9XXF0</td>\n",
       "      <td>Q9XXF0_CAEEL</td>\n",
       "      <td>Uncharacterized protein</td>\n",
       "      <td>LTPIITAQCSKKPKG</td>\n",
       "      <td>Q9XXF0_CAEEL_30_LTPIITAQCSKKPKG</td>\n",
       "      <td>LTPIITAQCSKKPKG</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2360816</th>\n",
       "      <td>305</td>\n",
       "      <td>Q9XXI0</td>\n",
       "      <td>Q9XXI0_CAEEL</td>\n",
       "      <td>Arrestin C-terminal-like domain-containing pro...</td>\n",
       "      <td>KKLRCELPITVGTIP</td>\n",
       "      <td>Q9XXI0_CAEEL_305_KKLRCELPITVGTIP</td>\n",
       "      <td>KKLRCELPITVGTIP</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>arrd-8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2390435</th>\n",
       "      <td>135</td>\n",
       "      <td>U4PR82</td>\n",
       "      <td>U4PR82_CAEEL</td>\n",
       "      <td>Pyr_redox_2 domain-containing protein</td>\n",
       "      <td>VVACGKTSPRKLQLP</td>\n",
       "      <td>U4PR82_CAEEL_135_VVACGKTSPRKLQLP</td>\n",
       "      <td>VVACGKTSPRKLQLP</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>osgn-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2398461</th>\n",
       "      <td>205</td>\n",
       "      <td>V6CK14</td>\n",
       "      <td>V6CK14_CAEEL</td>\n",
       "      <td>FBD domain-containing protein</td>\n",
       "      <td>CIVNLGIPKTKTFPL</td>\n",
       "      <td>V6CK14_CAEEL_205_CIVNLGIPKTKTFPL</td>\n",
       "      <td>CIVNLGIPKTKTFPL</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>14908</td>\n",
       "      <td>him-19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Start       Entry    Entry name  \\\n",
       "0            0  A0A061ACU2   PIEZ1_CAEEL   \n",
       "10947     4155      G4SLH0    TTN1_CAEEL   \n",
       "96923      330      P34271    CUP4_CAEEL   \n",
       "234219      30      Q9XUC2   IFT22_CAEEL   \n",
       "236662     150      Q9XW92    VATA_CAEEL   \n",
       "...        ...         ...           ...   \n",
       "2347550    580      Q9XW73  Q9XW73_CAEEL   \n",
       "2360262     30      Q9XXF0  Q9XXF0_CAEEL   \n",
       "2360816    305      Q9XXI0  Q9XXI0_CAEEL   \n",
       "2390435    135      U4PR82  U4PR82_CAEEL   \n",
       "2398461    205      V6CK14  V6CK14_CAEEL   \n",
       "\n",
       "                                             Protein names          Peptide  \\\n",
       "0        Piezo-type mechanosensitive ion channel compon...  MTVPPLLKSCVVKLL   \n",
       "10947                          Titin homolog (EC 2.7.11.1)  ATLKCIVVGTPLPDV   \n",
       "96923    Acetylcholine receptor-like protein cup-4 (Coe...  PLSVSEIPFCIKLIG   \n",
       "234219       Intraflagellar transport associated protein 2  ILVLGPPKAGKTTLC   \n",
       "236662   V-type proton ATPase catalytic subunit A (V-AT...  IKHKILLPPSACGTI   \n",
       "...                                                    ...              ...   \n",
       "2347550  Mediator of RNA polymerase II transcription su...  TPFIIVCIKSPSLLV   \n",
       "2360262                            Uncharacterized protein  LTPIITAQCSKKPKG   \n",
       "2360816  Arrestin C-terminal-like domain-containing pro...  KKLRCELPITVGTIP   \n",
       "2390435              Pyr_redox_2 domain-containing protein  VVACGKTSPRKLQLP   \n",
       "2398461                      FBD domain-containing protein  CIVNLGIPKTKTFPL   \n",
       "\n",
       "                             Prot_res_pep               aa  CA-Gly  Pro-MC  \\\n",
       "0           PIEZ1_CAEEL_0_MTVPPLLKSCVVKLL  MTVPPLLKSCVVKLL       0       4   \n",
       "10947     TTN1_CAEEL_4155_ATLKCIVVGTPLPDV  ATLKCIVVGTPLPDV       1       4   \n",
       "96923      CUP4_CAEEL_330_PLSVSEIPFCIKLIG  PLSVSEIPFCIKLIG       1       4   \n",
       "234219     IFT22_CAEEL_30_ILVLGPPKAGKTTLC  ILVLGPPKAGKTTLC       2       4   \n",
       "236662     VATA_CAEEL_150_IKHKILLPPSACGTI  IKHKILLPPSACGTI       1       4   \n",
       "...                                   ...              ...     ...     ...   \n",
       "2347550  Q9XW73_CAEEL_580_TPFIIVCIKSPSLLV  TPFIIVCIKSPSLLV       0       4   \n",
       "2360262   Q9XXF0_CAEEL_30_LTPIITAQCSKKPKG  LTPIITAQCSKKPKG       1       4   \n",
       "2360816  Q9XXI0_CAEEL_305_KKLRCELPITVGTIP  KKLRCELPITVGTIP       1       4   \n",
       "2390435  U4PR82_CAEEL_135_VVACGKTSPRKLQLP  VVACGKTSPRKLQLP       1       4   \n",
       "2398461  V6CK14_CAEEL_205_CIVNLGIPKTKTFPL  CIVNLGIPKTKTFPL       1       4   \n",
       "\n",
       "         CB  ...  S  NH3  Arg  MC  LastC  ManyH  SeCys  PyrroLys  Kclust  \\\n",
       "0        13  ...  1    2    0  56  False  False  False     False   14908   \n",
       "10947    12  ...  0    1    0  55  False  False  False     False   14908   \n",
       "96923    12  ...  0    1    0  55  False  False  False     False   14908   \n",
       "234219   11  ...  0    2    0  54   True  False  False     False   14908   \n",
       "236662   12  ...  0    2    0  55  False  False  False     False   14908   \n",
       "...      ..  ... ..  ...  ...  ..    ...    ...    ...       ...     ...   \n",
       "2347550  13  ...  0    1    0  56  False  False  False     False   14908   \n",
       "2360262  12  ...  0    3    0  55  False  False  False     False   14908   \n",
       "2360816  12  ...  0    2    4  55  False  False  False     False   14908   \n",
       "2390435  12  ...  0    2    4  55  False  False  False     False   14908   \n",
       "2398461  12  ...  0    2    0  55  False  False  False     False   14908   \n",
       "\n",
       "         Gene Names (primary)  \n",
       "0                      pezo-1  \n",
       "10947                   ttn-1  \n",
       "96923                   cup-4  \n",
       "234219                 ifta-2  \n",
       "236662                 vha-13  \n",
       "...                       ...  \n",
       "2347550                   NaN  \n",
       "2360262                   NaN  \n",
       "2360816                arrd-8  \n",
       "2390435                osgn-1  \n",
       "2398461                him-19  \n",
       "\n",
       "[95 rows x 32 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# indexed the dataframe df_pkl[ bravo] to give us peptides that have Kclust equal to 14908 only\n",
    "bravo = df_pkl[\"Kclust\"] == 14908\n",
    "df_pkl[bravo]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0fb01733-33de-4ea9-88dd-e73e03f7e53a",
   "metadata": {},
   "source": [
    "# Using the for loop to iterate through in order to get the excel file that contains number of GO Terms , start and without myos column\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9e0dd6b9-63fa-4559-8420-dc63301b9aee",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing: V6CK14_CAEEL\n",
      "Kclust 5042 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_0.xlsx\n",
      "Kclust 2458 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_5.xlsx\n",
      "Kclust 14342 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_10.xlsx\n",
      "Kclust 7670 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_15.xlsx\n",
      "Kclust 17018 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_20.xlsx\n",
      "Kclust 17018 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_25.xlsx\n",
      "Kclust 10197 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_30.xlsx\n",
      "Kclust 15469 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_35.xlsx\n",
      "Kclust 15469 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_40.xlsx\n",
      "Kclust 15873 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_45.xlsx\n",
      "Kclust 18014 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_50.xlsx\n",
      "Kclust 6137 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_55.xlsx\n",
      "Kclust 11387 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_60.xlsx\n",
      "Kclust 13096 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_65.xlsx\n",
      "Kclust 1768 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_70.xlsx\n",
      "Kclust 15472 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_75.xlsx\n",
      "Kclust 5057 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_80.xlsx\n",
      "Kclust 3554 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_85.xlsx\n",
      "Kclust 13885 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_90.xlsx\n",
      "Kclust 13167 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_95.xlsx\n",
      "Kclust 6327 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_100.xlsx\n",
      "Saved to pixel_V6CK14_CAEEL_105.xlsx\n",
      "Kclust 12999 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_110.xlsx\n",
      "Kclust 3077 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_115.xlsx\n",
      "Kclust 17871 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_120.xlsx\n",
      "Kclust 406 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_125.xlsx\n",
      "Kclust 7996 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_130.xlsx\n",
      "Saved to pixel_V6CK14_CAEEL_135.xlsx\n",
      "Kclust 11183 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_140.xlsx\n",
      "Kclust 18942 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_145.xlsx\n",
      "Kclust 19586 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_150.xlsx\n",
      "Saved to pixel_V6CK14_CAEEL_155.xlsx\n",
      "Kclust 18238 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_160.xlsx\n",
      "Kclust 2181 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_165.xlsx\n",
      "Kclust 2665 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_170.xlsx\n",
      "Kclust 5616 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_175.xlsx\n",
      "Kclust 19628 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_180.xlsx\n",
      "Kclust 16455 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_185.xlsx\n",
      "Kclust 5476 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_190.xlsx\n",
      "Kclust 19701 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_195.xlsx\n",
      "Kclust 5355 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_200.xlsx\n",
      "Kclust 14908 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_205.xlsx\n",
      "Kclust 2176 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_210.xlsx\n",
      "Kclust 16603 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_215.xlsx\n",
      "Kclust 11884 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_220.xlsx\n",
      "Kclust 6407 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_225.xlsx\n",
      "Kclust 2943 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_230.xlsx\n",
      "Kclust 3159 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_235.xlsx\n",
      "Saved to pixel_V6CK14_CAEEL_240.xlsx\n",
      "Kclust 2458 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_245.xlsx\n",
      "Kclust 1840 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_250.xlsx\n",
      "Kclust 15266 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_255.xlsx\n",
      "Kclust 18381 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_260.xlsx\n",
      "Kclust 511 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_265.xlsx\n",
      "Kclust 1913 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_270.xlsx\n",
      "Kclust 10265 does not have myosins.\n",
      "Saved to pixel_V6CK14_CAEEL_275.xlsx\n",
      "Kclust 560 does not have myosins.\n"
     ]
    },
    {
     "ename": "PermissionError",
     "evalue": "[Errno 13] Permission denied: 'pixel_V6CK14_CAEEL_280.xlsx'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mPermissionError\u001b[0m                           Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[7], line 38\u001b[0m\n\u001b[0;32m     36\u001b[0m \u001b[38;5;66;03m# Save the filtered Kclust group to Excel\u001b[39;00m\n\u001b[0;32m     37\u001b[0m output_filename \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpixel_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpeptide\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mrow[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mStart\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m.xlsx\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m---> 38\u001b[0m kclust_df\u001b[38;5;241m.\u001b[39mto_excel(output_filename, index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n\u001b[0;32m     39\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mSaved to \u001b[39m\u001b[38;5;132;01m{\u001b[39;00moutput_filename\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     41\u001b[0m output_rows\u001b[38;5;241m.\u001b[39mappend(result_row)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\util\\_decorators.py:333\u001b[0m, in \u001b[0;36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    327\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(args) \u001b[38;5;241m>\u001b[39m num_allow_args:\n\u001b[0;32m    328\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[0;32m    329\u001b[0m         msg\u001b[38;5;241m.\u001b[39mformat(arguments\u001b[38;5;241m=\u001b[39m_format_argument_list(allow_args)),\n\u001b[0;32m    330\u001b[0m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m,\n\u001b[0;32m    331\u001b[0m         stacklevel\u001b[38;5;241m=\u001b[39mfind_stack_level(),\n\u001b[0;32m    332\u001b[0m     )\n\u001b[1;32m--> 333\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m func(\u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\core\\generic.py:2417\u001b[0m, in \u001b[0;36mNDFrame.to_excel\u001b[1;34m(self, excel_writer, sheet_name, na_rep, float_format, columns, header, index, index_label, startrow, startcol, engine, merge_cells, inf_rep, freeze_panes, storage_options, engine_kwargs)\u001b[0m\n\u001b[0;32m   2404\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mio\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mformats\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mexcel\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m ExcelFormatter\n\u001b[0;32m   2406\u001b[0m formatter \u001b[38;5;241m=\u001b[39m ExcelFormatter(\n\u001b[0;32m   2407\u001b[0m     df,\n\u001b[0;32m   2408\u001b[0m     na_rep\u001b[38;5;241m=\u001b[39mna_rep,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   2415\u001b[0m     inf_rep\u001b[38;5;241m=\u001b[39minf_rep,\n\u001b[0;32m   2416\u001b[0m )\n\u001b[1;32m-> 2417\u001b[0m formatter\u001b[38;5;241m.\u001b[39mwrite(\n\u001b[0;32m   2418\u001b[0m     excel_writer,\n\u001b[0;32m   2419\u001b[0m     sheet_name\u001b[38;5;241m=\u001b[39msheet_name,\n\u001b[0;32m   2420\u001b[0m     startrow\u001b[38;5;241m=\u001b[39mstartrow,\n\u001b[0;32m   2421\u001b[0m     startcol\u001b[38;5;241m=\u001b[39mstartcol,\n\u001b[0;32m   2422\u001b[0m     freeze_panes\u001b[38;5;241m=\u001b[39mfreeze_panes,\n\u001b[0;32m   2423\u001b[0m     engine\u001b[38;5;241m=\u001b[39mengine,\n\u001b[0;32m   2424\u001b[0m     storage_options\u001b[38;5;241m=\u001b[39mstorage_options,\n\u001b[0;32m   2425\u001b[0m     engine_kwargs\u001b[38;5;241m=\u001b[39mengine_kwargs,\n\u001b[0;32m   2426\u001b[0m )\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\io\\formats\\excel.py:943\u001b[0m, in \u001b[0;36mExcelFormatter.write\u001b[1;34m(self, writer, sheet_name, startrow, startcol, freeze_panes, engine, storage_options, engine_kwargs)\u001b[0m\n\u001b[0;32m    941\u001b[0m     need_save \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[0;32m    942\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m--> 943\u001b[0m     writer \u001b[38;5;241m=\u001b[39m ExcelWriter(\n\u001b[0;32m    944\u001b[0m         writer,\n\u001b[0;32m    945\u001b[0m         engine\u001b[38;5;241m=\u001b[39mengine,\n\u001b[0;32m    946\u001b[0m         storage_options\u001b[38;5;241m=\u001b[39mstorage_options,\n\u001b[0;32m    947\u001b[0m         engine_kwargs\u001b[38;5;241m=\u001b[39mengine_kwargs,\n\u001b[0;32m    948\u001b[0m     )\n\u001b[0;32m    949\u001b[0m     need_save \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[0;32m    951\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\io\\excel\\_openpyxl.py:61\u001b[0m, in \u001b[0;36mOpenpyxlWriter.__init__\u001b[1;34m(self, path, engine, date_format, datetime_format, mode, storage_options, if_sheet_exists, engine_kwargs, **kwargs)\u001b[0m\n\u001b[0;32m     57\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mopenpyxl\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mworkbook\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Workbook\n\u001b[0;32m     59\u001b[0m engine_kwargs \u001b[38;5;241m=\u001b[39m combine_kwargs(engine_kwargs, kwargs)\n\u001b[1;32m---> 61\u001b[0m \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m(\n\u001b[0;32m     62\u001b[0m     path,\n\u001b[0;32m     63\u001b[0m     mode\u001b[38;5;241m=\u001b[39mmode,\n\u001b[0;32m     64\u001b[0m     storage_options\u001b[38;5;241m=\u001b[39mstorage_options,\n\u001b[0;32m     65\u001b[0m     if_sheet_exists\u001b[38;5;241m=\u001b[39mif_sheet_exists,\n\u001b[0;32m     66\u001b[0m     engine_kwargs\u001b[38;5;241m=\u001b[39mengine_kwargs,\n\u001b[0;32m     67\u001b[0m )\n\u001b[0;32m     69\u001b[0m \u001b[38;5;66;03m# ExcelWriter replaced \"a\" by \"r+\" to allow us to first read the excel file from\u001b[39;00m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;66;03m# the file and later write to it\u001b[39;00m\n\u001b[0;32m     71\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mr+\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_mode:  \u001b[38;5;66;03m# Load from existing workbook\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\io\\excel\\_base.py:1246\u001b[0m, in \u001b[0;36mExcelWriter.__init__\u001b[1;34m(self, path, engine, date_format, datetime_format, mode, storage_options, if_sheet_exists, engine_kwargs)\u001b[0m\n\u001b[0;32m   1242\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_handles \u001b[38;5;241m=\u001b[39m IOHandles(\n\u001b[0;32m   1243\u001b[0m     cast(IO[\u001b[38;5;28mbytes\u001b[39m], path), compression\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcompression\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;28;01mNone\u001b[39;00m}\n\u001b[0;32m   1244\u001b[0m )\n\u001b[0;32m   1245\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(path, ExcelWriter):\n\u001b[1;32m-> 1246\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_handles \u001b[38;5;241m=\u001b[39m get_handle(\n\u001b[0;32m   1247\u001b[0m         path, mode, storage_options\u001b[38;5;241m=\u001b[39mstorage_options, is_text\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[0;32m   1248\u001b[0m     )\n\u001b[0;32m   1249\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_cur_sheet \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1251\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m date_format \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\pandas\\io\\common.py:882\u001b[0m, in \u001b[0;36mget_handle\u001b[1;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[0;32m    873\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(\n\u001b[0;32m    874\u001b[0m             handle,\n\u001b[0;32m    875\u001b[0m             ioargs\u001b[38;5;241m.\u001b[39mmode,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    878\u001b[0m             newline\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m    879\u001b[0m         )\n\u001b[0;32m    880\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    881\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m--> 882\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n\u001b[0;32m    883\u001b[0m     handles\u001b[38;5;241m.\u001b[39mappend(handle)\n\u001b[0;32m    885\u001b[0m \u001b[38;5;66;03m# Convert BytesIO or file objects passed with an encoding\u001b[39;00m\n",
      "\u001b[1;31mPermissionError\u001b[0m: [Errno 13] Permission denied: 'pixel_V6CK14_CAEEL_280.xlsx'"
     ]
    }
   ],
   "source": [
    "# peptide that I am interested in (non-myosin one now)\n",
    "peptides_of_interest = [\"V6CK14_CAEEL\"] \n",
    "\n",
    "# List to store final results\n",
    "output_rows = []\n",
    "\n",
    "# Keywords for myosins to exclude\n",
    "myosin_keywords = \"myo-3|myo-1|unc-54|myo-2\"\n",
    "\n",
    "for peptide in peptides_of_interest:\n",
    "    print(f\"Processing: {peptide}\")\n",
    "\n",
    "    # Filter entries matching the peptide\n",
    "    matching_entries = df_pkl[df_pkl[\"Entry name\"] == peptide]\n",
    "\n",
    "    for _, row in matching_entries.iterrows():\n",
    "        result_row = {\n",
    "            \"Start\": row[\"Start\"],\n",
    "            \"Kclust\": row[\"Kclust\"]\n",
    "        }\n",
    "\n",
    "        # Get all entries with the same Kclust\n",
    "        kclust_filter = df_pkl[\"Kclust\"] == row[\"Kclust\"]\n",
    "        kclust_df = df_pkl[kclust_filter]\n",
    "\n",
    "        # Filter out rows containing myosin-related genes\n",
    "        no_generic_filter = ~kclust_df[\"Gene Names (primary)\"].str.contains(\"somestuff\", na=False)\n",
    "        no_myosin_filter = ~kclust_df[\"Gene Names (primary)\"].str.contains(myosin_keywords, na=False)\n",
    "\n",
    "        if no_myosin_filter.sum() == no_generic_filter.sum():\n",
    "            print(f\"Kclust {row['Kclust']} does not have myosins.\")\n",
    "            result_row[\"Othermyos\"] = False\n",
    "        else:\n",
    "            result_row[\"Othermyos\"] = True\n",
    "\n",
    "        # Save the filtered Kclust group to Excel\n",
    "        output_filename = f\"pixel_{peptide}_{row['Start']}.xlsx\"\n",
    "        kclust_df.to_excel(output_filename, index=False)\n",
    "        print(f\"Saved to {output_filename}\")\n",
    "\n",
    "        output_rows.append(result_row)\n",
    "\n",
    "# Save summary results\n",
    "summary_df = pd.DataFrame(output_rows)\n",
    "summary_df.to_excel(\"resdf3.xlsx\", index=False)\n",
    "print(\"Summary saved to resdf3.xlsx\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d6d517e-c762-4703-a07e-c42a380d04f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "kclust_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "345671ca-34f0-455a-a97a-37beb4dd0bbe",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dataframe containg the column of interest- Gene Names (primary)\n",
    "df_pkl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c134b4e2-63c0-42f3-bf64-f299807a0ccb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running enrichment for Kclust 5042...\n",
      "Saved enrichment results to cluster_5042_GOs.xlsx\n",
      "Running enrichment for Kclust 2458...\n",
      "Saved enrichment results to cluster_2458_GOs.xlsx\n",
      "Running enrichment for Kclust 14342...\n",
      "Saved enrichment results to cluster_14342_GOs.xlsx\n",
      "Running enrichment for Kclust 7670...\n",
      "Saved enrichment results to cluster_7670_GOs.xlsx\n",
      "Running enrichment for Kclust 17018...\n",
      "Saved enrichment results to cluster_17018_GOs.xlsx\n",
      "Running enrichment for Kclust 10197...\n",
      "Saved enrichment results to cluster_10197_GOs.xlsx\n",
      "Running enrichment for Kclust 15469...\n",
      "Saved enrichment results to cluster_15469_GOs.xlsx\n",
      "Running enrichment for Kclust 15873...\n",
      "Saved enrichment results to cluster_15873_GOs.xlsx\n",
      "Running enrichment for Kclust 18014...\n",
      "Saved enrichment results to cluster_18014_GOs.xlsx\n",
      "Running enrichment for Kclust 6137...\n",
      "Saved enrichment results to cluster_6137_GOs.xlsx\n",
      "Running enrichment for Kclust 11387...\n",
      "Saved enrichment results to cluster_11387_GOs.xlsx\n",
      "Running enrichment for Kclust 13096...\n",
      "Saved enrichment results to cluster_13096_GOs.xlsx\n",
      "Running enrichment for Kclust 1768...\n",
      "Saved enrichment results to cluster_1768_GOs.xlsx\n",
      "Running enrichment for Kclust 15472...\n",
      "Saved enrichment results to cluster_15472_GOs.xlsx\n",
      "Running enrichment for Kclust 5057...\n",
      "Saved enrichment results to cluster_5057_GOs.xlsx\n",
      "Running enrichment for Kclust 3554...\n",
      "Saved enrichment results to cluster_3554_GOs.xlsx\n",
      "Running enrichment for Kclust 13885...\n",
      "Saved enrichment results to cluster_13885_GOs.xlsx\n",
      "Running enrichment for Kclust 13167...\n",
      "Saved enrichment results to cluster_13167_GOs.xlsx\n",
      "Running enrichment for Kclust 6327...\n",
      "Saved enrichment results to cluster_6327_GOs.xlsx\n",
      "Running enrichment for Kclust 12999...\n",
      "Saved enrichment results to cluster_12999_GOs.xlsx\n",
      "Running enrichment for Kclust 3077...\n",
      "Saved enrichment results to cluster_3077_GOs.xlsx\n",
      "Running enrichment for Kclust 17871...\n",
      "Saved enrichment results to cluster_17871_GOs.xlsx\n",
      "Running enrichment for Kclust 406...\n",
      "Saved enrichment results to cluster_406_GOs.xlsx\n",
      "Running enrichment for Kclust 7996...\n",
      "Saved enrichment results to cluster_7996_GOs.xlsx\n",
      "Running enrichment for Kclust 11183...\n",
      "Saved enrichment results to cluster_11183_GOs.xlsx\n",
      "Running enrichment for Kclust 18942...\n",
      "Saved enrichment results to cluster_18942_GOs.xlsx\n",
      "Running enrichment for Kclust 19586...\n",
      "Saved enrichment results to cluster_19586_GOs.xlsx\n",
      "Running enrichment for Kclust 18238...\n",
      "Saved enrichment results to cluster_18238_GOs.xlsx\n",
      "Running enrichment for Kclust 2181...\n",
      "Saved enrichment results to cluster_2181_GOs.xlsx\n",
      "Running enrichment for Kclust 2665...\n",
      "Saved enrichment results to cluster_2665_GOs.xlsx\n",
      "Running enrichment for Kclust 5616...\n",
      "Saved enrichment results to cluster_5616_GOs.xlsx\n",
      "Running enrichment for Kclust 19628...\n",
      "Saved enrichment results to cluster_19628_GOs.xlsx\n",
      "Running enrichment for Kclust 16455...\n",
      "Saved enrichment results to cluster_16455_GOs.xlsx\n",
      "Running enrichment for Kclust 5476...\n",
      "Saved enrichment results to cluster_5476_GOs.xlsx\n",
      "Running enrichment for Kclust 19701...\n",
      "Saved enrichment results to cluster_19701_GOs.xlsx\n",
      "Running enrichment for Kclust 5355...\n",
      "Saved enrichment results to cluster_5355_GOs.xlsx\n",
      "Running enrichment for Kclust 14908...\n",
      "Saved enrichment results to cluster_14908_GOs.xlsx\n",
      "Running enrichment for Kclust 2176...\n",
      "Saved enrichment results to cluster_2176_GOs.xlsx\n",
      "Running enrichment for Kclust 16603...\n",
      "Saved enrichment results to cluster_16603_GOs.xlsx\n",
      "Running enrichment for Kclust 11884...\n",
      "Saved enrichment results to cluster_11884_GOs.xlsx\n",
      "Running enrichment for Kclust 6407...\n",
      "Saved enrichment results to cluster_6407_GOs.xlsx\n",
      "Running enrichment for Kclust 2943...\n",
      "Saved enrichment results to cluster_2943_GOs.xlsx\n",
      "Running enrichment for Kclust 3159...\n",
      "Saved enrichment results to cluster_3159_GOs.xlsx\n",
      "Running enrichment for Kclust 1840...\n",
      "Saved enrichment results to cluster_1840_GOs.xlsx\n",
      "Running enrichment for Kclust 15266...\n",
      "Saved enrichment results to cluster_15266_GOs.xlsx\n",
      "Running enrichment for Kclust 18381...\n",
      "Saved enrichment results to cluster_18381_GOs.xlsx\n",
      "Running enrichment for Kclust 511...\n",
      "Saved enrichment results to cluster_511_GOs.xlsx\n",
      "Running enrichment for Kclust 1913...\n",
      "Saved enrichment results to cluster_1913_GOs.xlsx\n",
      "Running enrichment for Kclust 10265...\n",
      "Saved enrichment results to cluster_10265_GOs.xlsx\n",
      "Running enrichment for Kclust 560...\n",
      "Saved enrichment results to cluster_560_GOs.xlsx\n",
      "Running enrichment for Kclust 6259...\n",
      "Saved enrichment results to cluster_6259_GOs.xlsx\n",
      "Running enrichment for Kclust 14580...\n",
      "Saved enrichment results to cluster_14580_GOs.xlsx\n",
      "Running enrichment for Kclust 6771...\n",
      "Saved enrichment results to cluster_6771_GOs.xlsx\n",
      "Running enrichment for Kclust 1333...\n",
      "Saved enrichment results to cluster_1333_GOs.xlsx\n",
      "Running enrichment for Kclust 11699...\n",
      "Saved enrichment results to cluster_11699_GOs.xlsx\n",
      "Running enrichment for Kclust 8419...\n",
      "Saved enrichment results to cluster_8419_GOs.xlsx\n",
      "Running enrichment for Kclust 8232...\n",
      "Saved enrichment results to cluster_8232_GOs.xlsx\n",
      "Running enrichment for Kclust 15422...\n",
      "Saved enrichment results to cluster_15422_GOs.xlsx\n",
      "Running enrichment for Kclust 14570...\n",
      "Saved enrichment results to cluster_14570_GOs.xlsx\n",
      "Running enrichment for Kclust 1493...\n",
      "Saved enrichment results to cluster_1493_GOs.xlsx\n",
      "Running enrichment for Kclust 2933...\n",
      "Saved enrichment results to cluster_2933_GOs.xlsx\n",
      "Running enrichment for Kclust 727...\n",
      "Saved enrichment results to cluster_727_GOs.xlsx\n",
      "Running enrichment for Kclust 561...\n",
      "Saved enrichment results to cluster_561_GOs.xlsx\n",
      "Running enrichment for Kclust 11823...\n",
      "Saved enrichment results to cluster_11823_GOs.xlsx\n",
      "Running enrichment for Kclust 5871...\n",
      "Saved enrichment results to cluster_5871_GOs.xlsx\n",
      "Running enrichment for Kclust 5480...\n",
      "Saved enrichment results to cluster_5480_GOs.xlsx\n",
      "Running enrichment for Kclust 19868...\n",
      "Saved enrichment results to cluster_19868_GOs.xlsx\n",
      "Running enrichment for Kclust 3562...\n",
      "Saved enrichment results to cluster_3562_GOs.xlsx\n",
      "Running enrichment for Kclust 19878...\n",
      "Saved enrichment results to cluster_19878_GOs.xlsx\n",
      "Running enrichment for Kclust 1489...\n",
      "Saved enrichment results to cluster_1489_GOs.xlsx\n",
      "Running enrichment for Kclust 12265...\n",
      "Saved enrichment results to cluster_12265_GOs.xlsx\n",
      "Saved combined enrichment results to cluster_V6CK14_GOs.xlsx\n"
     ]
    }
   ],
   "source": [
    "from gprofiler import GProfiler\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Loaded the summary results\n",
    "summary_df = pd.read_excel(\"resdf3.xlsx\")\n",
    "\n",
    "# Initialized GProfiler and results list\n",
    "gp = GProfiler(return_dataframe=True)\n",
    "GOs = []\n",
    "\n",
    "# Filtered clusters without myosins\n",
    "no_myosin_clusters = summary_df[summary_df[\"Othermyos\"] == False][\"Kclust\"].unique()\n",
    "\n",
    "# Run enrichment analysis for each Kclust\n",
    "for kclust_id in no_myosin_clusters:\n",
    "    print(f\"Running enrichment for Kclust {kclust_id}...\")\n",
    "\n",
    "    # Getting genes in the current Kclust\n",
    "    subdf = df_pkl[df_pkl[\"Kclust\"] == kclust_id]\n",
    "    gene_list = list(subdf[\"Gene Names (primary)\"].dropna().unique())\n",
    "\n",
    "    output_go_file = f\"cluster_{kclust_id}_GOs.xlsx\"\n",
    "\n",
    "    # If no valid gene names, write an empty dataframe with headers\n",
    "    if not gene_list:\n",
    "        print(f\"No valid gene names for Kclust {kclust_id}. Writing empty file.\")\n",
    "        pd.DataFrame(columns=[\"Kclust\", \"qgenelist\"]).to_excel(output_go_file, index=False)\n",
    "        continue\n",
    "\n",
    "    # Run GO enrichment\n",
    "    results = gp.profile(organism='celegans', query=gene_list, no_evidences=False)\n",
    "\n",
    "    # Filtered out unwanted sources\n",
    "    results = results[~results[\"source\"].isin([\"TF\", \"MIRNA\", \"HP\"])]\n",
    "\n",
    "    # Added metadata\n",
    "    results[\"Kclust\"] = kclust_id\n",
    "    results[\"qgenelist\"] = str(gene_list)\n",
    "\n",
    "    # Saved individual result, even if was empty\n",
    "    results.to_excel(output_go_file, index=False)\n",
    "    print(f\"Saved enrichment results to {output_go_file}\")\n",
    "\n",
    "    # Added to list only if enrichment results exist\n",
    "    if not results.empty:\n",
    "        GOs.append(results)\n",
    "\n",
    "# Saved combined results (only non-empty ones)\n",
    "if GOs:\n",
    "    pd.concat(GOs).to_excel(\"cluster_V6CK14_GOs.xlsx\", index=False)\n",
    "    print(\"Saved combined enrichment results to cluster_V6CK14_GOs.xlsx\")\n",
    "else:\n",
    "    print(\"No enrichment results to combine.\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "dfc2dd51-8cd5-4d9d-9a49-998688bb3251",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>qgenelist</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0016020</td>\n",
       "      <td>membrane</td>\n",
       "      <td>1.564515e-12</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A lipid bilayer along with all the proteins a...</td>\n",
       "      <td>6920</td>\n",
       "      <td>154</td>\n",
       "      <td>128</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.831169</td>\n",
       "      <td>0.018497</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0110165]</td>\n",
       "      <td>[pbo-5, tax-2, ced-5, glp-1, unc-101, maco-1, ...</td>\n",
       "      <td>[[IBA, IEA], [IDA, IBA, IEA], [IBA, TAS], [IDA...</td>\n",
       "      <td>5042</td>\n",
       "      <td>['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0004888</td>\n",
       "      <td>transmembrane signaling receptor activity</td>\n",
       "      <td>1.852546e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Combining with an extracellular or intracellu...</td>\n",
       "      <td>821</td>\n",
       "      <td>112</td>\n",
       "      <td>33</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.294643</td>\n",
       "      <td>0.040195</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0038023]</td>\n",
       "      <td>[pbo-5, glp-1, dop-3, npr-8, lgc-19, npr-4, lg...</td>\n",
       "      <td>[[IEA], [ISS, IC], [IBA, IEA], [IEA], [IEA], [...</td>\n",
       "      <td>5042</td>\n",
       "      <td>['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0038023</td>\n",
       "      <td>signaling receptor activity</td>\n",
       "      <td>2.282523e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Receiving a signal and transmitting it in the...</td>\n",
       "      <td>989</td>\n",
       "      <td>112</td>\n",
       "      <td>36</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.321429</td>\n",
       "      <td>0.036400</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0060089]</td>\n",
       "      <td>[pbo-5, daf-12, glp-1, dop-3, npr-8, lgc-19, n...</td>\n",
       "      <td>[[IBA, IEA], [IDA, IBA], [ISS, IC], [IDA, IBA,...</td>\n",
       "      <td>5042</td>\n",
       "      <td>['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0060089</td>\n",
       "      <td>molecular transducer activity</td>\n",
       "      <td>2.282523e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A compound molecular function in which an eff...</td>\n",
       "      <td>989</td>\n",
       "      <td>112</td>\n",
       "      <td>36</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.321429</td>\n",
       "      <td>0.036400</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0003674]</td>\n",
       "      <td>[pbo-5, daf-12, glp-1, dop-3, npr-8, lgc-19, n...</td>\n",
       "      <td>[[IBA, IEA], [IDA, IBA], [ISS, IC], [IDA, IBA,...</td>\n",
       "      <td>5042</td>\n",
       "      <td>['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0015075</td>\n",
       "      <td>monoatomic ion transmembrane transporter activity</td>\n",
       "      <td>1.233801e-09</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Enables the transfer of an ion from one side ...</td>\n",
       "      <td>532</td>\n",
       "      <td>112</td>\n",
       "      <td>26</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.232143</td>\n",
       "      <td>0.048872</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0022857]</td>\n",
       "      <td>[pbo-5, tax-2, cng-1, spe-41, lgc-19, catp-5, ...</td>\n",
       "      <td>[[IDA, IEA], [IDA, IBA, IEA], [IBA], [IDA, IMP...</td>\n",
       "      <td>5042</td>\n",
       "      <td>['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005886</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>4.172496e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The membrane surrounding a cell that separate...</td>\n",
       "      <td>1730</td>\n",
       "      <td>106</td>\n",
       "      <td>28</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.264151</td>\n",
       "      <td>0.016185</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0016020, GO:0071944]</td>\n",
       "      <td>[sax-3, eat-4, pgp-1, pgp-3, sos-1, spe-9, vps...</td>\n",
       "      <td>[[IDA, IBA], [IEA], [IDA], [IDA, IEA], [IBA], ...</td>\n",
       "      <td>1489</td>\n",
       "      <td>['mlk-1', 'sax-3', 'sma-5', 'rme-8', 'glh-3', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>GO:BP</td>\n",
       "      <td>GO:0055085</td>\n",
       "      <td>transmembrane transport</td>\n",
       "      <td>4.228529e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The process in which a solute is transported ...</td>\n",
       "      <td>983</td>\n",
       "      <td>101</td>\n",
       "      <td>22</td>\n",
       "      <td>10936</td>\n",
       "      <td>0.217822</td>\n",
       "      <td>0.022380</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0006810, GO:0009987]</td>\n",
       "      <td>[eat-4, pgp-1, pgp-3, vha-13, kvs-4, pgp-5, cl...</td>\n",
       "      <td>[[ISS, IEA], [IBA, IEA], [IBA, IEA], [IBA, IEA...</td>\n",
       "      <td>1489</td>\n",
       "      <td>['mlk-1', 'sax-3', 'sma-5', 'rme-8', 'glh-3', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0003824</td>\n",
       "      <td>catalytic activity</td>\n",
       "      <td>4.900273e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Catalysis of a biochemical reaction at physio...</td>\n",
       "      <td>4722</td>\n",
       "      <td>105</td>\n",
       "      <td>61</td>\n",
       "      <td>11728</td>\n",
       "      <td>0.580952</td>\n",
       "      <td>0.012918</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0003674]</td>\n",
       "      <td>[mlk-1, sma-5, glh-3, hda-4, pgp-1, pgp-3, gcy...</td>\n",
       "      <td>[[IBA, IEA], [IBA, IEA], [IBA, IEA], [ISS, IEA...</td>\n",
       "      <td>1489</td>\n",
       "      <td>['mlk-1', 'sax-3', 'sma-5', 'rme-8', 'glh-3', ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005886</td>\n",
       "      <td>plasma membrane</td>\n",
       "      <td>8.603723e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The membrane surrounding a cell that separate...</td>\n",
       "      <td>1730</td>\n",
       "      <td>87</td>\n",
       "      <td>26</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.298851</td>\n",
       "      <td>0.015029</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0016020, GO:0071944]</td>\n",
       "      <td>[fmi-1, piki-1, sup-17, bli-3, scd-2, mua-3, c...</td>\n",
       "      <td>[[IEA], [IBA, IEA], [IDA, IBA, IEA], [IDA, IBA...</td>\n",
       "      <td>12265</td>\n",
       "      <td>['sca-1', 'fmi-1', 'piki-1', 'sup-17', 'max-2'...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0071944</td>\n",
       "      <td>cell periphery</td>\n",
       "      <td>3.465086e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"The broad region around and including the pla...</td>\n",
       "      <td>1879</td>\n",
       "      <td>87</td>\n",
       "      <td>26</td>\n",
       "      <td>12900</td>\n",
       "      <td>0.298851</td>\n",
       "      <td>0.013837</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0110165]</td>\n",
       "      <td>[fmi-1, piki-1, sup-17, bli-3, scd-2, mua-3, c...</td>\n",
       "      <td>[[IEA], [IBA, IEA], [IDA, IBA, IEA], [IDA, IBA...</td>\n",
       "      <td>12265</td>\n",
       "      <td>['sca-1', 'fmi-1', 'piki-1', 'sup-17', 'max-2'...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1054 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   source      native                                               name  \\\n",
       "0   GO:CC  GO:0016020                                           membrane   \n",
       "1   GO:MF  GO:0004888          transmembrane signaling receptor activity   \n",
       "2   GO:MF  GO:0038023                        signaling receptor activity   \n",
       "3   GO:MF  GO:0060089                      molecular transducer activity   \n",
       "4   GO:MF  GO:0015075  monoatomic ion transmembrane transporter activity   \n",
       "..    ...         ...                                                ...   \n",
       "55  GO:CC  GO:0005886                                    plasma membrane   \n",
       "56  GO:BP  GO:0055085                            transmembrane transport   \n",
       "59  GO:MF  GO:0003824                                 catalytic activity   \n",
       "0   GO:CC  GO:0005886                                    plasma membrane   \n",
       "3   GO:CC  GO:0071944                                     cell periphery   \n",
       "\n",
       "         p_value  significant  \\\n",
       "0   1.564515e-12         True   \n",
       "1   1.852546e-10         True   \n",
       "2   2.282523e-10         True   \n",
       "3   2.282523e-10         True   \n",
       "4   1.233801e-09         True   \n",
       "..           ...          ...   \n",
       "55  4.172496e-02         True   \n",
       "56  4.228529e-02         True   \n",
       "59  4.900273e-02         True   \n",
       "0   8.603723e-03         True   \n",
       "3   3.465086e-02         True   \n",
       "\n",
       "                                          description  term_size  query_size  \\\n",
       "0   \"A lipid bilayer along with all the proteins a...       6920         154   \n",
       "1   \"Combining with an extracellular or intracellu...        821         112   \n",
       "2   \"Receiving a signal and transmitting it in the...        989         112   \n",
       "3   \"A compound molecular function in which an eff...        989         112   \n",
       "4   \"Enables the transfer of an ion from one side ...        532         112   \n",
       "..                                                ...        ...         ...   \n",
       "55  \"The membrane surrounding a cell that separate...       1730         106   \n",
       "56  \"The process in which a solute is transported ...        983         101   \n",
       "59  \"Catalysis of a biochemical reaction at physio...       4722         105   \n",
       "0   \"The membrane surrounding a cell that separate...       1730          87   \n",
       "3   \"The broad region around and including the pla...       1879          87   \n",
       "\n",
       "    intersection_size  effective_domain_size  precision    recall    query  \\\n",
       "0                 128                  12900   0.831169  0.018497  query_1   \n",
       "1                  33                  11728   0.294643  0.040195  query_1   \n",
       "2                  36                  11728   0.321429  0.036400  query_1   \n",
       "3                  36                  11728   0.321429  0.036400  query_1   \n",
       "4                  26                  11728   0.232143  0.048872  query_1   \n",
       "..                ...                    ...        ...       ...      ...   \n",
       "55                 28                  12900   0.264151  0.016185  query_1   \n",
       "56                 22                  10936   0.217822  0.022380  query_1   \n",
       "59                 61                  11728   0.580952  0.012918  query_1   \n",
       "0                  26                  12900   0.298851  0.015029  query_1   \n",
       "3                  26                  12900   0.298851  0.013837  query_1   \n",
       "\n",
       "                     parents  \\\n",
       "0               [GO:0110165]   \n",
       "1               [GO:0038023]   \n",
       "2               [GO:0060089]   \n",
       "3               [GO:0003674]   \n",
       "4               [GO:0022857]   \n",
       "..                       ...   \n",
       "55  [GO:0016020, GO:0071944]   \n",
       "56  [GO:0006810, GO:0009987]   \n",
       "59              [GO:0003674]   \n",
       "0   [GO:0016020, GO:0071944]   \n",
       "3               [GO:0110165]   \n",
       "\n",
       "                                        intersections  \\\n",
       "0   [pbo-5, tax-2, ced-5, glp-1, unc-101, maco-1, ...   \n",
       "1   [pbo-5, glp-1, dop-3, npr-8, lgc-19, npr-4, lg...   \n",
       "2   [pbo-5, daf-12, glp-1, dop-3, npr-8, lgc-19, n...   \n",
       "3   [pbo-5, daf-12, glp-1, dop-3, npr-8, lgc-19, n...   \n",
       "4   [pbo-5, tax-2, cng-1, spe-41, lgc-19, catp-5, ...   \n",
       "..                                                ...   \n",
       "55  [sax-3, eat-4, pgp-1, pgp-3, sos-1, spe-9, vps...   \n",
       "56  [eat-4, pgp-1, pgp-3, vha-13, kvs-4, pgp-5, cl...   \n",
       "59  [mlk-1, sma-5, glh-3, hda-4, pgp-1, pgp-3, gcy...   \n",
       "0   [fmi-1, piki-1, sup-17, bli-3, scd-2, mua-3, c...   \n",
       "3   [fmi-1, piki-1, sup-17, bli-3, scd-2, mua-3, c...   \n",
       "\n",
       "                                            evidences  Kclust  \\\n",
       "0   [[IBA, IEA], [IDA, IBA, IEA], [IBA, TAS], [IDA...    5042   \n",
       "1   [[IEA], [ISS, IC], [IBA, IEA], [IEA], [IEA], [...    5042   \n",
       "2   [[IBA, IEA], [IDA, IBA], [ISS, IC], [IDA, IBA,...    5042   \n",
       "3   [[IBA, IEA], [IDA, IBA], [ISS, IC], [IDA, IBA,...    5042   \n",
       "4   [[IDA, IEA], [IDA, IBA, IEA], [IBA], [IDA, IMP...    5042   \n",
       "..                                                ...     ...   \n",
       "55  [[IDA, IBA], [IEA], [IDA], [IDA, IEA], [IBA], ...    1489   \n",
       "56  [[ISS, IEA], [IBA, IEA], [IBA, IEA], [IBA, IEA...    1489   \n",
       "59  [[IBA, IEA], [IBA, IEA], [IBA, IEA], [ISS, IEA...    1489   \n",
       "0   [[IEA], [IBA, IEA], [IDA, IBA, IEA], [IDA, IBA...   12265   \n",
       "3   [[IEA], [IBA, IEA], [IDA, IBA, IEA], [IDA, IBA...   12265   \n",
       "\n",
       "                                            qgenelist  \n",
       "0   ['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...  \n",
       "1   ['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...  \n",
       "2   ['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...  \n",
       "3   ['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...  \n",
       "4   ['pbo-5', 'tax-2', 'ced-5', 'daf-12', 'glp-1',...  \n",
       "..                                                ...  \n",
       "55  ['mlk-1', 'sax-3', 'sma-5', 'rme-8', 'glh-3', ...  \n",
       "56  ['mlk-1', 'sax-3', 'sma-5', 'rme-8', 'glh-3', ...  \n",
       "59  ['mlk-1', 'sax-3', 'sma-5', 'rme-8', 'glh-3', ...  \n",
       "0   ['sca-1', 'fmi-1', 'piki-1', 'sup-17', 'max-2'...  \n",
       "3   ['sca-1', 'fmi-1', 'piki-1', 'sup-17', 'max-2'...  \n",
       "\n",
       "[1054 rows x 18 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# This give the gene onotology terms of the proteins after being enrinched\n",
    "pd.concat(GOs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "72f5b70b-9988-4e53-9793-1089e01aa4d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf3= summary_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a1d9f929-1482-4373-9804-02ba7658ea14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Start</th>\n",
       "      <th>Kclust</th>\n",
       "      <th>Othermyos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>5042</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>2458</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>14342</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15</td>\n",
       "      <td>7670</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20</td>\n",
       "      <td>17018</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>370</td>\n",
       "      <td>19868</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>375</td>\n",
       "      <td>3562</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>380</td>\n",
       "      <td>19878</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>385</td>\n",
       "      <td>1489</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>390</td>\n",
       "      <td>12265</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>79 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Start  Kclust  Othermyos\n",
       "0       0    5042      False\n",
       "1       5    2458      False\n",
       "2      10   14342      False\n",
       "3      15    7670      False\n",
       "4      20   17018      False\n",
       "..    ...     ...        ...\n",
       "74    370   19868      False\n",
       "75    375    3562      False\n",
       "76    380   19878      False\n",
       "77    385    1489      False\n",
       "78    390   12265      False\n",
       "\n",
       "[79 rows x 3 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resdf3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "48395a1d-867f-4204-b8b2-a4d31e21ba36",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File cluster_2395_GOs.xlsx not found. Assigning 0 GO terms.\n",
      "File cluster_6399_GOs.xlsx not found. Assigning 0 GO terms.\n",
      "File cluster_1339_GOs.xlsx not found. Assigning 0 GO terms.\n",
      "File cluster_18949_GOs.xlsx not found. Assigning 0 GO terms.\n",
      "File cluster_18356_GOs.xlsx not found. Assigning 0 GO terms.\n",
      "Saved resdf4.xlsx with GO term counts.\n"
     ]
    }
   ],
   "source": [
    "# Used for loop to iterate over the resdf3 file and count the number of GO terms in each cluster file\n",
    "import numpy as np\n",
    "lengo = []\n",
    "for i, row in resdf3.iterrows():\n",
    "    Kclust = row[\"Kclust\"]\n",
    "    filename = f\"cluster_{Kclust}_GOs.xlsx\"\n",
    "    \n",
    "    # Assign 0 if file is not found, to avoid limiting computation\n",
    "    try:\n",
    "        godf = pd.read_excel(filename)\n",
    "        lengo.append(len(godf))\n",
    "    except FileNotFoundError:\n",
    "        print(f\"File {filename} not found. Assigning 0 GO terms.\")\n",
    "        lengo.append(0)\n",
    "\n",
    "resdf3[\"Number of GO term\"] = lengo\n",
    "resdf3.to_excel(\"resdf4.xlsx\", index=False)\n",
    "print(\"Saved resdf4.xlsx with GO term counts.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "533fdc7a-26bf-458d-b576-447afd2a598c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "70ab65cd-ca77-4856-8a2a-9606cb1644bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib qt\n",
    "\n",
    "fig,ax = plt.subplots(1, figsize=(16,10))\n",
    "ax.plot(resdf3[\"Start\"], resdf3[\"Number of GO term\"])\n",
    "\n",
    "font1 ={\"size\": 40}\n",
    "font2 = {\"size\": 20}\n",
    "\n",
    "ax.set_title(\"GO terms- function of sequence position(V6CK14)\", fontdict= font1)\n",
    "ax.set_xlabel(\"Start\", fontdict = font2)\n",
    "ax.set_ylabel(\"Number of GO term\", fontdict = font2)\n",
    "ax.tick_params(axis='both', which='major', labelsize=18)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c57f8813-6592-43af-b3e6-e1bb295d8414",
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "bd653e2e-b742-48ae-9f6c-2e3a787d53e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example bin size (I can choose to adjust this when I want)\n",
    "bin_size = 50\n",
    "\n",
    "# Bin amino acid positions (I binned the amino acid positions in order to accomodate all start ids opn the map)\n",
    "resdf3[\"AA_bin\"] = (resdf3[\"Start\"] // bin_size) * bin_size\n",
    "\n",
    "# Aggregate the number of GO terms per bin\n",
    "binned_data = resdf3.groupby(\"AA_bin\")[\"Number of GO term\"].sum().reset_index()\n",
    "\n",
    "#  Then Prepared 2D array for heatmap (1 row, many columns)\n",
    "heat_data = np.array([binned_data[\"Number of GO term\"].values])\n",
    "\n",
    "# Plot heatmap\n",
    "#plt.figure(figsize=(20, 3)) - this gave a smaller heat map\n",
    "fig,ax = plt.subplots(1, figsize=(12, 8))\n",
    "ax = sns.heatmap(\n",
    "    heat_data,\n",
    "    cmap=\"YlGnBu\",\n",
    "    cbar_kws={'label': 'Number of GO terms'},\n",
    "    xticklabels=binned_data[\"AA_bin\"].astype(str),\n",
    "    yticklabels=False\n",
    ")\n",
    "# Increasing the colorbar label (legend title) font size\n",
    "colorbar = ax.collections[0].colorbar\n",
    "colorbar.ax.set_ylabel('Number of GO terms', fontsize=25) \n",
    "\n",
    "\n",
    "# formatting the map to my choice settings\n",
    "plt.title(\"GO terms - function of sequence position (V6CK14)\", fontsize=40)\n",
    "plt.xlabel(f\"Amino Acid Position (bins of {bin_size})\", fontsize=20, fontweight=\"bold\" )\n",
    "plt.ylabel(\"\")\n",
    "ax.tick_params(axis='x', labelsize=12, rotation=90)\n",
    "plt.tight_layout()\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f0625fbd-80dd-4363-a6e2-684f26f8d0cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.patches import Patch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "861115c0-a2dc-44d3-8ced-c792462d061f",
   "metadata": {},
   "outputs": [],
   "source": [
    "columns_to_extract = [\n",
    "    'CA-Gly', 'Pro-MC', 'CB', 'CB-Pro', 'Carboxyl',\n",
    "    'Amide', 'His', 'Trp', 'Phe-Tyr', 'OH-Tyr', 'CG',\n",
    "    'CD', 'CE', 'OH', 'SH', 'S', 'NH3', 'Arg', 'MC'\n",
    "]\n",
    "\n",
    "def standardize_max(reference_data, data):\n",
    "    max_values = np.max(reference_data, axis=1)\n",
    "    min_values = np.min(reference_data, axis=1)\n",
    "    return [\n",
    "        (col - min_val) / (max_val - min_val) if max_val != min_val else col * 0\n",
    "        for col, max_val, min_val in zip(data, max_values, min_values)\n",
    "    ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a2b0b275-5f9b-4345-912a-d96e21ed7bbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_20_000 = pd.read_pickle('mergedbf_unusual_marked_with_20000cluster_label (1).pkl')\n",
    "df_20_000 = df_20_000.rename(columns={'Kclust':'cluster'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "8cfafae7-ca3f-4f76-b3e1-8210004d3d8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 560] # c.elegans 280 start(AA position) THE HIGHEST NUMBER OF GO TERMS of an amino acid position in highly enrinched region\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_V6CK14_CAEEL_280 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_560_pos_280'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "a91e3d51-d136-4da8-9a86-7ca5f2b75b85",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_clust = df_20_000\n",
    "clust = df_clust[df_clust.cluster == 2933] # c.elegans 340 start(AA position)/THE HIGHEST NUMBER OF GO TERMS of an amino acid position in less enrinched region\n",
    "full_proteome_data = [df_clust[col] for col in columns_to_extract]\n",
    "clust_data = standardize_max(full_proteome_data, [clust[col] for col in columns_to_extract])\n",
    "normalized_full = standardize_max(full_proteome_data, full_proteome_data)\n",
    "\n",
    "boxplot_data = []\n",
    "positions = []\n",
    "xtick_labels = []\n",
    "offset = 0.2\n",
    "\n",
    "for i, (feature, clust_col, full_col) in enumerate(zip(columns_to_extract, clust_data, normalized_full)):\n",
    "    base_pos = i\n",
    "    positions.append(base_pos - offset)\n",
    "    boxplot_data.append(clust_col)\n",
    "    positions.append(base_pos + offset)\n",
    "    boxplot_data.append(full_col)\n",
    "    xtick_labels.append(feature)\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(12, 8))\n",
    "\n",
    "colors = ['steelblue', 'gray'] * len(columns_to_extract)\n",
    "bp = ax.boxplot(\n",
    "    boxplot_data,\n",
    "    positions=positions,\n",
    "    widths=0.35,\n",
    "    patch_artist=True,\n",
    "    whis=(0, 100),\n",
    "    boxprops=dict(linewidth=1.2),\n",
    "    medianprops=dict(linewidth=1.5, color='red')\n",
    ")\n",
    "\n",
    "\n",
    "for patch, color in zip(bp['boxes'], colors):\n",
    "    patch.set_facecolor(color)\n",
    "\n",
    "whiskers = [True,True,False,False] * len(columns_to_extract)\n",
    "\n",
    "for patch, whisker in zip(bp['whiskers'], whiskers):\n",
    "    \n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "for patch, whisker in zip(bp['caps'], whiskers):\n",
    "    if whisker==False:\n",
    "        patch.remove()\n",
    "    else:\n",
    "        patch.set_color('steelblue')\n",
    "        patch.set_lw(2.0)\n",
    "\n",
    "ax.set_xticks(range(len(columns_to_extract)))\n",
    "ax.set_xticklabels(columns_to_extract, fontsize=20, rotation=90)\n",
    "ax.set_yticks(np.linspace(0, 1, 11))\n",
    "ax.set_ylabel(\"Normalized feature count\", fontsize=25)\n",
    "ax.set_xlabel(\"Functional groups or atoms\", fontsize=25)\n",
    "ax.tick_params(axis='y', labelsize=18)\n",
    "ax.yaxis.grid(True, linestyle='--', alpha=0.5)\n",
    "for spine in ax.spines.values():\n",
    "    spine.set_linewidth(2)\n",
    "ax.tick_params(width=2)\n",
    "\n",
    "legend_elements = [\n",
    "    Patch(facecolor='steelblue', label='Cluster'),\n",
    "    Patch(facecolor='gray', label='Proteome')\n",
    "]\n",
    "ax.legend(handles=legend_elements, loc='upper center', \n",
    "          #bbox_to_anchor=(0.5, 1.1),\n",
    "          ncol=2, fontsize=20)\n",
    "\n",
    "plt.tight_layout(pad=2)\n",
    "plt.title(\"Standardized Feature Boxplots for pixel_V6CK14_CAEEL_340 \", fontsize=22, fontweight='bold')\n",
    "filename='feat_clust_2933_pos_340'\n",
    "fig.savefig(filename+'.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba87df71-f3d0-450a-abdb-b80d9b10b5f1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
